{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating the Gene Table\n",
    "This notebook is copied from the [Pymodulon GitHub repository](https://github.com/SBRG/pymodulon/blob/master/docs/tutorials/creating_the_gene_table.ipynb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get information from GFF files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymodulon.gene_util import *\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "\n",
    "from os import path\n",
    "from pathlib import Path  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, download the FASTA and GFF files for your organism and its plasmids from NCBI."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter the location of all your GFF files here:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gff_files = [os.path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene/genomic.gff')]\n",
    "gff_files = [os.path.join('/Users/louxuwen/Desktop/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene/genomic.gff')]\n",
    "\n",
    "#gene_file = path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene/gene.csv') # Enter metadata filename here\n",
    "#Kegg_file = path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene/Kegg_ID.csv') # Enter metadata filename here\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following cell will convert all the GFF files into a single Pandas DataFrame for easy manipulation. Pseudogenes have multiple rows in a GFF file (one for each fragment), but only the first fragment will be kept."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>old_locus_tag</th>\n",
       "      <th>gene_product</th>\n",
       "      <th>ncbi_protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>locus_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>cg0001</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1575.0</td>\n",
       "      <td>+</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>None</td>\n",
       "      <td>CHROMOSOMAL REPLICATION INITIATOR PROTEIN</td>\n",
       "      <td>CAF18566.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0002</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>1594.0</td>\n",
       "      <td>1920.0</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical protein predicted by Glimmer</td>\n",
       "      <td>CAF18567.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0004</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>2292.0</td>\n",
       "      <td>3476.0</td>\n",
       "      <td>+</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>None</td>\n",
       "      <td>DNA POLYMERASE III%2C BETA SUBUNIT</td>\n",
       "      <td>CAF18568.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0005</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3585.0</td>\n",
       "      <td>4769.0</td>\n",
       "      <td>+</td>\n",
       "      <td>recF</td>\n",
       "      <td>None</td>\n",
       "      <td>DNA REPAIR AND GENETIC RECOMBINATION PROTEIN</td>\n",
       "      <td>CAF18569.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0006</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>4814.0</td>\n",
       "      <td>5302.0</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>CONSERVED HYPOTHETICAL PROTEIN</td>\n",
       "      <td>CAF18570.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3430</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3280996.0</td>\n",
       "      <td>3281295.0</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>conserved hypothetical protein</td>\n",
       "      <td>CAF19037.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3431</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3281276.0</td>\n",
       "      <td>3281677.0</td>\n",
       "      <td>-</td>\n",
       "      <td>rnpA</td>\n",
       "      <td>None</td>\n",
       "      <td>RNase P protein component</td>\n",
       "      <td>CAF19038.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3432</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3281717.0</td>\n",
       "      <td>3281860.0</td>\n",
       "      <td>-</td>\n",
       "      <td>rpmH</td>\n",
       "      <td>None</td>\n",
       "      <td>50S RIBOSOMAL PROTEIN L34</td>\n",
       "      <td>CAF19039.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3433</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3282127.0</td>\n",
       "      <td>3282348.0</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical protein predicted by Glimmer/Critica</td>\n",
       "      <td>CAF19040.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3434</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3282519.0</td>\n",
       "      <td>3282632.0</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical protein predicted by Glimmer/Critica</td>\n",
       "      <td>CAF19041.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3058 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            accession      start        end strand gene_name old_locus_tag  \\\n",
       "locus_tag                                                                    \n",
       "cg0001     BX927147.1        1.0     1575.0      +      dnaA          None   \n",
       "cg0002     BX927147.1     1594.0     1920.0      -      None          None   \n",
       "cg0004     BX927147.1     2292.0     3476.0      +      dnaN          None   \n",
       "cg0005     BX927147.1     3585.0     4769.0      +      recF          None   \n",
       "cg0006     BX927147.1     4814.0     5302.0      +      None          None   \n",
       "...               ...        ...        ...    ...       ...           ...   \n",
       "cg3430     BX927147.1  3280996.0  3281295.0      -      None          None   \n",
       "cg3431     BX927147.1  3281276.0  3281677.0      -      rnpA          None   \n",
       "cg3432     BX927147.1  3281717.0  3281860.0      -      rpmH          None   \n",
       "cg3433     BX927147.1  3282127.0  3282348.0      -      None          None   \n",
       "cg3434     BX927147.1  3282519.0  3282632.0      +      None          None   \n",
       "\n",
       "                                                gene_product ncbi_protein  \n",
       "locus_tag                                                                  \n",
       "cg0001             CHROMOSOMAL REPLICATION INITIATOR PROTEIN   CAF18566.1  \n",
       "cg0002             hypothetical protein predicted by Glimmer   CAF18567.1  \n",
       "cg0004                    DNA POLYMERASE III%2C BETA SUBUNIT   CAF18568.1  \n",
       "cg0005          DNA REPAIR AND GENETIC RECOMBINATION PROTEIN   CAF18569.1  \n",
       "cg0006                        CONSERVED HYPOTHETICAL PROTEIN   CAF18570.1  \n",
       "...                                                      ...          ...  \n",
       "cg3430                        conserved hypothetical protein   CAF19037.1  \n",
       "cg3431                             RNase P protein component   CAF19038.1  \n",
       "cg3432                             50S RIBOSOMAL PROTEIN L34   CAF19039.1  \n",
       "cg3433     hypothetical protein predicted by Glimmer/Critica   CAF19040.1  \n",
       "cg3434     hypothetical protein predicted by Glimmer/Critica   CAF19041.1  \n",
       "\n",
       "[3058 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keep_cols = ['accession','start','end','strand','gene_name','old_locus_tag','gene_product','ncbi_protein']\n",
    "\n",
    "DF_annot = gff2pandas(gff_files,index='locus_tag')\n",
    "DF_annot = DF_annot[keep_cols]\n",
    "\n",
    "DF_annot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "06/30 update: turns out there are two c. glutamicum in kegg and ncbi, used the cgb organism will get you the right locus tag and genes.\n",
    "\n",
    "Since the microarray data use a different kind of annotation, it might be necessary to provide add a column of the gene ID that they used to match this locus tag\n",
    "1. in gene_ID, the start and end position also indicates the direction. Solution is to add the direction of the strand, but start position will now always be smalle than the end position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gene_ID = pd.read_csv(gene_file, index_col=0).fillna(-1)\n",
    "#Kegg_ID = pd.read_csv(Kegg_file, index_col=0,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Kegg_ID"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To ensure that the gene index used is identical to the expression matrix, load in your data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM5197020_1</th>\n",
       "      <th>GSM5197020_2</th>\n",
       "      <th>GSM5197021_1</th>\n",
       "      <th>GSM5197021_2</th>\n",
       "      <th>GSM5197022_1</th>\n",
       "      <th>GSM5197022_2</th>\n",
       "      <th>GSM5197023_1</th>\n",
       "      <th>GSM5197023_2</th>\n",
       "      <th>GSM5197024_2</th>\n",
       "      <th>GSM5197025_2</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM5197943_1</th>\n",
       "      <th>GSM5197943_2</th>\n",
       "      <th>GSM5197944_1</th>\n",
       "      <th>GSM5197944_2</th>\n",
       "      <th>GSM5197945_1</th>\n",
       "      <th>GSM5197945_2</th>\n",
       "      <th>GSM5197946_1</th>\n",
       "      <th>GSM5197946_2</th>\n",
       "      <th>GSM5197947_1</th>\n",
       "      <th>GSM5197947_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>cg0001</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1540.986</td>\n",
       "      <td>2186.654</td>\n",
       "      <td>839.000</td>\n",
       "      <td>1872.000</td>\n",
       "      <td>1723.954</td>\n",
       "      <td>2059.085</td>\n",
       "      <td>...</td>\n",
       "      <td>1296.8247</td>\n",
       "      <td>826.9054</td>\n",
       "      <td>1150.1929</td>\n",
       "      <td>848.4191</td>\n",
       "      <td>362.9568</td>\n",
       "      <td>326.9718</td>\n",
       "      <td>366.9057</td>\n",
       "      <td>550.2436</td>\n",
       "      <td>416.097</td>\n",
       "      <td>444.4621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0002</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>176.9562</td>\n",
       "      <td>227.2580</td>\n",
       "      <td>238.8252</td>\n",
       "      <td>243.9302</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0004</th>\n",
       "      <td>4711.377</td>\n",
       "      <td>6665.822</td>\n",
       "      <td>6738.429</td>\n",
       "      <td>6371.215</td>\n",
       "      <td>5917.581</td>\n",
       "      <td>6109.358</td>\n",
       "      <td>4897.064</td>\n",
       "      <td>8340.172</td>\n",
       "      <td>6469.821</td>\n",
       "      <td>6613.652</td>\n",
       "      <td>...</td>\n",
       "      <td>1994.0833</td>\n",
       "      <td>1810.2925</td>\n",
       "      <td>2091.9613</td>\n",
       "      <td>2049.7641</td>\n",
       "      <td>1391.4229</td>\n",
       "      <td>665.8254</td>\n",
       "      <td>1790.5143</td>\n",
       "      <td>1346.9563</td>\n",
       "      <td>1324.000</td>\n",
       "      <td>1327.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0005</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>457.000</td>\n",
       "      <td>714.000</td>\n",
       "      <td>...</td>\n",
       "      <td>363.3212</td>\n",
       "      <td>309.6552</td>\n",
       "      <td>402.8218</td>\n",
       "      <td>417.6308</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>178.000</td>\n",
       "      <td>192.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0006</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>216.0000</td>\n",
       "      <td>224.0000</td>\n",
       "      <td>239.0000</td>\n",
       "      <td>267.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 853 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GSM5197020_1  GSM5197020_2  GSM5197021_1  GSM5197021_2  GSM5197022_1  \\\n",
       "cg0001         0.000         0.000         0.000         0.000      1540.986   \n",
       "cg0002         0.000         0.000         0.000         0.000         0.000   \n",
       "cg0004      4711.377      6665.822      6738.429      6371.215      5917.581   \n",
       "cg0005         0.000         0.000         0.000         0.000         0.000   \n",
       "cg0006         0.000         0.000         0.000         0.000         0.000   \n",
       "\n",
       "        GSM5197022_2  GSM5197023_1  GSM5197023_2  GSM5197024_2  GSM5197025_2  \\\n",
       "cg0001      2186.654       839.000      1872.000      1723.954      2059.085   \n",
       "cg0002         0.000         0.000         0.000         0.000         0.000   \n",
       "cg0004      6109.358      4897.064      8340.172      6469.821      6613.652   \n",
       "cg0005         0.000         0.000         0.000       457.000       714.000   \n",
       "cg0006         0.000         0.000         0.000         0.000         0.000   \n",
       "\n",
       "        ...  GSM5197943_1  GSM5197943_2  GSM5197944_1  GSM5197944_2  \\\n",
       "cg0001  ...     1296.8247      826.9054     1150.1929      848.4191   \n",
       "cg0002  ...      176.9562      227.2580      238.8252      243.9302   \n",
       "cg0004  ...     1994.0833     1810.2925     2091.9613     2049.7641   \n",
       "cg0005  ...      363.3212      309.6552      402.8218      417.6308   \n",
       "cg0006  ...      216.0000      224.0000      239.0000      267.0000   \n",
       "\n",
       "        GSM5197945_1  GSM5197945_2  GSM5197946_1  GSM5197946_2  GSM5197947_1  \\\n",
       "cg0001      362.9568      326.9718      366.9057      550.2436       416.097   \n",
       "cg0002        0.0000        0.0000        0.0000        0.0000         0.000   \n",
       "cg0004     1391.4229      665.8254     1790.5143     1346.9563      1324.000   \n",
       "cg0005        0.0000        0.0000        0.0000        0.0000       178.000   \n",
       "cg0006        0.0000        0.0000        0.0000        0.0000         0.000   \n",
       "\n",
       "        GSM5197947_2  \n",
       "cg0001      444.4621  \n",
       "cg0002        0.0000  \n",
       "cg0004     1327.0000  \n",
       "cg0005      192.0000  \n",
       "cg0006        0.0000  \n",
       "\n",
       "[5 rows x 853 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "log_tpm_file = os.path.join('/Users/louxuwen/Desktop/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/log_tpm.csv')\n",
    "\n",
    "#log_tpm_file = os.path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/log_tpm.csv')\n",
    "DF_log_tpm = pd.read_csv(log_tpm_file,index_col=0)\n",
    "DF_log_tpm.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that the genes are the same in the expression dataset as in the annotation dataframe. Mismatched genes are listed below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_gene = DF_annot.sort_index().index.tolist()\n",
    "tpm_gene = DF_log_tpm.sort_index().index.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing = []\n",
    "for i in DF_gene:\n",
    "    if i not in tpm_gene:\n",
    "        missing.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>old_locus_tag</th>\n",
       "      <th>gene_product</th>\n",
       "      <th>ncbi_protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>locus_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>cg0001</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1575.0</td>\n",
       "      <td>+</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>None</td>\n",
       "      <td>CHROMOSOMAL REPLICATION INITIATOR PROTEIN</td>\n",
       "      <td>CAF18566.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0002</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>1594.0</td>\n",
       "      <td>1920.0</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical protein predicted by Glimmer</td>\n",
       "      <td>CAF18567.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0004</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>2292.0</td>\n",
       "      <td>3476.0</td>\n",
       "      <td>+</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>None</td>\n",
       "      <td>DNA POLYMERASE III%2C BETA SUBUNIT</td>\n",
       "      <td>CAF18568.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0005</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3585.0</td>\n",
       "      <td>4769.0</td>\n",
       "      <td>+</td>\n",
       "      <td>recF</td>\n",
       "      <td>None</td>\n",
       "      <td>DNA REPAIR AND GENETIC RECOMBINATION PROTEIN</td>\n",
       "      <td>CAF18569.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0006</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>4814.0</td>\n",
       "      <td>5302.0</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>CONSERVED HYPOTHETICAL PROTEIN</td>\n",
       "      <td>CAF18570.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3430</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3280996.0</td>\n",
       "      <td>3281295.0</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>conserved hypothetical protein</td>\n",
       "      <td>CAF19037.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3431</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3281276.0</td>\n",
       "      <td>3281677.0</td>\n",
       "      <td>-</td>\n",
       "      <td>rnpA</td>\n",
       "      <td>None</td>\n",
       "      <td>RNase P protein component</td>\n",
       "      <td>CAF19038.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3432</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3281717.0</td>\n",
       "      <td>3281860.0</td>\n",
       "      <td>-</td>\n",
       "      <td>rpmH</td>\n",
       "      <td>None</td>\n",
       "      <td>50S RIBOSOMAL PROTEIN L34</td>\n",
       "      <td>CAF19039.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3433</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3282127.0</td>\n",
       "      <td>3282348.0</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical protein predicted by Glimmer/Critica</td>\n",
       "      <td>CAF19040.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg3434</th>\n",
       "      <td>BX927147.1</td>\n",
       "      <td>3282519.0</td>\n",
       "      <td>3282632.0</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>hypothetical protein predicted by Glimmer/Critica</td>\n",
       "      <td>CAF19041.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3047 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            accession      start        end strand gene_name old_locus_tag  \\\n",
       "locus_tag                                                                    \n",
       "cg0001     BX927147.1        1.0     1575.0      +      dnaA          None   \n",
       "cg0002     BX927147.1     1594.0     1920.0      -      None          None   \n",
       "cg0004     BX927147.1     2292.0     3476.0      +      dnaN          None   \n",
       "cg0005     BX927147.1     3585.0     4769.0      +      recF          None   \n",
       "cg0006     BX927147.1     4814.0     5302.0      +      None          None   \n",
       "...               ...        ...        ...    ...       ...           ...   \n",
       "cg3430     BX927147.1  3280996.0  3281295.0      -      None          None   \n",
       "cg3431     BX927147.1  3281276.0  3281677.0      -      rnpA          None   \n",
       "cg3432     BX927147.1  3281717.0  3281860.0      -      rpmH          None   \n",
       "cg3433     BX927147.1  3282127.0  3282348.0      -      None          None   \n",
       "cg3434     BX927147.1  3282519.0  3282632.0      +      None          None   \n",
       "\n",
       "                                                gene_product ncbi_protein  \n",
       "locus_tag                                                                  \n",
       "cg0001             CHROMOSOMAL REPLICATION INITIATOR PROTEIN   CAF18566.1  \n",
       "cg0002             hypothetical protein predicted by Glimmer   CAF18567.1  \n",
       "cg0004                    DNA POLYMERASE III%2C BETA SUBUNIT   CAF18568.1  \n",
       "cg0005          DNA REPAIR AND GENETIC RECOMBINATION PROTEIN   CAF18569.1  \n",
       "cg0006                        CONSERVED HYPOTHETICAL PROTEIN   CAF18570.1  \n",
       "...                                                      ...          ...  \n",
       "cg3430                        conserved hypothetical protein   CAF19037.1  \n",
       "cg3431                             RNase P protein component   CAF19038.1  \n",
       "cg3432                             50S RIBOSOMAL PROTEIN L34   CAF19039.1  \n",
       "cg3433     hypothetical protein predicted by Glimmer/Critica   CAF19040.1  \n",
       "cg3434     hypothetical protein predicted by Glimmer/Critica   CAF19041.1  \n",
       "\n",
       "[3047 rows x 8 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#test = DF_annot.sort_index().index == DF_log_tpm.sort_index().index\n",
    "#  some genes in the DF_annot does not exist in DF_log_tpm\n",
    "DF_annot = DF_annot.drop(missing, axis=0)\n",
    "DF_annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path  \n",
    "filepath = Path('/Users/louxuwen/Desktop//Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene_info.csv')  \n",
    "\n",
    "#filepath = Path('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene_info.csv')  \n",
    "filepath.parent.mkdir(parents=True, exist_ok=True)  \n",
    "DF_annot.to_csv(filepath)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## (Optional) KEGG and COGs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate nucleotide fasta files for CDS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter the location of all your fasta files here:\n",
    "https://www.genome.jp/brite/br08601+cgb\n",
    "https://rest.kegg.jp/link/cgl/pathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fasta_files = [os.path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene/genomic.fna')]\n",
    "#fasta_files = Path('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/Gene/genomic.fna')\n",
    "\n",
    "fasta_files\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following code generates CDS files using your FASTA and GFF3 files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "cds_list = []\n",
    "for fasta in fasta_files:\n",
    "    seq = SeqIO.read(fasta,'fasta')\n",
    "\n",
    "    # Get gene information for genes in this fasta file\n",
    "    df_genes = DF_annot[DF_annot.accession == seq.id]\n",
    "    \n",
    "    for i,row in df_genes.iterrows():\n",
    "        cds = seq[int(row.start)-1:int(row.end)] #Added int() heredue to errors\n",
    "        if row.strand == '-':\n",
    "            cds = seq[int(row.start)-1:int(row.end)].reverse_complement()  #Added int() heredue to errors\n",
    "        cds.id = row.name\n",
    "        cds.description = row.gene_name if pd.notnull(row.gene_name) else row.name\n",
    "        cds_list.append(cds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_list[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save the CDS file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cds_file = os.path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/gene','CDS.fna')\n",
    "SeqIO.write(cds_list, cds_file, 'fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run EggNOG Mapper\n",
    "1. Go to http://eggnog-mapper.embl.de/.\n",
    "1. Upload the CDS.fna file from your organism directory (within the sequence_files folder)\n",
    "1. Make sure to limit the taxonomy to the correct level\n",
    "1. After the job is submitted, you must follow the link in your email to run the job.\n",
    "1. Once the job completes (after ~4 hrs), download the annotations file.\n",
    "1. Save the annotation file\n",
    "\n",
    "06/30 update: done and complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get KEGG IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once you have the EggNOG annotations, load the annotation file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "eggnog_file = os.path.join('/Users/louxuwen/Desktop/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/annotations.tsv')\n",
    "#eggnog_file = os.path.join('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/annotations.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seed eggNOG ortholog</th>\n",
       "      <th>seed ortholog evalue</th>\n",
       "      <th>seed ortholog score</th>\n",
       "      <th>eggNOG OGs</th>\n",
       "      <th>Max Annotation Level</th>\n",
       "      <th>COG</th>\n",
       "      <th>Description</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>EC number</th>\n",
       "      <th>KEGG_orth</th>\n",
       "      <th>KEGG_pathway</th>\n",
       "      <th>KEGG_module</th>\n",
       "      <th>KEGG_reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>locus_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>cg0001</th>\n",
       "      <td>196627.cg0001</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>998.0</td>\n",
       "      <td>COG0593@1|root,COG0593@2|Bacteria,2GJKI@201174...</td>\n",
       "      <td>201174|Actinobacteria</td>\n",
       "      <td>L</td>\n",
       "      <td>it binds specifically double-stranded DNA at a...</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>GO:0000166,GO:0003674,GO:0003676,GO:0003677,GO...</td>\n",
       "      <td>-</td>\n",
       "      <td>ko:K02313</td>\n",
       "      <td>ko02020,ko04112,map02020,map04112</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03032,ko03036</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Bac_DnaA,Bac_DnaA_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0004</th>\n",
       "      <td>196627.cg0004</td>\n",
       "      <td>5.280000e-282</td>\n",
       "      <td>771.0</td>\n",
       "      <td>COG0592@1|root,COG0592@2|Bacteria,2GJK3@201174...</td>\n",
       "      <td>201174|Actinobacteria</td>\n",
       "      <td>L</td>\n",
       "      <td>Confers DNA tethering and processivity to DNA ...</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>GO:0005575,GO:0005576,GO:0005618,GO:0005623,GO...</td>\n",
       "      <td>2.7.7.7</td>\n",
       "      <td>ko:K02338</td>\n",
       "      <td>ko00230,ko00240,ko01100,ko03030,ko03430,ko0344...</td>\n",
       "      <td>M00260</td>\n",
       "      <td>R00375,R00376,R00377,R00378</td>\n",
       "      <td>RC02795</td>\n",
       "      <td>ko00000,ko00001,ko00002,ko01000,ko03032,ko03400</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>DNA_pol3_beta,DNA_pol3_beta_2,DNA_pol3_beta_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0005</th>\n",
       "      <td>196627.cg0005</td>\n",
       "      <td>2.060000e-279</td>\n",
       "      <td>764.0</td>\n",
       "      <td>COG1195@1|root,COG1195@2|Bacteria,2GJCS@201174...</td>\n",
       "      <td>201174|Actinobacteria</td>\n",
       "      <td>L</td>\n",
       "      <td>it is required for DNA replication and normal ...</td>\n",
       "      <td>recF</td>\n",
       "      <td>GO:0000731,GO:0005575,GO:0005622,GO:0005623,GO...</td>\n",
       "      <td>-</td>\n",
       "      <td>ko:K03629</td>\n",
       "      <td>ko03440,map03440</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03400</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>SMC_N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0006</th>\n",
       "      <td>196627.cg0006</td>\n",
       "      <td>1.900000e-115</td>\n",
       "      <td>331.0</td>\n",
       "      <td>COG5512@1|root,COG5512@2|Bacteria,2GNQ4@201174...</td>\n",
       "      <td>201174|Actinobacteria</td>\n",
       "      <td>S</td>\n",
       "      <td>Zn-ribbon-containing, possibly RNA-binding pro...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>DUF721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cg0007</th>\n",
       "      <td>196627.cg0007</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1348.0</td>\n",
       "      <td>COG0187@1|root,COG0187@2|Bacteria,2GKGP@201174...</td>\n",
       "      <td>201174|Actinobacteria</td>\n",
       "      <td>L</td>\n",
       "      <td>A type II topoisomerase that negatively superc...</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>GO:0003674,GO:0003824,GO:0003916,GO:0003918,GO...</td>\n",
       "      <td>5.99.1.3</td>\n",
       "      <td>ko:K02470</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko01000,ko03032,ko03400</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>DNA_gyraseB,DNA_gyraseB_C,HATPase_c,Toprim</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          seed eggNOG ortholog  seed ortholog evalue  seed ortholog score  \\\n",
       "locus_tag                                                                   \n",
       "cg0001           196627.cg0001          0.000000e+00                998.0   \n",
       "cg0004           196627.cg0004         5.280000e-282                771.0   \n",
       "cg0005           196627.cg0005         2.060000e-279                764.0   \n",
       "cg0006           196627.cg0006         1.900000e-115                331.0   \n",
       "cg0007           196627.cg0007          0.000000e+00               1348.0   \n",
       "\n",
       "                                                  eggNOG OGs  \\\n",
       "locus_tag                                                      \n",
       "cg0001     COG0593@1|root,COG0593@2|Bacteria,2GJKI@201174...   \n",
       "cg0004     COG0592@1|root,COG0592@2|Bacteria,2GJK3@201174...   \n",
       "cg0005     COG1195@1|root,COG1195@2|Bacteria,2GJCS@201174...   \n",
       "cg0006     COG5512@1|root,COG5512@2|Bacteria,2GNQ4@201174...   \n",
       "cg0007     COG0187@1|root,COG0187@2|Bacteria,2GKGP@201174...   \n",
       "\n",
       "            Max Annotation Level COG  \\\n",
       "locus_tag                              \n",
       "cg0001     201174|Actinobacteria   L   \n",
       "cg0004     201174|Actinobacteria   L   \n",
       "cg0005     201174|Actinobacteria   L   \n",
       "cg0006     201174|Actinobacteria   S   \n",
       "cg0007     201174|Actinobacteria   L   \n",
       "\n",
       "                                                 Description Gene Name  \\\n",
       "locus_tag                                                                \n",
       "cg0001     it binds specifically double-stranded DNA at a...      dnaA   \n",
       "cg0004     Confers DNA tethering and processivity to DNA ...      dnaN   \n",
       "cg0005     it is required for DNA replication and normal ...      recF   \n",
       "cg0006     Zn-ribbon-containing, possibly RNA-binding pro...         -   \n",
       "cg0007     A type II topoisomerase that negatively superc...      gyrB   \n",
       "\n",
       "                                                         GOs EC number  \\\n",
       "locus_tag                                                                \n",
       "cg0001     GO:0000166,GO:0003674,GO:0003676,GO:0003677,GO...         -   \n",
       "cg0004     GO:0005575,GO:0005576,GO:0005618,GO:0005623,GO...   2.7.7.7   \n",
       "cg0005     GO:0000731,GO:0005575,GO:0005622,GO:0005623,GO...         -   \n",
       "cg0006                                                     -         -   \n",
       "cg0007     GO:0003674,GO:0003824,GO:0003916,GO:0003918,GO...  5.99.1.3   \n",
       "\n",
       "           KEGG_orth                                       KEGG_pathway  \\\n",
       "locus_tag                                                                 \n",
       "cg0001     ko:K02313                  ko02020,ko04112,map02020,map04112   \n",
       "cg0004     ko:K02338  ko00230,ko00240,ko01100,ko03030,ko03430,ko0344...   \n",
       "cg0005     ko:K03629                                   ko03440,map03440   \n",
       "cg0006             -                                                  -   \n",
       "cg0007     ko:K02470                                                  -   \n",
       "\n",
       "          KEGG_module                KEGG_reaction KEGG_rclass  \\\n",
       "locus_tag                                                        \n",
       "cg0001              -                            -           -   \n",
       "cg0004         M00260  R00375,R00376,R00377,R00378     RC02795   \n",
       "cg0005              -                            -           -   \n",
       "cg0006              -                            -           -   \n",
       "cg0007              -                            -           -   \n",
       "\n",
       "                                                     BRITE KEGG_TC CAZy  \\\n",
       "locus_tag                                                                 \n",
       "cg0001                     ko00000,ko00001,ko03032,ko03036       -    -   \n",
       "cg0004     ko00000,ko00001,ko00002,ko01000,ko03032,ko03400       -    -   \n",
       "cg0005                             ko00000,ko00001,ko03400       -    -   \n",
       "cg0006                                                   -       -    -   \n",
       "cg0007                     ko00000,ko01000,ko03032,ko03400       -    -   \n",
       "\n",
       "          BiGG Reaction                                          PFAMs  \n",
       "locus_tag                                                               \n",
       "cg0001                -                            Bac_DnaA,Bac_DnaA_C  \n",
       "cg0004                -  DNA_pol3_beta,DNA_pol3_beta_2,DNA_pol3_beta_3  \n",
       "cg0005                -                                          SMC_N  \n",
       "cg0006                -                                         DUF721  \n",
       "cg0007                -     DNA_gyraseB,DNA_gyraseB_C,HATPase_c,Toprim  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DF_eggnog = pd.read_csv(eggnog_file,sep='\\t',skiprows=5,header=None) #changed 4 to 5 due to version update\n",
    "eggnog_cols = ['query_name','seed eggNOG ortholog','seed ortholog evalue','seed ortholog score',\n",
    "               'eggNOG OGs','Max Annotation Level','COG',\n",
    "               'Description', 'Gene Name', 'GOs','EC number',\n",
    "               'KEGG_orth','KEGG_pathway','KEGG_module','KEGG_reaction',\n",
    "               'KEGG_rclass','BRITE','KEGG_TC','CAZy','BiGG Reaction','PFAMs']  \n",
    "# \"predicted taxonomic group\" -> eggNOG OGs (orthologous groups)\n",
    "# \"predicted protein name\" -> max annotation level\n",
    "# \"Gene Ontology terms\" -> COG categroy\"\n",
    "# \"EC number\" -> \"Description\"\n",
    "# add \"gene name\"\n",
    "# add \"GOs\"\n",
    "# \"tax_scope\" -> PFAMs, the PFAMs database is now interpro. Not sure how useful this column is\n",
    "# delete \"eggNOG OGs\" it's in previous column, \"bestOF _deprecated\"\n",
    "# delete 'COG', it's added to other places\n",
    "\n",
    "\n",
    "#deleted 'eggNOG free text description' column at the end due to bug\n",
    "\n",
    "DF_eggnog.columns = eggnog_cols\n",
    "\n",
    "# Strip last three rows as they are comments\n",
    "DF_eggnog = DF_eggnog.iloc[:-3]\n",
    "\n",
    "# Set locus tag as index\n",
    "DF_eggnog = DF_eggnog.set_index('query_name')\n",
    "DF_eggnog.index.name = 'locus_tag'\n",
    "\n",
    "DF_eggnog.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will pull the KEGG information from the eggNOG file, including orthology, pathway, module, and reactions for each gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>database</th>\n",
       "      <th>kegg_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cg0006</td>\n",
       "      <td>KEGG_orth</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>cg0008</td>\n",
       "      <td>KEGG_orth</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>cg0010</td>\n",
       "      <td>KEGG_orth</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>cg0012</td>\n",
       "      <td>KEGG_orth</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>cg0013</td>\n",
       "      <td>KEGG_orth</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_id   database kegg_id\n",
       "3  cg0006  KEGG_orth       -\n",
       "5  cg0008  KEGG_orth       -\n",
       "7  cg0010  KEGG_orth       -\n",
       "8  cg0012  KEGG_orth       -\n",
       "9  cg0013  KEGG_orth       -"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DF_kegg = DF_eggnog[['KEGG_orth','KEGG_pathway','KEGG_module','KEGG_reaction']]\n",
    "\n",
    "# Melt dataframe\n",
    "DF_kegg = DF_kegg.reset_index().melt(id_vars='locus_tag') \n",
    "\n",
    "# Remove null values\n",
    "DF_kegg = DF_kegg[DF_kegg.value.notnull()]\n",
    "\n",
    "# Split comma-separated values into their own rows\n",
    "list2struct = []\n",
    "for name,row in DF_kegg.iterrows():\n",
    "    for val in row.value.split(','):\n",
    "        list2struct.append([row.locus_tag,row.variable,val])\n",
    "\n",
    "DF_kegg = pd.DataFrame(list2struct,columns=['gene_id','database','kegg_id'])\n",
    "\n",
    "# Remove ko entries, as only map entries are searchable in KEGG pathway\n",
    "DF_kegg = DF_kegg[~DF_kegg.kegg_id.str.startswith('ko')]\n",
    "\n",
    "DF_kegg.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save KEGG information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "filepath = Path('/Users/louxuwen/Desktop/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/kegg_mapping.csv')  \n",
    "\n",
    "#filepath = Path('/home/amy/Documents/GitHub/modulome-C_Glutamicum_Microarray_clean/7_characterizing_imodulons/Data/kegg_mapping.csv')  \n",
    "filepath.parent.mkdir(parents=True, exist_ok=True)  \n",
    "DF_kegg.to_csv(filepath)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save COGs to annotation dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_annot['COG'] = DF_eggnog.COG\n",
    "\n",
    "# Make sure COG only has one entry per gene\n",
    "DF_annot['COG'] = [item[0] if isinstance(item,str) else item for item in DF_annot['COG']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Uniprot ID mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The ``uniprot_id_mapping`` function is a python wrapper for the [Uniprot ID mapping tool](https://www.uniprot.org/uploadlists/). Use ``input_id=P_REFSEQ_AC`` if the FASTA/GFF files are from RefSeq, and ``input_id=EMBL`` if the files are from Genbank.\n",
    "\n",
    "06/30 completed but no results found\n",
    "\n",
    "07/03 cannot access using uniprot_id_mapping, used url access and obtained ~20 hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "mapping_uniprot = uniprot_id_mapping(DF_annot.ncbi_protein.fillna(''),input_id='P_REFSEQ_AC',output_id='ACC', input_name='ncbi_protein',output_name='uniprot')\n",
    "mapping_uniprot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAF18566.1\tQ8NUD8\n",
      "CAF18569.1\tQ6M8X7\n",
      "CAF18570.1\tQ8NUD4\n",
      "CAF18580.1\tQ6M8W7\n",
      "CAF18608.1\tQ8NU99\n",
      "CAF18609.1\tQ8NU98\n",
      "CAF18610.1\tQ8NU97\n",
      "CAF18640.1\tP46396\n",
      "CAF18652.1\tQ9RHM6\n",
      "CAF18653.1\tQ79VJ4\n",
      "CAF18654.1\tQ79VJ3\n",
      "CAF18655.1\tQ9RHM3\n",
      "CAF18656.1\tQ79VJ2\n",
      "CAF18657.1\tQ79VJ1\n",
      "CAF18658.1\tQ79VJ0\n",
      "CAF18681.1\tQ9X713\n",
      "CAF18682.1\tQ9X712\n",
      "CAF18684.1\tQ8NU33\n",
      "CAF18702.1\tQ9X4N0\n",
      "CAF18731.1\tQ8NTY7\n",
      "CAF18758.1\tQ8NTW4\n",
      "CAF18782.1\tQ8NTU1\n",
      "CAF18789.1\tQ8NTT4\n",
      "CAF18807.1\tQ8NTR6\n",
      "CAF18814.1\tQ8NTQ9\n",
      "['CAF18566.1', 'CAF18567.1', 'CAF18568.1', 'CAF18569.1', 'CAF18570.1', 'CAF18571.1', 'CAF18572.1', 'CAF18573.1', 'CAF18574.1', 'CAF18575.1', 'CAF18576.1', 'CAF18577.1', 'CAF18578.1', 'CAF18579.1', 'CAF18580.1', 'CAF18581.1', 'CAF18582.1', 'CAF18583.1', 'CAF18584.1', 'CAF18585.1', 'CAF18586.1', 'CAF18587.1', 'CAF18588.1', 'CAF18589.1', 'CAF18590.1', 'CAF18591.1', 'CAF18592.1', 'CAF18593.1', 'CAF18594.1', 'CAF18595.1', 'CAF18596.1', 'CAF18597.1', 'CAF18598.1', 'CAF18599.1', 'CAF18600.1', 'CAF18601.1', 'CAF18602.1', 'CAF18603.1', 'CAF18605.1', 'CAF18604.1', 'CAF18606.1', 'CAF18607.1', 'CAF18608.1', 'CAF18609.1', 'CAF18610.1', 'CAF18611.1', 'CAF18612.1', 'CAF18613.1', 'CAF18614.1', 'CAF18615.1', 'CAF18616.1', 'CAF18617.1', 'CAF18619.1', 'CAF18618.1', 'CAF18620.1', 'CAF18621.1', 'CAF18622.1', 'CAF18623.1', 'CAF18624.1', 'CAF18625.1', 'CAF18626.1', 'CAF18627.1', 'CAF18628.1', 'CAF18629.1', 'CAF18630.1', 'CAF18631.1', 'CAF18632.1', 'CAF18633.1', 'CAF18634.1', 'CAF18635.1', 'CAF18636.1', 'CAF18637.1', 'CAF18638.1', 'CAF18639.1', 'CAF18640.1', 'CAF18641.1', 'CAF18642.1', 'CAF18643.1', 'CAF18644.1', 'CAF18645.1', 'CAF18646.1', 'CAF18647.1', 'CAF18648.1', 'CAF18649.1', 'CAF18650.1', 'CAF18651.1', 'CAF18652.1', 'CAF18653.1', 'CAF18654.1', 'CAF18655.1', 'CAF18656.1', 'CAF18657.1', 'CAF18658.1', 'CAF18659.1', 'CAF18660.1', 'CAF18661.1', 'CAF18662.1', 'CAF18663.1', 'CAF18664.1', 'CAF18665.1', 'CAF18666.1', 'CAF18667.1', 'CAF18668.1', 'CAF18669.1', 'CAF18670.1', 'CAF18671.1', 'CAF18672.1', 'CAF18673.1', 'CAF18674.1', 'CAF18675.1', 'CAF18676.1', 'CAF18677.1', 'CAF18678.1', 'CAF18679.1', 'CAF18680.1', 'CAF18681.1', 'CAF18682.1', 'CAF18683.1', 'CAF18684.1', 'CAF18685.1', 'CAF18686.1', 'CAF18687.1', 'CAF18688.1', 'CAF18689.1', 'CAF18690.1', 'CAF18691.1', 'CAF18692.1', 'CAF18693.1', 'CAF18694.1', 'CAF18695.1', 'CAF18696.1', 'CAF18697.1', 'CAF18698.1', 'CAF18699.1', 'CAF18700.1', 'CAF18701.1', 'CAF18702.1', 'CAF18703.1', 'CAF18704.1', 'CAF18705.1', 'CAF18706.1', 'CAF18707.1', 'CAF18708.1', 'CAF18709.1', 'CAF18710.1', 'CAF18711.1', 'CAF18712.1', 'CAF18713.1', 'CAF18714.1', 'CAF18715.1', 'CAF18716.1', 'CAF18717.1', 'CAF18718.1', 'CAF18719.1', 'CAF18720.1', 'CAF18721.1', 'CAF18722.1', 'CAF18723.1', 'CAF18724.1', 'CAF18725.1', 'CAF18726.1', 'CAF18727.1', 'CAF18728.1', 'CAF18729.1', 'CAF18730.1', 'CAF18731.1', 'CAF18732.1', 'CAF18733.1', 'CAF18734.1', 'CAF18735.1', 'CAF18736.1', 'CAF18737.1', 'CAF18738.1', 'CAF18739.1', 'CAF18740.1', 'CAF18741.1', 'CAF18742.1', 'CAF18743.1', 'CAF18744.1', 'CAF18745.1', 'CAF18746.1', 'CAF18747.1', 'CAF18748.1', 'CAF18749.1', 'CAF18750.1', 'CAF18751.1', 'CAF18752.1', 'CAF18753.1', 'CAF18754.1', 'CAF18755.1', 'CAF18756.1', 'CAF18757.1', 'CAF18758.1', 'CAF18759.1', 'CAF18760.1', 'CAF18761.1', 'CAF18762.1', 'CAF18763.1', 'CAF18764.1', 'CAF18765.1', 'CAF18766.1', 'CAF18767.1', 'CAF18768.1', 'CAF18769.1', 'CAF18770.1', 'CAF18771.1', 'CAF18772.1', 'CAF18773.1', 'CAF18774.1', 'CAF18775.1', 'CAF18776.1', 'CAF18777.1', 'CAF18778.1', 'CAF18779.1', 'CAF18780.1', 'CAF18781.1', 'CAF18782.1', 'CAF18783.1', 'CAF18784.1', 'CAF18785.1', 'CAF18786.1', 'CAF18787.1', 'CAF18788.1', 'CAF18789.1', 'CAF18790.1', 'CAF18791.1', 'CAF18792.1', 'CAF18793.1', 'CAF18794.1', 'CAF18795.1', 'CAF18796.1', 'CAF18797.1', 'CAF18798.1', 'CAF18799.1', 'CAF18800.1', 'CAF18801.1', 'CAF18802.1', 'CAF18803.1', 'CAF18804.1', 'CAF18805.1', 'CAF18806.1', 'CAF18807.1', 'CAF18808.1', 'CAF18809.1', 'CAF18810.1', 'CAF18811.1', 'CAF18812.1', 'CAF18813.1', 'CAF18814.1', 'CAF18815.1', 'CAF18816.1', 'CAF18817.1', 'CAF18818.1', 'CAF18819.1', 'CAF18820.1', 'CAF18821.1', 'CAF18822.1', 'CAF18823.1', 'CAF18824.1', 'CAF18825.1', 'CAF18826.1', 'CAF18827.1', 'CAF18828.1', 'CAF18829.1', 'CAF18830.1', 'CAF18831.1', 'CAF18832.1', 'CAF18833.1', 'CAF18834.1', 'CAF18835.1', 'CAF18836.1', 'CAF18837.1', 'CAF18838.1', 'CAF18839.1', 'CAF18840.1', 'CAF18841.1', 'CAF18842.1', 'CAF18843.1', 'CAF18844.1', 'CAF18845.1', 'CAF18846.1', 'CAF18847.1', 'CAF18848.1', 'CAF18849.1', 'CAF18850.1', 'CAF18851.1', 'CAF18852.1', 'CAF18853.1', 'CAF18854.1', 'CAF18855.1', 'CAF18856.1', 'CAF18857.1', 'CAF18858.1', 'CAF18859.1', 'CAF18860.1', 'CAF18861.1', 'CAF18862.1', 'CAF18863.1', 'CAF18864.1', 'CAF18865.1', 'CAF18866.1', 'CAF18867.1', 'CAF18868.1', 'CAF18869.1', 'CAF18870.1', 'CAF18871.1', 'CAF18872.1', 'CAF18873.1', 'CAF18874.1', 'CAF18875.1', 'CAF18876.1', 'CAF18877.1', 'CAF18878.1', 'CAF18879.1', 'CAF18880.1', 'CAF18881.1', 'CAF18883.1', 'CAF18882.1', 'CAF18884.1', 'CAF18885.1', 'CAF18886.1', 'CAF18887.1', 'CAF18888.1', 'CAF18889.1', 'CAF18890.1', 'CAF18891.1', 'CAF18892.1', 'CAF18893.1', 'CAF18894.1', 'CAF18895.1', 'CAF18896.1', 'CAF18897.1', 'CAF18898.1', 'CAF18899.1', 'CAF18900.1', 'CAF18901.1', 'CAF18902.1', 'CAF19046.1', 'CAF19047.1', 'CAF19048.1', 'CAF19049.1', 'CAF19050.1', 'CAF19051.1', 'CAF19052.1', 'CAF19053.1', 'CAF19054.1', 'CAF19055.1', 'CAF19056.1', 'CAF19057.1', 'CAF19058.1', 'CAF19059.1', 'CAF19060.1', 'CAF19061.1', 'CAF19062.1', 'CAF19063.1', 'CAF19064.1', 'CAF19065.1', 'CAF19066.1', 'CAF19067.1', 'CAF19068.1', 'CAF19069.1', 'CAF19070.1', 'CAF19071.1', 'CAF19072.1', 'CAF19073.1', 'CAF19074.1', 'CAF19075.1', 'CAF19076.1', 'CAF19077.1', 'CAF19078.1', 'CAF19079.1', 'CAF19080.1', 'CAF19081.1', 'CAF19082.1', 'CAF19083.1', 'CAF19084.1', 'CAF19085.1', 'CAF19086.1', 'CAF19087.1', 'CAF19089.1', 'CAF19088.1', 'CAF19090.1', 'CAF19091.1', 'CAF19092.1', 'CAF19093.1', 'CAF19094.1', 'CAF19095.1', 'CAF19096.1', 'CAF19097.1', 'CAF19098.1', 'CAF19099.1', 'CAF19100.1', 'CAF19101.1', 'CAF19102.1', 'CAF19103.1', 'CAF19104.1', 'CAF19105.1', 'CAF19106.1', 'CAF19107.1', 'CAF19108.1', 'CAF19109.1', 'CAF19110.1', 'CAF19111.1', 'CAF19112.1', 'CAF19113.1', 'CAF19114.1', 'CAF19115.1', 'CAF19116.1', 'CAF19117.1', 'CAF19118.1', 'CAF19119.1', 'CAF19120.1', 'CAF19121.1', 'CAF19122.1', 'CAF19123.1', 'CAF19124.1', 'CAF19125.1', 'CAF19126.1', 'CAF19127.1', 'CAF19128.1', 'CAF19129.1', 'CAF19130.1', 'CAF19131.1', 'CAF19132.1', 'CAF19133.1', 'CAF19134.1', 'CAF19135.1', 'CAF19136.1', 'CAF19137.1', 'CAF19138.1', 'CAF19139.1', 'CAF19140.1', 'CAF19141.1', 'CAF19142.1', 'CAF19143.1', 'CAF19144.1', 'CAF19145.1', 'CAF19146.1', 'CAF19147.1', 'CAF19148.1', 'CAF19149.1', 'CAF19150.1', 'CAF19151.1', 'CAF19152.1', 'CAF19153.1', 'CAF19154.1', 'CAF19155.1', 'CAF19156.1', 'CAF19157.1', 'CAF19158.1', 'CAF19159.1', 'CAF19160.1', 'CAF19161.1', 'CAF19162.1', 'CAF19163.1', 'CAF19164.1', 'CAF19165.1', 'CAF19166.1', 'CAF19167.1', 'CAF19168.1', 'CAF19169.1', 'CAF19170.1', 'CAF19171.1', 'CAF19172.1', 'CAF19173.1', 'CAF19174.1', 'CAF19175.1', 'CAF19176.1', 'CAF19177.1', 'CAF19179.1', 'CAF19178.1', 'CAF19180.1', 'CAF19181.1', 'CAF19182.1', 'CAF19183.1', 'CAF19184.1', 'CAF19185.1', 'CAF19186.1', 'CAF19187.1', 'CAF19188.1', 'CAF19189.1', 'CAF19190.1', 'CAF19191.1', 'CAF19192.1', 'CAF19193.1', 'CAF19194.1', 'CAF19195.1', 'CAF19196.1', 'CAF19197.1', 'CAF19198.1', 'CAF19199.1', 'CAF19200.1', 'CAF19201.1', 'CAF19202.1', 'CAF19203.1', 'CAF19204.1', 'CAF19205.1', 'CAF19206.1', 'CAF19207.1', 'CAF19208.1', 'CAF19209.1', 'CAF19210.1', 'CAF19211.1', 'CAF19212.1', 'CAF19213.1', 'CAF19214.1', 'CAF19215.1', 'CAF19216.1', 'CAF19217.1', 'CAF19218.1', 'CAF19219.1', 'CAF19220.1', 'CAF19221.1', 'CAF19222.1', 'CAF19223.1', 'CAF19224.1', 'CAF19225.1', 'CAF19226.1', 'CAF19227.1', 'CAF19228.1', 'CAF19229.1', 'CAF19230.1', 'CAF19231.1', 'CAF19232.1', 'CAF19233.1', 'CAF19234.1', 'CAF19235.1', 'CAF19236.1', 'CAF19237.1', 'CAF19238.1', 'CAF19239.1', 'CAF19240.1', 'CAF19241.1', 'CAF19242.1', 'CAF19243.1', 'CAF19244.1', 'CAF19245.1', 'CAF19246.1', 'CAF19247.1', 'CAF19248.1', 'CAF19249.1', 'CAF19250.1', 'CAF19251.1', 'CAF19252.1', 'CAF19253.1', 'CAF19254.1', 'CAF19255.1', 'CAF19256.1', 'CAF19257.1', 'CAF19258.1', 'CAF19259.1', 'CAF19260.1', 'CAF19261.1', 'CAF19262.1', 'CAF19263.1', 'CAF19264.1', 'CAF19265.1', 'CAF19266.1', 'CAF19267.1', 'CAF19268.1', 'CAF19269.1', 'CAF19270.1', 'CAF19271.1', 'CAF19272.1', 'CAF19273.1', 'CAF19274.1', 'CAF19275.1', 'CAF19276.1', 'CAF19277.1', 'CAF19278.1', 'CAF19279.1', 'CAF19280.1', 'CAF19281.1', 'CAF19282.1', 'CAF19283.1', 'CAF19284.1', 'CAF19285.1', 'CAF19286.1', 'CAF19287.1', 'CAF19288.1', 'CAF19289.1', 'CAF19290.1', 'CAF19291.1', 'CAF19292.1', 'CAF19293.1', 'CAF19294.1', 'CAF19295.1', 'CAF19296.1', 'CAF19297.1', 'CAF19298.1', 'CAF19299.1', 'CAF19300.1', 'CAF19301.1', 'CAF19302.1', 'CAF19303.1', 'CAF19304.1', 'CAF19305.1', 'CAF19306.1', 'CAF19307.1', 'CAF19308.1', 'CAF19309.1', 'CAF19310.1', 'CAF19311.1', 'CAF19312.1', 'CAF19313.1', 'CAF19314.1', 'CAF19315.1', 'CAF19316.1', 'CAF19317.1', 'CAF19318.1', 'CAF19319.1', 'CAF19320.1', 'CAF19321.1', 'CAF19322.1', 'CAF19323.1', 'CAF19324.1', 'CAF19325.1', 'CAF19326.1', 'CAF19327.1', 'CAF19328.1', 'CAF19329.1', 'CAF19330.1', 'CAF19331.1', 'CAF19332.1', 'CAF19333.1', 'CAF19334.1', 'CAF19335.1', 'CAF19336.1', 'CAF19337.1', 'CAF19338.1', 'CAF19339.1', 'CAF19340.1', 'CAF19341.1', 'CAF19342.1', 'CAF19343.1', 'CAF19344.1', 'CAF19345.1', 'CAF19346.1', 'CAF19347.1', 'CAF19348.1', 'CAF19349.1', 'CAF19350.1', 'CAF19351.1', 'CAF19352.1', 'CAF19353.1', 'CAF19354.1', 'CAF19355.1', 'CAF19356.1', 'CAF19357.1', 'CAF19358.1', 'CAF19359.1', 'CAF19360.1', 'CAF19361.1', 'CAF19362.1', 'CAF19363.1', 'CAF19364.1', 'CAF19365.1', 'CAF19366.1', 'CAF19367.1', 'CAF19368.1', 'CAF19369.1', 'CAF19370.1', 'CAF19371.1', 'CAF19372.1', 'CAF19373.1', 'CAF19374.1', 'CAF19375.1', 'CAF19376.1', 'CAF19377.1', 'CAF19378.1', 'CAF19379.1', 'CAF19380.1', 'CAF19381.1', 'CAF19382.1', 'CAF19383.1', 'CAF19384.1', 'CAF19385.1', 'CAF19386.1', 'CAF19387.1', 'CAF19388.1', 'CAF19389.1', 'CAF19390.1', 'CAF19391.1', 'CAF19392.1', 'CAF19393.1', 'CAF19394.1', 'CAF19395.1', 'CAF19396.1', 'CAF19397.1', 'CAF19398.1', 'CAF19399.1', 'CAF19400.1', 'CAF19401.1', 'CAF19402.1', 'CAF19403.1', 'CAF19404.1', 'CAF19405.1', 'CAF19406.1', 'CAF19407.1', 'CAF19408.1', 'CAF19409.1', 'CAF19410.1', 'CAF19411.1', 'CAF19412.1', 'CAF19413.1', 'CAF19414.1', 'CAF19415.1', 'CAF19416.1', 'CAF19417.1', 'CAF19418.1', 'CAF19419.1', 'CAF19420.1', 'CAF19421.1', 'CAF19422.1', 'CAF19423.1', 'CAF19424.1', 'CAF19425.1', 'CAF19426.1', 'CAF19427.1', 'CAF19428.1', 'CAF19429.1', 'CAF19430.1', 'CAF19431.1', 'CAF19432.1', 'CAF19433.1', 'CAF19434.1', 'CAF19435.1', 'CAF19436.1', 'CAF19437.1', 'CAF19438.1', 'CAF19439.1', 'CAF19440.1', 'CAF19441.1', 'CAF19442.1', 'CAF19443.1', 'CAF19444.1', 'CAF19445.1', 'CAF19446.1', 'CAF19447.1', 'CAF19448.1', 'CAF19449.1', 'CAF19450.1', 'CAF19451.1', 'CAF19452.1', 'CAF19453.1', 'CAF19454.1', 'CAF19455.1', 'CAF19456.1', 'CAF19457.1', 'CAF19458.1', 'CAF19459.1', 'CAF19460.1', 'CAF19461.1', 'CAF19462.1', 'CAF19463.1', 'CAF19464.1', 'CAF19465.1', 'CAF19466.1', 'CAF19467.1', 'CAF19468.1', 'CAF19469.1', 'CAF19470.1', 'CAF19471.1', 'CAF19472.1', 'CAF19473.1', 'CAF19474.1', 'CAF19475.1', 'CAF19476.1', 'CAF19477.1', 'CAF19478.1', 'CAF19479.1', 'CAF19480.1', 'CAF19481.1', 'CAF19482.1', 'CAF19483.1', 'CAF19484.1', 'CAF19485.1', 'CAF19486.1', 'CAF19487.1', 'CAF19488.1', 'CAF19489.1', 'CAF19490.1', 'CAF19491.1', 'CAF19492.1', 'CAF19493.1', 'CAF19494.1', 'CAF19495.1', 'CAF19496.1', 'CAF19497.1', 'CAF19498.1', 'CAF19499.1', 'CAF19500.1', 'CAF19501.1', 'CAF19502.1', 'CAF19503.1', 'CAF19504.1', 'CAF19505.1', 'CAF19506.1', 'CAF19507.1', 'CAF19508.1', 'CAF19509.1', 'CAF19510.1', 'CAF19511.1', 'CAF19512.1', 'CAF19513.1', 'CAF19514.1', 'CAF19515.1', 'CAF19516.1', 'CAF19517.1', 'CAF19518.1', 'CAF19519.1', 'CAF19520.1', 'CAF19521.1', 'CAF19522.1', 'CAF19523.1', 'CAF19524.1', 'CAF19525.1', 'CAF19526.1', 'CAF19527.1', 'CAF19528.1', 'CAF19529.1', 'CAF19530.1', 'CAF19531.1', 'CAF19532.1', 'CAF19533.1', 'CAF19534.1', 'CAF19535.1', 'CAF19536.1', 'CAF19537.1', 'CAF19538.1', 'CAF19539.1', 'CAF19540.1', 'CAF19541.1', 'CAF19542.1', 'CAF19543.1', 'CAF19544.1', 'CAF19545.1', 'CAF19546.1', 'CAF19547.1', 'CAF19548.1', 'CAF19549.1', 'CAF19550.1', 'CAF19551.1', 'CAF19552.1', 'CAF19553.1', 'CAF19554.1', 'CAF19555.1', 'CAF19556.1', 'CAF19557.1', 'CAF19558.1', 'CAF19559.1', 'CAF19560.1', 'CAF19561.1', 'CAF19562.1', 'CAF19563.1', 'CAF19564.1', 'CAF19565.1', 'CAF19566.1', 'CAF19567.1', 'CAF19568.1', 'CAF19569.1', 'CAF19570.1', 'CAF19571.1', 'CAF19572.1', 'CAF19573.1', 'CAF19574.1', 'CAF19575.1', 'CAF19576.1', 'CAF19577.1', 'CAF19578.1', 'CAF19579.1', 'CAF19580.1', 'CAF19581.1', 'CAF19582.1', 'CAF19583.1', 'CAF19584.1', 'CAF19585.1', 'CAF19586.1', 'CAF19587.1', 'CAF19588.1', 'CAF19589.1', 'CAF19590.1', 'CAF19591.1', 'CAF19592.1', 'CAF19593.1', 'CAF19594.1', 'CAF19595.1', 'CAF19596.1', 'CAF19597.1', 'CAF19598.1', 'CAF19599.1', 'CAF19600.1', 'CAF19601.1', 'CAF19602.1', 'CAF19603.1', 'CAF19604.1', 'CAF19605.1', 'CAF19606.1', 'CAF19607.1', 'CAF19608.1', 'CAF19609.1', 'CAF19612.1', 'CAF19613.1', 'CAF19614.1', 'CAF19615.1', 'CAF19616.1', 'CAF19617.1', 'CAF19618.1', 'CAF19619.1', 'CAF19620.1', 'CAF19621.1', 'CAF19622.1', 'CAF19623.1', 'CAF19624.1', 'CAF19625.1', 'CAF19626.1', 'CAF19627.1', 'CAF19628.1', 'CAF19629.1', 'CAF19630.1', 'CAF19631.1', 'CAF19632.1', 'CAF19633.1', 'CAF19634.1', 'CAF19635.1', 'CAF19636.1', 'CAF19637.1', 'CAF19638.1', 'CAF19639.1', 'CAF19640.1', 'CAF19641.1', 'CAF19642.1', 'CAF19643.1', 'CAF19644.1', 'CAF19645.1', 'CAF19646.1', 'CAF19647.1', 'CAF19648.1', 'CAF19649.1', 'CAF19650.1', 'CAF19651.1', 'CAF19652.1', 'CAF19653.1', 'CAF19654.1', 'CAF19655.1', 'CAF19656.1', 'CAF19657.1', 'CAF19658.1', 'CAF19659.1', 'CAF19660.1', 'CAF19661.1', 'CAF19662.1', 'CAF19663.1', 'CAF19664.1', 'CAF19665.1', 'CAF19666.1', 'CAF19667.1', 'CAF19668.1', 'CAF19669.1', 'CAF19670.1', 'CAF19671.1', 'CAF19672.1', 'CAF19673.1', 'CAF19674.1', 'CAF19675.1', 'CAF19676.1', 'CAF19677.1', 'CAF19678.1', 'CAF19679.1', 'CAF19680.1', 'CAF19681.1', 'CAF19682.1', 'CAF19683.1', 'CAF19684.1', 'CAF19685.1', 'CAF19686.1', 'CAF19687.1', 'CAF19688.1', 'CAF19689.1', 'CAF19690.1', 'CAF19691.1', 'CAF19692.1', 'CAF19693.1', 'CAF19694.1', 'CAF19695.1', 'CAF19696.1', 'CAF19697.1', 'CAF19698.1', 'CAF19699.1', 'CAF19700.1', 'CAF19701.1', 'CAF19702.1', 'CAF19703.1', 'CAF19704.1', 'CAF19705.1', 'CAF19706.1', 'CAF19707.1', 'CAF19708.1', 'CAF19709.1', 'CAF19710.1', 'CAF19711.1', 'CAF19712.1', 'CAF19713.1', 'CAF19714.1', 'CAF19715.1', 'CAF19716.1', 'CAF19717.1', 'CAF19718.1', 'CAF19719.1', 'CAF19720.1', 'CAF19721.1', 'CAF19722.1', 'CAF19723.1', 'CAF19725.1', 'CAF19724.1', 'CAF19726.1', 'CAF19727.1', 'CAF19728.1', 'CAF19729.1', 'CAF19730.1', 'CAF19731.1', 'CAF19732.1', 'CAF19733.1', 'CAF19734.1', 'CAF19735.1', 'CAF19736.1', 'CAF19737.1', 'CAF19738.1', 'CAF19739.1', 'CAF19740.1', 'CAF19741.1', 'CAF19742.1', 'CAF19743.1', 'CAF19744.1', 'CAF19745.1', 'CAF19746.1', 'CAF19747.1', 'CAF19748.1', 'CAF19749.1', 'CAF19750.1', 'CAF19751.1', 'CAF19752.1', 'CAF19753.1', 'CAF19754.1', 'CAF19755.1', 'CAF19756.1', 'CAF19757.1', 'CAF19758.1', 'CAF19759.1', 'CAF19760.1', 'CAF19761.1', 'CAF19762.1', 'CAF19763.1', 'CAF19764.1', 'CAF19765.1', 'CAF19766.1', 'CAF19767.1', 'CAF19768.1', 'CAF19769.1', 'CAF19770.1', 'CAF19771.1', 'CAF19772.1', 'CAF19773.1', 'CAF19774.1', 'CAF19775.1', 'CAF19776.1', 'CAF19777.1', 'CAF19778.1', 'CAF19779.1', 'CAF19780.1', 'CAF19781.1', 'CAF19782.1', 'CAF19783.1', 'CAF19784.1', 'CAF19785.1', 'CAF19786.1', 'CAF19787.1', 'CAF19788.1', 'CAF19789.1', 'CAF19790.1', 'CAF19791.1', 'CAF19792.1', 'CAF19793.1', 'CAF19794.1', 'CAF19795.1', 'CAF19796.1', 'CAF19797.1', 'CAF19798.1', 'CAF19799.1', 'CAF19800.1', 'CAF19801.1', 'CAF19802.1', 'CAF19803.1', 'CAF19804.1', 'CAF19805.1', 'CAF19806.1', 'CAF19807.1', 'CAF19808.1', 'CAF19809.1', 'CAF19810.1', 'CAF19811.1', 'CAF19812.1', 'CAF19813.1', 'CAF19814.1', 'CAF19815.1', 'CAF19816.1', 'CAF19817.1', 'CAF19818.1', 'CAF19819.1', 'CAF19820.1', 'CAF19821.1', 'CAF19822.1', 'CAF19823.1', 'CAF19824.1', 'CAF19825.1', 'CAF19826.1', 'CAF19827.1', 'CAF19828.1', 'CAF19829.1', 'CAF19830.1', 'CAF19831.1', 'CAF19832.1', 'CAF19833.1', 'CAF19834.1', 'CAF19835.1', 'CAF19836.1', 'CAF19837.1', 'CAF19838.1', 'CAF19839.1', 'CAF19840.1', 'CAF19841.1', 'CAF19842.1', 'CAF19843.1', 'CAF19844.1', 'CAF19845.1', 'CAF19846.1', 'CAF19847.1', 'CAF19848.1', 'CAF19849.1', 'CAF19850.1', 'CAF19851.1', 'CAF19852.1', 'CAF19853.1', 'CAF19854.1', 'CAF19855.1', 'CAF19856.1', 'CAF19857.1', 'CAF19858.1', 'CAF19859.1', 'CAF19860.1', 'CAF19861.1', 'CAF19862.1', 'CAF19863.1', 'CAF19864.1', 'CAF19865.1', 'CAF19866.1', 'CAF19867.1', 'CAF19868.1', 'CAF19869.1', 'CAF19870.1', 'CAF19871.1', 'CAF19872.1', 'CAF19873.1', 'CAF19874.1', 'CAF19875.1', 'CAF19876.1', 'CAF19877.1', 'CAF19878.1', 'CAF19879.1', 'CAF19880.1', 'CAF19881.1', 'CAF19882.1', 'CAF19883.1', 'CAF19884.1', 'CAF19885.1', 'CAF19886.1', 'CAF19887.1', 'CAF19888.1', 'CAF19889.1', 'CAF19890.1', 'CAF19891.1', 'CAF19892.1', 'CAF19893.1', 'CAF19894.1', 'CAF19895.1', 'CAF19896.1', 'CAF19897.1', 'CAF19898.1', 'CAF19899.1', 'CAF19900.1', 'CAF19901.1', 'CAF19902.1', 'CAF19903.1', 'CAF19904.1', 'CAF19905.1', 'CAF19906.1', 'CAF19907.1', 'CAF19908.1', 'CAF19909.1', 'CAF19910.1', 'CAF19911.1', 'CAF19912.1', 'CAF19913.1', 'CAF19914.1', 'CAF19915.1', 'CAF19916.1', 'CAF19917.1', 'CAF19918.1', 'CAF19919.1', 'CAF19920.1', 'CAF19921.1', 'CAF19922.1', 'CAF19923.1', 'CAF19924.1', 'CAF19925.1', 'CAF19926.1', 'CAF19927.1', 'CAF19928.1', 'CAF19929.1', 'CAF19930.1', 'CAF19931.1', 'CAF19932.1', 'CAF19933.1', 'CAF19934.1', 'CAF19935.1', 'CAF19936.1', 'CAF19937.1', 'CAF19938.1', 'CAF19939.1', 'CAF19940.1', 'CAF19941.1', 'CAF19942.1', 'CAF19943.1', 'CAF19944.1', 'CAF19945.1', 'CAF19946.1', 'CAF19947.1', 'CAF19948.1', 'CAF19949.1', 'CAF19950.1', 'CAF19951.1', 'CAF19952.1', 'CAF19953.1', 'CAF19954.1', 'CAF19955.1', 'CAF19956.1', 'CAF19957.1', 'CAF19958.1', 'CAF19959.1', 'CAF19960.1', 'CAF19961.1', 'CAF19962.1', 'CAF19963.1', 'CAF19964.1', 'CAF19965.1', 'CAF19966.1', 'CAF19967.1', 'CAF19968.1', 'CAF19969.1', 'CAF19970.1', 'CAF19971.1', 'CAF19972.1', 'CAF19973.1', 'CAF19974.1', 'CAF19975.1', 'CAF19976.1', 'CAF19977.1', 'CAF19978.1', 'CAF19979.1', 'CAF19980.1', 'CAF19981.1', 'CAF19982.1', 'CAF19983.1', 'CAF19984.1', 'CAF19985.1', 'CAF19986.1', 'CAF19987.1', 'CAF19988.1', 'CAF19989.1', 'CAF19990.1', 'CAF19991.1', 'CAF19992.1', 'CAF19993.1', 'CAF19994.1', 'CAF19995.1', 'CAF19996.1', 'CAF19997.1', 'CAF19998.1', 'CAF19999.1', 'CAF20000.1', 'CAF20001.1', 'CAF20002.1', 'CAF20003.1', 'CAF20004.1', 'CAF20005.1', 'CAF20006.1', 'CAF20007.1', 'CAF20008.1', 'CAF20009.1', 'CAF20010.1', 'CAF20011.1', 'CAF20012.1', 'CAF20013.1', 'CAF20014.1', 'CAF20015.1', 'CAF20016.1', 'CAF20017.1', 'CAF20018.1', 'CAF20019.1', 'CAF20020.1', 'CAF20021.1', 'CAF20022.1', 'CAF20023.1', 'CAF20024.1', 'CAF20025.1', 'CAF20026.1', 'CAF20027.1', 'CAF21338.1', 'CAF21339.1', 'CAF21340.1', 'CAF21341.1', 'CAF21342.1', 'CAF21343.1', 'CAF21344.1', 'CAF21345.1', 'CAF21346.1', 'CAF21347.1', 'CAF21348.1', 'CAF21349.1', 'CAF21350.1', 'CAF21351.1', 'CAF21352.1', 'CAF21353.1', 'CAF21354.1', 'CAF21355.1', 'CAF21356.1', 'CAF21357.1', 'CAF21358.1', 'CAF21359.1', 'CAF21360.1', 'CAF21361.1', 'CAF21362.1', 'CAF21363.1', 'CAF21364.1', 'CAF21365.1', 'CAF21366.1', 'CAF21367.1', 'CAF21368.1', 'CAF21369.1', 'CAF21370.1', 'CAF21371.1', 'CAF21372.1', 'CAF21373.1', 'CAF21374.1', 'CAF21375.1', 'CAF21376.1', 'CAF21377.1', 'CAF21378.1', 'CAF21379.1', 'CAF21380.1', 'CAF21381.1', 'CAF21382.1', 'CAF21383.1', 'CAF21384.1', 'CAF21385.1', 'CAF21386.1', 'CAF21387.1', 'CAF21388.1', 'CAF21389.1', 'CAF21390.1', 'CAF21391.1', 'CAF21392.1', 'CAF21393.1', 'CAF21394.1', 'CAF21395.1', 'CAF21396.1', 'CAF21397.1', 'CAF21398.1', 'CAF21399.1', 'CAF21400.1', 'CAF21401.1', 'CAF21402.1', 'CAF21403.1', 'CAF21404.1', 'CAF21405.1', 'CAF21406.1', 'CAF21407.1', 'CAF21408.1', 'CAF21409.1', 'CAF21410.1', 'CAF21411.1', 'CAF21412.1', 'CAF21413.1', 'CAF21414.1', 'CAF21415.1', 'CAF21416.1', 'CAF21417.1', 'CAF21418.1', 'CAF21419.1', 'CAF21420.1', 'CAF21421.1', 'CAF21422.1', 'CAF21423.1', 'CAF21424.1', 'CAF21425.1', 'CAF21426.1', 'CAF21427.1', 'CAF21428.1', 'CAF21429.1', 'CAF21430.1', 'CAF21431.1', 'CAF21432.1', 'CAF21433.1', 'CAF21434.1', 'CAF21435.1', 'CAF21436.1', 'CAF21437.1', 'CAF21438.1', 'CAF21439.1', 'CAF21440.1', 'CAF21441.1', 'CAF21442.1', 'CAF21443.1', 'CAF21444.1', 'CAF21445.1', 'CAF21446.1', 'CAF21447.1', 'CAF21448.1', 'CAF21449.1', 'CAF21450.1', 'CAF21451.1', 'CAF21452.1', 'CAF21453.1', 'CAF21454.1', 'CAF21455.1', 'CAF21456.1', 'CAF21457.1', 'CAF21458.1', 'CAF21459.1', 'CAF21460.1', 'CAF21461.1', 'CAF21462.1', 'CAF21463.1', 'CAF21464.1', 'CAF21465.1', 'CAF21466.1', 'CAF21467.1', 'CAF21468.1', 'CAF21469.1', 'CAF21470.1', 'CAF21471.1', 'CAF21472.1', 'CAF21473.1', 'CAF21474.1', 'CAF21475.1', 'CAF21476.1', 'CAF21477.1', 'CAF21478.1', 'CAF21479.1', 'CAF21480.1', 'CAF21481.1', 'CAF21482.1', 'CAF21483.1', 'CAF21484.1', 'CAF21485.1', 'CAF21486.1', 'CAF21487.1', 'CAF21488.1', 'CAF21489.1', 'CAF21490.1', 'CAF21491.1', 'CAF21492.1', 'CAF21493.1', 'CAF21494.1', 'CAF21495.1', 'CAF21496.1', 'CAF21497.1', 'CAF21498.1', 'CAF21499.1', 'CAF21500.1', 'CAF21501.1', 'CAF21502.1', 'CAF21503.1', 'CAF21504.1', 'CAF21505.1', 'CAF21506.1', 'CAF21507.1', 'CAF21508.1', 'CAF21509.1', 'CAF21510.1', 'CAF21511.1', 'CAF21512.1', 'CAF21513.1', 'CAF21514.1', 'CAF21515.1', 'CAF21516.1', 'CAF21517.1', 'CAF21518.2', 'CAF21519.1', 'CAF21520.1', 'CAF21521.1', 'CAF21522.1', 'CAF21523.1', 'CAF21524.1', 'CAF21525.1', 'CAF21526.1', 'CAF21527.1', 'CAF21528.1', 'CAF21529.1', 'CAF21530.1', 'CAF21531.1', 'CAF21532.1', 'CAF21533.1', 'CAF21534.1', 'CAF21535.1', 'CAF21536.1', 'CAF21537.1', 'CAF21538.1', 'CAF21539.1', 'CAF21540.1', 'CAF21541.1', 'CAF21542.1', 'CAF21543.1', 'CAF21544.1', 'CAF21545.1', 'CAF21546.1', 'CAF21547.1', 'CAF21548.1', 'CAF21549.1', 'CAF21550.1', 'CAF21551.1', 'CAF21552.1', 'CAF21553.1', 'CAF21554.1', 'CAF21555.1', 'CAF21556.1', 'CAF21557.1', 'CAF21558.1', 'CAF21559.1', 'CAF21560.1', 'CAF21562.1', 'CAF21561.1', 'CAF21563.1', 'CAF21564.1', 'CAF21565.1', 'CAF21567.1', 'CAF21568.1', 'CAF21569.1', 'CAF21570.1', 'CAF21571.1', 'CAF21572.1', 'CAF21573.1', 'CAF21574.1', 'CAF21575.1', 'CAF21576.1', 'CAF21577.1', 'CAF21578.1', 'CAF21579.1', 'CAF21580.1', 'CAF21581.1', 'CAF21582.1', 'CAF21583.1', 'CAF21584.1', 'CAF21585.1', 'CAF21586.1', 'CAF21587.1', 'CAF21588.1', 'CAF21589.1', 'CAF21590.1', 'CAF21591.1', 'CAF21592.1', 'CAF21593.1', 'CAF21594.1', 'CAF21595.1', 'CAF21596.1', 'CAF21597.1', 'CAF21598.1', 'CAF21599.1', 'CAF21600.1', 'CAF21601.1', 'CAF21602.1', 'CAF21603.1', 'CAF21604.1', 'CAF21605.1', 'CAF21606.1', 'CAF21607.1', 'CAF21608.1', 'CAF21609.1', 'CAF21610.1', 'CAF21611.1', 'CAF21612.1', 'CAF21613.1', 'CAF21614.1', 'CAF21615.1', 'CAF21616.1', 'CAF21617.1', 'CAF21618.1', 'CAF21619.1', 'CAF21620.1', 'CAF21621.1', 'CAF21622.1', 'CAF21623.1', 'CAF21624.1', 'CAF21625.1', 'CAF21626.1', 'CAF21627.1', 'CAF21628.1', 'CAF21629.1', 'CAF21630.1', 'CAF21631.1', 'CAF21632.1', 'CAF21633.1', 'CAF21634.1', 'CAF21635.1', 'CAF21636.1', 'CAF21637.1', 'CAF21638.1', 'CAF21639.1', 'CAF21640.1', 'CAF21641.1', 'CAF21642.1', 'CAF21643.1', 'CAF21644.1', 'CAF21645.1', 'CAF21646.1', 'CAF21647.1', 'CAF21648.1', 'CAF21649.1', 'CAF21650.1', 'CAF21651.1', 'CAF21652.1', 'CAF20028.1', 'CAF20029.1', 'CAF20030.1', 'CAF20031.1', 'CAF20032.1', 'CAF20033.1', 'CAF20034.1', 'CAF20035.1', 'CAF20036.1', 'CAF20037.1', 'CAF20038.1', 'CAF20039.1', 'CAF20040.1', 'CAF20041.1', 'CAF20042.1', 'CAF20043.1', 'CAF20044.1', 'CAF20045.1', 'CAF20046.1', 'CAF20047.1', 'CAF20048.1', 'CAF20049.1', 'CAF20050.1', 'CAF20051.1', 'CAF20052.1', 'CAF20053.1', 'CAF20054.1', 'CAF20055.1', 'CAF20056.1', 'CAF20057.1', 'CAF20058.1', 'CAF20059.1', 'CAF20060.1', 'CAF20061.1', 'CAF20062.1', 'CAF20063.1', 'CAF20064.1', 'CAF20065.1', 'CAF20066.1', 'CAF20067.1', 'CAF20068.1', 'CAF20069.1', 'CAF20070.1', 'CAF20071.1', 'CAF20072.1', 'CAF20073.1', 'CAF20074.1', 'CAF20075.1', 'CAF20076.1', 'CAF20077.1', 'CAF20078.1', 'CAF20079.1', 'CAF20080.1', 'CAF20081.1', 'CAF20082.1', 'CAF20083.1', 'CAF20084.1', 'CAF20085.1', 'CAF20086.1', 'CAF20087.1', 'CAF20088.1', 'CAF20089.1', 'CAF20090.1', 'CAF20091.1', 'CAF20092.1', 'CAF20093.1', 'CAF20094.1', 'CAF20095.1', 'CAF20096.1', 'CAF20097.1', 'CAF20098.1', 'CAF20099.1', 'CAF20100.1', 'CAF20101.1', 'CAF20102.1', 'CAF20103.1', 'CAF20104.1', 'CAF20105.1', 'CAF20106.1', 'CAF20107.1', 'CAF20108.1', 'CAF20109.1', 'CAF20110.1', 'CAF20111.1', 'CAF20112.1', 'CAF20113.1', 'CAF20114.1', 'CAF20115.1', 'CAF20116.1', 'CAF20117.1', 'CAF20118.1', 'CAF20119.1', 'CAF20120.1', 'CAF20121.1', 'CAF20122.1', 'CAF20123.1', 'CAF20124.1', 'CAF20125.1', 'CAF20126.1', 'CAF20127.1', 'CAF20128.1', 'CAF20129.1', 'CAF20130.1', 'CAF20131.1', 'CAF20132.1', 'CAF20133.1', 'CAF20134.1', 'CAF20135.1', 'CAF20136.1', 'CAF20137.1', 'CAF20138.1', 'CAF20139.1', 'CAF20140.1', 'CAF20141.1', 'CAF20142.1', 'CAF20143.1', 'CAF20144.1', 'CAF20145.1', 'CAF20146.1', 'CAF20147.1', 'CAF20148.1', 'CAF20149.1', 'CAF20150.1', 'CAF20151.1', 'CAF20152.1', 'CAF20153.1', 'CAF20154.1', 'CAF20155.1', 'CAF20156.1', 'CAF20157.1', 'CAF20158.1', 'CAF20159.1', 'CAF20160.1', 'CAF20161.1', 'CAF20162.1', 'CAF20163.1', 'CAF20164.1', 'CAF20165.1', 'CAF20166.1', 'CAF20167.1', 'CAF20168.1', 'CAF20169.1', 'CAF20170.1', 'CAF20171.1', 'CAF20172.1', 'CAF20173.1', 'CAF20174.1', 'CAF20175.1', 'CAF20176.1', 'CAF20177.1', 'CAF20178.1', 'CAF20179.1', 'CAF20180.1', 'CAF20181.1', 'CAF20182.1', 'CAF20183.1', 'CAF20184.1', 'CAF20185.1', 'CAF20186.1', 'CAF20187.1', 'CAF20188.1', 'CAF20189.1', 'CAF20190.1', 'CAF20191.1', 'CAF20192.1', 'CAF20193.1', 'CAF20194.1', 'CAF20195.1', 'CAF20196.1', 'CAF20197.1', 'CAF20198.1', 'CAF20199.1', 'CAF20200.1', 'CAF20201.1', 'CAF20202.1', 'CAF20203.1', 'CAF20204.1', 'CAF20205.1', 'CAF20206.1', 'CAF20207.1', 'CAF20208.1', 'CAF20209.1', 'CAF20210.1', 'CAF20211.1', 'CAF20212.1', 'CAF20213.1', 'CAF20214.1', 'CAF20215.1', 'CAF20216.1', 'CAF20217.1', 'CAF20218.1', 'CAF20219.1', 'CAF20220.1', 'CAF20221.1', 'CAF20222.1', 'CAF20223.1', 'CAF20224.1', 'CAF20225.1', 'CAF20226.1', 'CAF20227.1', 'CAF20228.1', 'CAF20229.1', 'CAF20230.1', 'CAF20231.1', 'CAF20232.1', 'CAF20233.1', 'CAF20234.1', 'CAF20235.1', 'CAF20236.1', 'CAF20237.1', 'CAF20238.1', 'CAF20239.1', 'CAF20240.1', 'CAF20241.1', 'CAF20242.1', 'CAF20243.1', 'CAF20244.1', 'CAF20245.1', 'CAF20246.1', 'CAF20247.1', 'CAF20248.1', 'CAF20249.1', 'CAF20250.1', 'CAF20251.1', 'CAF20252.1', 'CAF20253.1', 'CAF20254.1', 'CAF20255.1', 'CAF20256.1', 'CAF20257.1', 'CAF20258.1', 'CAF20259.1', 'CAF20260.1', 'CAF20261.1', 'CAF20262.1', 'CAF20263.1', 'CAF20264.1', 'CAF20265.1', 'CAF20266.1', 'CAF20267.1', 'CAF20268.1', 'CAF20269.1', 'CAF20270.1', 'CAF20271.1', 'CAF20272.1', 'CAF20273.1', 'CAF20274.1', 'CAF20275.1', 'CAF20276.1', 'CAF20277.1', 'CAF20278.1', 'CAF20279.1', 'CAF20280.1', 'CAF20281.1', 'CAF20282.1', 'CAF20283.1', 'CAF20284.1', 'CAF20285.1', 'CAF20286.1', 'CAF20287.1', 'CAF20288.1', 'CAF20289.1', 'CAF20290.1', 'CAF20291.1', 'CAF20292.1', 'CAF20293.1', 'CAF20294.1', 'CAF20295.1', 'CAF20296.1', 'CAF20297.1', 'CAF20298.1', 'CAF20299.1', 'CAF20300.1', 'CAF20301.1', 'CAF20302.1', 'CAF20303.1', 'CAF20304.1', 'CAF20305.1', 'CAF20306.1', 'CAF20307.1', 'CAF20308.1', 'CAF20309.1', 'CAF20310.1', 'CAF20311.1', 'CAF20312.1', 'CAF20313.1', 'CAF20314.1', 'CAF20315.1', 'CAF20316.1', 'CAF20317.1', 'CAF20318.1', 'CAF20319.1', 'CAF20320.1', 'CAF20321.1', 'CAF20322.1', 'CAF20323.1', 'CAF20324.1', 'CAF20325.1', 'CAF20326.1', 'CAF20327.1', 'CAF20328.1', 'CAF20329.1', 'CAF20330.1', 'CAF20331.1', 'CAF20332.1', 'CAF20333.1', 'CAF20334.1', 'CAF20335.1', 'CAF20336.1', 'CAF20337.1', 'CAF20338.1', 'CAF20339.1', 'CAF20340.1', 'CAF20341.1', 'CAF20342.1', 'CAF20343.1', 'CAF20344.1', 'CAF20345.1', 'CAF20346.1', 'CAF20347.1', 'CAF20348.1', 'CAF20349.1', 'CAF20350.1', 'CAF20351.1', 'CAF20352.1', 'CAF20353.1', 'CAF20354.1', 'CAF20355.1', 'CAF20356.1', 'CAF20357.1', 'CAF20358.1', 'CAF20359.1', 'CAF20360.1', 'CAF20361.1', 'CAF20362.1', 'CAF20363.1', 'CAF20364.1', 'CAF20365.1', 'CAF20366.1', 'CAF20367.1', 'CAF20368.1', 'CAF20369.1', 'CAF20370.1', 'CAF20371.1', 'CAF20372.1', 'CAF20373.1', 'CAF20374.1', 'CAF20375.1', 'CAF20376.1', 'CAF20377.1', 'CAF20378.1', 'CAF20379.1', 'CAF20380.1', 'CAF20381.1', 'CAF20382.1', 'CAF20383.1', 'CAF20384.1', 'CAF20385.1', 'CAF20386.1', 'CAF20387.1', 'CAF20388.1', 'CAF20389.1', 'CAF20390.1', 'CAF20391.1', 'CAF20392.1', 'CAF20393.1', 'CAF20394.1', 'CAF20395.1', 'CAF20396.1', 'CAF20397.1', 'CAF20398.1', 'CAF20399.1', 'CAF20400.1', 'CAF20401.1', 'CAF20402.1', 'CAF20403.1', 'CAF20404.1', 'CAF20405.1', 'CAF20406.1', 'CAF20407.1', 'CAF20408.1', 'CAF20409.1', 'CAF20410.1', 'CAF20411.1', 'CAF20412.1', 'CAF20413.1', 'CAF20414.1', 'CAF20415.1', 'CAF20416.1', 'CAF20417.1', 'CAF20418.1', 'CAF20419.1', 'CAF20420.1', 'CAF20421.1', 'CAF20422.1', 'CAF20423.1', 'CAF20424.1', 'CAF20425.1', 'CAF20426.1', 'CAF20427.1', 'CAF20428.1', 'CAF20429.1', 'CAF20430.1', 'CAF20431.1', 'CAF20432.1', 'CAF20433.1', 'CAF20434.1', 'CAF20435.1', 'CAF20436.1', 'CAF20437.1', 'CAF20438.1', 'CAF20439.1', 'CAF20440.1', 'CAF20441.1', 'CAF20442.1', 'CAF20443.1', 'CAF20444.1', 'CAF20445.1', 'CAF20446.1', 'CAF20447.1', 'CAF20448.1', 'CAF20449.1', 'CAF20450.1', 'CAF20451.1', 'CAF20452.1', 'CAF20453.1', 'CAF20454.1', 'CAF20455.1', 'CAF20456.1', 'CAF20457.1', 'CAF20458.1', 'CAF20459.1', 'CAF20460.1', 'CAF20461.1', 'CAF20462.1', 'CAF20463.1', 'CAF20464.1', 'CAF20465.1', 'CAF20466.1', 'CAF20467.1', 'CAF20468.1', 'CAF20469.1', 'CAF20470.1', 'CAF20471.1', 'CAF20472.1', 'CAF20473.1', 'CAF20474.1', 'CAF20475.1', 'CAF20476.1', 'CAF20477.1', 'CAF20478.1', 'CAF20479.1', 'CAF20480.1', 'CAF20481.1', 'CAF20482.1', 'CAF20484.1', 'CAF20485.1', 'CAF20486.1', 'CAF20487.1', 'CAF20488.1', 'CAF20489.1', 'CAF20490.1', 'CAF20491.1', 'CAF20492.1', 'CAF20493.1', 'CAF20494.1', 'CAF20495.1', 'CAF20496.1', 'CAF20497.1', 'CAF20498.1', 'CAF20499.1', 'CAF20500.1', 'CAF20501.1', 'CAF20502.1', 'CAF20503.1', 'CAF20504.1', 'CAF20505.1', 'CAF20506.1', 'CAF20507.1', 'CAF20508.1', 'CAF20509.1', 'CAF20510.1', 'CAF20511.1', 'CAF20512.1', 'CAF20513.1', 'CAF20514.1', 'CAF20515.1', 'CAF20516.1', 'CAF20517.1', 'CAF20518.1', 'CAF20519.1', 'CAF20520.1', 'CAF20521.1', 'CAF20522.1', 'CAF20523.1', 'CAF20524.1', 'CAF20525.1', 'CAF20526.1', 'CAF20527.1', 'CAF20528.1', 'CAF20529.1', 'CAF20530.1', 'CAF20531.1', 'CAF20532.1', 'CAF20533.1', 'CAF20534.1', 'CAF20535.1', 'CAF20536.1', 'CAF20537.1', 'CAF20538.1', 'CAF20539.1', 'CAF20540.1', 'CAF20541.1', 'CAF20542.1', 'CAF20543.1', 'CAF20544.1', 'CAF20545.1', 'CAF20546.1', 'CAF20547.1', 'CAF20548.1', 'CAF20549.1', 'CAF20550.1', 'CAF20551.1', 'CAF20553.1', 'CAF20554.1', 'CAF20555.1', 'CAF20556.1', 'CAF20557.1', 'CAF20558.1', 'CAF20559.1', 'CAF20560.1', 'CAF20562.1', 'CAF20561.1', 'CAF20563.1', 'CAF20564.1', 'CAF20565.1', 'CAF20566.1', 'CAF20567.1', 'CAF20568.1', 'CAF20569.1', 'CAF20570.1', 'CAF20571.1', 'CAF20572.1', 'CAF20574.1', 'CAF20573.1', 'CAF20575.1', 'CAF20576.1', 'CAF20577.1', 'CAF20578.1', 'CAF20579.1', 'CAF20580.1', 'CAF20581.1', 'CAF20582.1', 'CAF20583.1', 'CAF20584.1', 'CAF20585.1', 'CAF20586.1', 'CAF20587.1', 'CAF20588.1', 'CAF20589.1', 'CAF20590.1', 'CAF20591.1', 'CAF20592.1', 'CAF20593.1', 'CAF20594.1', 'CAF20595.1', 'CAF20596.1', 'CAF20597.1', 'CAF20598.1', 'CAF20599.1', 'CAF20600.1', 'CAF20601.1', 'CAF20602.1', 'CAF20603.1', 'CAF20604.1', 'CAF20605.1', 'CAF20606.1', 'CAF20607.1', 'CAF20608.1', 'CAF20609.1', 'CAF20610.1', 'CAF20611.1', 'CAF20612.1', 'CAF20613.1', 'CAF20614.1', 'CAF20615.1', 'CAF20616.1', 'CAF20617.1', 'CAF20618.1', 'CAF20619.1', 'CAF20620.1', 'CAF20621.1', 'CAF20622.1', 'CAF20623.1', 'CAF20624.1', 'CAF20625.1', 'CAF20626.1', 'CAF20627.1', 'CAF20628.1', 'CAF20629.1', 'CAF20630.1', 'CAF20631.1', 'CAF20632.1', 'CAF20633.1', 'CAF20634.1', 'CAF20635.1', 'CAF20636.1', 'CAF20637.1', 'CAF20638.1', 'CAF20639.1', 'CAF20640.1', 'CAF20641.1', 'CAF20642.1', 'CAF20643.1', 'CAF20644.1', 'CAF20645.1', 'CAF20646.1', 'CAF20647.1', 'CAF20648.1', 'CAF20649.1', 'CAF20650.1', 'CAF20651.1', 'CAF20652.1', 'CAF20653.1', 'CAF20654.1', 'CAF20655.1', 'CAF20656.1', 'CAF20657.1', 'CAF20658.1', 'CAF20659.1', 'CAF20660.1', 'CAF20662.1', 'CAF20661.1', 'CAF20663.1', 'CAF20664.1', 'CAF20665.1', 'CAF20666.1', 'CAF20667.1', 'CAF20668.1', 'CAF20669.1', 'CAF20670.1', 'CAF20671.1', 'CAF20672.1', 'CAF20673.1', 'CAF20674.1', 'CAF20675.1', 'CAF20997.1', 'CAF20998.1', 'CAF20999.1', 'CAF21000.1', 'CAF21001.1', 'CAF21002.1', 'CAF21003.1', 'CAF21004.1', 'CAF21005.1', 'CAF21006.1', 'CAF21007.1', 'CAF21008.1', 'CAF21009.1', 'CAF21010.1', 'CAF21011.1', 'CAF21012.1', 'CAF21013.1', 'CAF21014.1', 'CAF21015.1', 'CAF21016.1', 'CAF21017.1', 'CAF21018.1', 'CAF21019.1', 'CAF21020.1', 'CAF21021.1', 'CAF21022.1', 'CAF21023.1', 'CAF21024.1', 'CAF21025.1', 'CAF21026.1', 'CAF21027.1', 'CAF21028.1', 'CAF21029.1', 'CAF21030.1', 'CAF21031.1', 'CAF21032.1', 'CAF21033.1', 'CAF21034.1', 'CAF21035.1', 'CAF21036.1', 'CAF21037.1', 'CAF21038.1', 'CAF21039.1', 'CAF21040.1', 'CAF21041.1', 'CAF21042.1', 'CAF21043.1', 'CAF21044.1', 'CAF21045.1', 'CAF21046.1', 'CAF21047.1', 'CAF21048.1', 'CAF21049.1', 'CAF21050.1', 'CAF21051.1', 'CAF21052.1', 'CAF21053.1', 'CAF21054.1', 'CAF21055.1', 'CAF21056.1', 'CAF21057.1', 'CAF21058.1', 'CAF21059.1', 'CAF21060.1', 'CAF21061.1', 'CAF21062.1', 'CAF21063.1', 'CAF21064.1', 'CAF21065.1', 'CAF21066.1', 'CAF21067.1', 'CAF21068.1', 'CAF21069.1', 'CAF21070.1', 'CAF21071.1', 'CAF21072.1', 'CAF21073.1', 'CAF21074.1', 'CAF21075.1', 'CAF21076.1', 'CAF21077.1', 'CAF21078.1', 'CAF21079.1', 'CAF21080.1', 'CAF21081.1', 'CAF21082.1', 'CAF21083.1', 'CAF21084.1', 'CAF21085.1', 'CAF21086.1', 'CAF21087.1', 'CAF21088.1', 'CAF21089.1', 'CAF21090.1', 'CAF21091.1', 'CAF21092.1', 'CAF21093.1', 'CAF21094.1', 'CAF21095.1', 'CAF21096.1', 'CAF21097.1', 'CAF21098.1', 'CAF21099.1', 'CAF21100.1', 'CAF21101.1', 'CAF21102.1', 'CAF21103.1', 'CAF21104.1', 'CAF21105.1', 'CAF21106.1', 'CAF21107.1', 'CAF21108.1', 'CAF21109.1', 'CAF21110.1', 'CAF21111.1', 'CAF21112.1', 'CAF21113.1', 'CAF21114.1', 'CAF21115.1', 'CAF21116.1', 'CAF21117.1', 'CAF21118.1', 'CAF21119.1', 'CAF21120.1', 'CAF21121.1', 'CAF21122.1', 'CAF21123.1', 'CAF21124.1', 'CAF21125.1', 'CAF22220.1', 'CAF21128.1', 'CAF21129.1', 'CAF21130.1', 'CAF21131.1', 'CAF21132.1', 'CAF21133.1', 'CAF21134.1', 'CAF21135.1', 'CAF21136.1', 'CAF21137.1', 'CAF21138.1', 'CAF21139.1', 'CAF21141.1', 'CAF21142.1', 'CAF21143.1', 'CAF21144.1', 'CAF21145.1', 'CAF21146.1', 'CAF21147.1', 'CAF21148.1', 'CAF21149.1', 'CAF21150.1', 'CAF21151.1', 'CAF21152.1', 'CAF21153.1', 'CAF21154.1', 'CAF21155.1', 'CAF21156.1', 'CAF21157.1', 'CAF21158.1', 'CAF21159.1', 'CAF21160.1', 'CAF21161.1', 'CAF21162.1', 'CAF21163.1', 'CAF21164.1', 'CAF21165.1', 'CAF21166.1', 'CAF21167.1', 'CAF21168.1', 'CAF21169.1', 'CAF21170.1', 'CAF21171.1', 'CAF21172.1', 'CAF21173.1', 'CAF21174.1', 'CAF21175.1', 'CAF21176.1', 'CAF21177.1', 'CAF21178.1', 'CAF21179.1', 'CAF21180.1', 'CAF21181.1', 'CAF21182.1', 'CAF21183.1', 'CAF21184.1', 'CAF21185.1', 'CAF21186.1', 'CAF21187.1', 'CAF21188.1', 'CAF21189.1', 'CAF21190.1', 'CAF21191.1', 'CAF21192.1', 'CAF21193.1', 'CAF21194.1', 'CAF21195.1', 'CAF21196.1', 'CAF21197.1', 'CAF21198.1', 'CAF21199.1', 'CAF21200.1', 'CAF21201.1', 'CAF21202.1', 'CAF21203.1', 'CAF21204.1', 'CAF21205.1', 'CAF21206.1', 'CAF21207.1', 'CAF21208.1', 'CAF21209.1', 'CAF21210.1', 'CAF21212.1', 'CAF21213.1', 'CAF21214.1', 'CAF21215.1', 'CAF21216.1', 'CAF21217.1', 'CAF21218.1', 'CAF21219.1', 'CAF21220.1', 'CAF21221.1', 'CAF21222.1', 'CAF21223.1', 'CAF21224.1', 'CAF21225.1', 'CAF21226.1', 'CAF21227.1', 'CAF21228.1', 'CAF21229.1', 'CAF21230.1', 'CAF21231.1', 'CAF21232.1', 'CAF21233.1', 'CAF21234.1', 'CAF21235.1', 'CAF21236.1', 'CAF21237.1', 'CAF21238.1', 'CAF21240.1', 'CAF21239.1', 'CAF21241.1', 'CAF21242.1', 'CAF21243.1', 'CAF21245.1', 'CAF21246.1', 'CAF21247.1', 'CAF21248.1', 'CAF21249.1', 'CAF21250.1', 'CAF21251.1', 'CAF21252.1', 'CAF21253.1', 'CAF21254.1', 'CAF21255.1', 'CAF21256.1', 'CAF21257.1', 'CAF21258.1', 'CAF21259.1', 'CAF21260.1', 'CAF21261.1', 'CAF21262.1', 'CAF21263.1', 'CAF21264.1', 'CAF21265.1', 'CAF21266.1', 'CAF21267.1', 'CAF21268.1', 'CAF21269.1', 'CAF21270.1', 'CAF21271.1', 'CAF21272.1', 'CAF21273.1', 'CAF21274.1', 'CAF21275.1', 'CAF21276.1', 'CAF21277.1', 'CAF21278.1', 'CAF21279.1', 'CAF21280.1', 'CAF21281.1', 'CAF21282.1', 'CAF21283.1', 'CAF21284.1', 'CAF21285.1', 'CAF21286.1', 'CAF21287.1', 'CAF21288.1', 'CAF21289.1', 'CAF21290.1', 'CAF21291.1', 'CAF21292.1', 'CAF21294.1', 'CAF21295.1', 'CAF21296.1', 'CAF21297.1', 'CAF21298.1', 'CAF21299.1', 'CAF21300.1', 'CAF21301.1', 'CAF21302.1', 'CAF21303.1', 'CAF21304.1', 'CAF21305.1', 'CAF21306.1', 'CAF21307.1', 'CAF21308.1', 'CAF21309.1', 'CAF21310.1', 'CAF21311.1', 'CAF21312.1', 'CAF20676.1', 'CAF20677.1', 'CAF20678.1', 'CAF20679.1', 'CAF20680.1', 'CAF20681.1', 'CAF20682.1', 'CAF20683.1', 'CAF20684.1', 'CAF20685.1', 'CAF20686.1', 'CAF20687.1', 'CAF20688.1', 'CAF20689.1', 'CAF20690.1', 'CAF20691.1', 'CAF20692.1', 'CAF20693.1', 'CAF20694.1', 'CAF20695.1', 'CAF20696.1', 'CAF20697.1', 'CAF20698.1', 'CAF20699.1', 'CAF20700.1', 'CAF20701.1', 'CAF20702.1', 'CAF20703.1', 'CAF20704.1', 'CAF20705.1', 'CAF20706.1', 'CAF20707.1', 'CAF20708.1', 'CAF20709.1', 'CAF20710.1', 'CAF20711.1', 'CAF20712.1', 'CAF20713.1', 'CAF20714.1', 'CAF20715.1', 'CAF20716.1', 'CAF20717.1', 'CAF20718.1', 'CAF20719.1', 'CAF20720.1', 'CAF20721.1', 'CAF20722.1', 'CAF20723.1', 'CAF20724.1', 'CAF20725.1', 'CAF20726.1', 'CAF20727.1', 'CAF20728.1', 'CAF20729.1', 'CAF20730.1', 'CAF20731.1', 'CAF20732.1', 'CAF20733.1', 'CAF20734.1', 'CAF20735.1', 'CAF20736.1', 'CAF20737.1', 'CAF20738.1', 'CAF20739.1', 'CAF20740.1', 'CAF20741.1', 'CAF20742.1', 'CAF20743.1', 'CAF20744.1', 'CAF20745.1', 'CAF20746.1', 'CAF20747.1', 'CAF20748.1', 'CAF20749.1', 'CAF20750.1', 'CAF20751.1', 'CAF20752.1', 'CAF20753.1', 'CAF20754.1', 'CAF20755.1', 'CAF20756.1', 'CAF20757.1', 'CAF20758.1', 'CAF20759.1', 'CAF20760.1', 'CAF20761.1', 'CAF20762.1', 'CAF20763.1', 'CAF20764.1', 'CAF20765.1', 'CAF20766.1', 'CAF20767.1', 'CAF20768.1', 'CAF20769.1', 'CAF20770.1', 'CAF20771.1', 'CAF20772.1', 'CAF20773.1', 'CAF20774.1', 'CAF20775.1', 'CAF20776.1', 'CAF20777.1', 'CAF20778.1', 'CAF20779.1', 'CAF20780.1', 'CAF20781.1', 'CAF20782.1', 'CAF20783.1', 'CAF20784.1', 'CAF20785.1', 'CAF20786.1', 'CAF20787.1', 'CAF20788.1', 'CAF20789.1', 'CAF20790.1', 'CAF20791.1', 'CAF20792.1', 'CAF20793.1', 'CAF20794.1', 'CAF20795.1', 'CAF20796.1', 'CAF20797.1', 'CAF20798.1', 'CAF20799.1', 'CAF20800.1', 'CAF20801.1', 'CAF20802.1', 'CAF20803.1', 'CAF20804.1', 'CAF20805.1', 'CAF20806.1', 'CAF20807.1', 'CAF20808.1', 'CAF20809.1', 'CAF20810.1', 'CAF20811.1', 'CAF20812.1', 'CAF20813.1', 'CAF20814.1', 'CAF20815.1', 'CAF20816.1', 'CAF20817.1', 'CAF20818.1', 'CAF20819.1', 'CAF20820.1', 'CAF20821.1', 'CAF20822.1', 'CAF20823.1', 'CAF20824.1', 'CAF20825.1', 'CAF20826.1', 'CAF20827.1', 'CAF20828.1', 'CAF20830.1', 'CAF20829.1', 'CAF20831.1', 'CAF20832.1', 'CAF20833.1', 'CAF20834.1', 'CAF20835.1', 'CAF20836.1', 'CAF20837.1', 'CAF20838.1', 'CAF20839.1', 'CAF20840.1', 'CAF20841.1', 'CAF20842.1', 'CAF20843.1', 'CAF20844.1', 'CAF20845.1', 'CAF20846.1', 'CAF20847.1', 'CAF20848.1', 'CAF20849.1', 'CAF20850.1', 'CAF20851.1', 'CAF20852.1', 'CAF20853.1', 'CAF20854.1', 'CAF20855.1', 'CAF20856.1', 'CAF20857.1', 'CAF20858.1', 'CAF20859.1', 'CAF20860.1', 'CAF20861.1', 'CAF20862.1', 'CAF20863.1', 'CAF20864.1', 'CAF20865.1', 'CAF20866.1', 'CAF20867.1', 'CAF20869.1', 'CAF20868.1', 'CAF20870.1', 'CAF20872.1', 'CAF20873.1', 'CAF20874.1', 'CAF20875.1', 'CAF20876.1', 'CAF20877.1', 'CAF20878.1', 'CAF20879.1', 'CAF20880.1', 'CAF20881.1', 'CAF20882.1', 'CAF20883.1', 'CAF20884.1', 'CAF20885.1', 'CAF20886.1', 'CAF20887.1', 'CAF20888.1', 'CAF20889.1', 'CAF20890.1', 'CAF20891.1', 'CAF20892.1', 'CAF20893.1', 'CAF20894.1', 'CAF20895.1', 'CAF20896.1', 'CAF20897.1', 'CAF20898.1', 'CAF20899.1', 'CAF20900.1', 'CAF20901.1', 'CAF20902.1', 'CAF20903.1', 'CAF20904.1', 'CAF20905.1', 'CAF20906.1', 'CAF20907.1', 'CAF20908.1', 'CAF20909.1', 'CAF20910.1', 'CAF20911.1', 'CAF20912.1', 'CAF20913.1', 'CAF20914.1', 'CAF20915.1', 'CAF20916.1', 'CAF20917.1', 'CAF20918.1', 'CAF20919.1', 'CAF20920.1', 'CAF20921.1', 'CAF20922.1', 'CAF20923.1', 'CAF20924.1', 'CAF20925.1', 'CAF20926.1', 'CAF20927.1', 'CAF20928.1', 'CAF20929.1', 'CAF20930.1', 'CAF20931.1', 'CAF20932.1', 'CAF20933.1', 'CAF20934.1', 'CAF20935.1', 'CAF20936.1', 'CAF20937.1', 'CAF20938.1', 'CAF20939.1', 'CAF20940.1', 'CAF20941.1', 'CAF20942.1', 'CAF20943.1', 'CAF20944.1', 'CAF20945.1', 'CAF20946.1', 'CAF20947.1', 'CAF20948.1', 'CAF20949.1', 'CAF20950.1', 'CAF20951.1', 'CAF20952.1', 'CAF20953.1', 'CAF20954.1', 'CAF20955.1', 'CAF20956.1', 'CAF20957.1', 'CAF20958.1', 'CAF20959.1', 'CAF20961.1', 'CAF20960.1', 'CAF20962.1', 'CAF20963.1', 'CAF20964.1', 'CAF20965.1', 'CAF20966.1', 'CAF20967.1', 'CAF20968.1', 'CAF20970.1', 'CAF20969.1', 'CAF20971.1', 'CAF20972.1', 'CAF20973.1', 'CAF20974.1', 'CAF20975.1', 'CAF20976.1', 'CAF20977.1', 'CAF20978.1', 'CAF20979.1', 'CAF20980.1', 'CAF20981.1', 'CAF20982.1', 'CAF20983.1', 'CAF20984.1', 'CAF20985.1', 'CAF20986.1', 'CAF20987.1', 'CAF20988.1', 'CAF20989.1', 'CAF20990.1', 'CAF20991.1', 'CAF20992.1', 'CAF20993.1', 'CAF20994.1', 'CAF20995.1', 'CAF20996.1', 'CAF18903.1', 'CAF18904.1', 'CAF18905.1', 'CAF18906.1', 'CAF18907.1', 'CAF18908.1', 'CAF18909.1', 'CAF18910.1', 'CAF18911.1', 'CAF18912.1', 'CAF18913.1', 'CAF18914.1', 'CAF18915.1', 'CAF18916.1', 'CAF18917.1', 'CAF18918.1', 'CAF18919.1', 'CAF18920.1', 'CAF18921.1', 'CAF18922.1', 'CAF18923.1', 'CAF18924.1', 'CAF18925.1', 'CAF18926.1', 'CAF18927.1', 'CAF18928.1', 'CAF18929.1', 'CAF18930.1', 'CAF18931.1', 'CAF18932.1', 'CAF18933.1', 'CAF18934.1', 'CAF18935.1', 'CAF18936.1', 'CAF18937.1', 'CAF18938.1', 'CAF18939.1', 'CAF18940.1', 'CAF18941.1', 'CAF18942.1', 'CAF18943.1', 'CAF18944.1', 'CAF18945.1', 'CAF18946.1', 'CAF18947.1', 'CAF18948.1', 'CAF18949.1', 'CAF18950.1', 'CAF18951.1', 'CAF18952.1', 'CAF18953.1', 'CAF18954.1', 'CAF18955.1', 'CAF18956.1', 'CAF18957.1', 'CAF18958.1', 'CAF18959.1', 'CAF18960.1', 'CAF18961.1', 'CAF18962.1', 'CAF18963.1', 'CAF18964.1', 'CAF18965.1', 'CAF18966.1', 'CAF18967.1', 'CAF18968.1', 'CAF18970.1', 'CAF18969.1', 'CAF18971.1', 'CAF18972.1', 'CAF18973.1', 'CAF18974.1', 'CAF18975.1', 'CAF18976.1', 'CAF18977.1', 'CAF18978.1', 'CAF18979.1', 'CAF18980.1', 'CAF18981.1', 'CAF18982.1', 'CAF18983.1', 'CAF18984.1', 'CAF18985.1', 'CAF18986.1', 'CAF18987.1', 'CAF18988.1', 'CAF18989.1', 'CAF18990.1', 'CAF18991.1', 'CAF18992.1', 'CAF18993.1', 'CAF18994.1', 'CAF18995.1', 'CAF18996.1', 'CAF18997.1', 'CAF18998.1', 'CAF18999.1', 'CAF19000.1', 'CAF19001.1', 'CAF19002.1', 'CAF19003.1', 'CAF19004.1', 'CAF19005.1', 'CAF19006.1', 'CAF19007.1', 'CAF19008.1', 'CAF19009.1', 'CAF19010.1', 'CAF19011.1', 'CAF19012.1', 'CAF19013.1', 'CAF19014.1', 'CAF19015.1', 'CAF19016.1', 'CAF19017.1', 'CAF19018.1', 'CAF19019.1', 'CAF19020.1', 'CAF19021.1', 'CAF19022.1', 'CAF19023.1', 'CAF19024.1', 'CAF19025.1', 'CAF19026.1', 'CAF19027.1', 'CAF19028.1', 'CAF19029.1', 'CAF19030.1', 'CAF19031.1', 'CAF19032.1', 'CAF19033.1', 'CAF19034.1', 'CAF19035.1', 'CAF19036.1', 'CAF19037.1', 'CAF19038.1', 'CAF19039.1', 'CAF19040.1', 'CAF19041.1']\n",
      "{'jobId': 'cc09d8dbd461815ca34595258553bca2e5f3c570'}\n",
      "{'results': [{'from': 'CAF18566.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NUD8', 'uniProtkbId': 'DNAA_CORGL', 'entryAudit': {'firstPublicDate': '2003-11-07', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 115, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00377'}], 'value': 'Chromosomal replication initiator protein DnaA'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00377'}], 'value': 'dnaA'}, 'orderedLocusNames': [{'value': 'Cgl0001'}, {'value': 'cg0001'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00377'}], 'value': \"Plays an important role in the initiation and regulation of chromosomal replication. Binds to the origin of replication; it binds specifically double-stranded DNA at a 9 bp consensus (dnaA box): 5'-TTATC[CA]A[CA]A-3'. DnaA binds to ATP and to acidic phospholipids\"}], 'commentType': 'FUNCTION'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00377'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00377'}], 'value': 'Belongs to the DnaA family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 524, 'modifier': 'EXACT'}}, 'description': 'Chromosomal replication initiator protein DnaA', 'featureId': 'PRO_0000114169'}, {'type': 'Region', 'location': {'start': {'value': 95, 'modifier': 'EXACT'}, 'end': {'value': 183, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Binding site', 'location': {'start': {'value': 224, 'modifier': 'EXACT'}, 'end': {'value': 231, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00377'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}], 'keywords': [{'id': 'KW-0067', 'category': 'Ligand', 'name': 'ATP-binding'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0235', 'category': 'Biological process', 'name': 'DNA replication'}, {'id': 'KW-0238', 'category': 'Molecular function', 'name': 'DNA-binding'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97394.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18566.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599253.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013309.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NUD8', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NUD8', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0001', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97394', 'properties': [{'key': 'ProteinId', 'value': 'BAB97394'}, {'key': 'GeneId', 'value': 'BAB97394'}]}, {'database': 'KEGG', 'id': 'cgb:cg0001', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0001', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.1', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0593', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_026910_2_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'DFIHFYQ', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9807019at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0005524', 'properties': [{'key': 'GoTerm', 'value': 'F:ATP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0003688', 'properties': [{'key': 'GoTerm', 'value': 'F:DNA replication origin binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006270', 'properties': [{'key': 'GoTerm', 'value': 'P:DNA replication initiation'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006275', 'properties': [{'key': 'GoTerm', 'value': 'P:regulation of DNA replication'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd00009', 'properties': [{'key': 'EntryName', 'value': 'AAA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'CDD', 'id': 'cd06571', 'properties': [{'key': 'EntryName', 'value': 'Bac_DnaA_C'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '1.10.1750.10', 'properties': [{'key': 'EntryName', 'value': '-'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '1.10.8.60', 'properties': [{'key': 'EntryName', 'value': '-'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.40.50.300', 'properties': [{'key': 'EntryName', 'value': 'P-loop containing nucleotide triphosphate hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00377', 'properties': [{'key': 'EntryName', 'value': 'DnaA_bact'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR003593', 'properties': [{'key': 'EntryName', 'value': 'AAA+_ATPase'}]}, {'database': 'InterPro', 'id': 'IPR001957', 'properties': [{'key': 'EntryName', 'value': 'Chromosome_initiator_DnaA'}]}, {'database': 'InterPro', 'id': 'IPR020591', 'properties': [{'key': 'EntryName', 'value': 'Chromosome_initiator_DnaA-like'}]}, {'database': 'InterPro', 'id': 'IPR018312', 'properties': [{'key': 'EntryName', 'value': 'Chromosome_initiator_DnaA_CS'}]}, {'database': 'InterPro', 'id': 'IPR013317', 'properties': [{'key': 'EntryName', 'value': 'DnaA'}]}, {'database': 'InterPro', 'id': 'IPR013159', 'properties': [{'key': 'EntryName', 'value': 'DnaA_C'}]}, {'database': 'InterPro', 'id': 'IPR027417', 'properties': [{'key': 'EntryName', 'value': 'P-loop_NTPase'}]}, {'database': 'InterPro', 'id': 'IPR010921', 'properties': [{'key': 'EntryName', 'value': 'Trp_repressor/repl_initiator'}]}, {'database': 'PANTHER', 'id': 'PTHR30050', 'properties': [{'key': 'EntryName', 'value': 'CHROMOSOMAL REPLICATION INITIATOR PROTEIN DNAA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR30050:SF2', 'properties': [{'key': 'EntryName', 'value': 'CHROMOSOMAL REPLICATION INITIATOR PROTEIN DNAA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00308', 'properties': [{'key': 'EntryName', 'value': 'Bac_DnaA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF08299', 'properties': [{'key': 'EntryName', 'value': 'Bac_DnaA_C'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PRINTS', 'id': 'PR00051', 'properties': [{'key': 'EntryName', 'value': 'DNAA'}]}, {'database': 'SMART', 'id': 'SM00382', 'properties': [{'key': 'EntryName', 'value': 'AAA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SMART', 'id': 'SM00760', 'properties': [{'key': 'EntryName', 'value': 'Bac_DnaA_C'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF52540', 'properties': [{'key': 'EntryName', 'value': 'P-loop containing nucleoside triphosphate hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF48295', 'properties': [{'key': 'EntryName', 'value': 'TrpR-like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00362', 'properties': [{'key': 'EntryName', 'value': 'DnaA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS01008', 'properties': [{'key': 'EntryName', 'value': 'DNAA'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSQNSSSLLETWRQVVADLTTLSQQADSGFDPLTPTQRAYLNLTKPIAIVDGYAVLSTPNAMAKNVIENDLGDALTRVLSLRMGRSFSLAVSVEPEQEIPETPAQQEFKYQPDAPVISSNKAPKQYEVGGRGEASTSDGWERTHSAPAPEPHPAPIADPEPELATPQRIPRETPAHNPNREVSLNPKYTFESFVIGPFNRFANAAAVAVAESPAKAFNPLFISGGSGLGKTHLLHAVGNYAQELQPGLRIKYVSSEEFTNDYINSVRDDRQETFKRRYRNLDILMVDDIQFLAGKEGTQEEFFHTFNALHQADKQIILSSDRPPKQLTTLEDRLRTRFEGGLITDIQPPDLETRIAILMKKAQTDGTHVDREVLELIASRFESSIRELEGALIRVSAYSSLINQPIDKEMAIVALRDILPEPEDMEITAPVIMEVTAEYFEISVDTLRGAGKTRAVAHARQLAMYLCRELTDMSLPKIGDVFGGKDHTTVMYADRKIRQEMTEKRDTYDEIQQLTQLIKSRGRN', 'length': 524, 'molWeight': 58508, 'crc64': '3B934D1E3F93CD2E', 'md5': 'A137BEF0BD3B42E3FCDA02E0B63D92DB'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBCELLULAR LOCATION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Region': 1, 'Binding site': 1}, 'uniParcId': 'UPI0000039FAF'}}}, {'from': 'CAF18569.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q6M8X7', 'secondaryAccessions': ['Q8NUD5'], 'uniProtkbId': 'RECF_CORGL', 'entryAudit': {'firstPublicDate': '2004-08-16', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 109, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00365'}], 'value': 'DNA replication and repair protein RecF'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00365'}], 'value': 'recF'}, 'orderedLocusNames': [{'value': 'Cgl0004'}, {'value': 'cg0005'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00365'}], 'value': 'The RecF protein is involved in DNA metabolism; it is required for DNA replication and normal SOS inducibility. RecF binds preferentially to single-stranded, linear DNA. It also seems to bind ATP'}], 'commentType': 'FUNCTION'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00365'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00365'}], 'value': 'Belongs to the RecF family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 394, 'modifier': 'EXACT'}}, 'description': 'DNA replication and repair protein RecF', 'featureId': 'PRO_0000196411'}, {'type': 'Binding site', 'location': {'start': {'value': 30, 'modifier': 'EXACT'}, 'end': {'value': 37, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00365'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}], 'keywords': [{'id': 'KW-0067', 'category': 'Ligand', 'name': 'ATP-binding'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0227', 'category': 'Biological process', 'name': 'DNA damage'}, {'id': 'KW-0234', 'category': 'Biological process', 'name': 'DNA repair'}, {'id': 'KW-0235', 'category': 'Biological process', 'name': 'DNA replication'}, {'id': 'KW-0238', 'category': 'Molecular function', 'name': 'DNA-binding'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0742', 'category': 'Biological process', 'name': 'SOS response'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97397.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18569.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599255.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013310.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q6M8X7', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q6M8X7', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0005', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97397', 'properties': [{'key': 'ProteinId', 'value': 'BAB97397'}, {'key': 'GeneId', 'value': 'BAB97397'}]}, {'database': 'KEGG', 'id': 'cgb:cg0005', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0004', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.3', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG1195', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_040267_1_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'GESWSYA', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9803889at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0005524', 'properties': [{'key': 'GoTerm', 'value': 'F:ATP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0003697', 'properties': [{'key': 'GoTerm', 'value': 'F:single-stranded DNA binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006281', 'properties': [{'key': 'GoTerm', 'value': 'P:DNA repair'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006260', 'properties': [{'key': 'GoTerm', 'value': 'P:DNA replication'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0009432', 'properties': [{'key': 'GoTerm', 'value': 'P:SOS response'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd03242', 'properties': [{'key': 'EntryName', 'value': 'ABC_RecF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.40.50.300', 'properties': [{'key': 'EntryName', 'value': 'P-loop containing nucleotide triphosphate hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '1.20.1050.90', 'properties': [{'key': 'EntryName', 'value': 'RecF/RecN/SMC, N-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00365', 'properties': [{'key': 'EntryName', 'value': 'RecF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR001238', 'properties': [{'key': 'EntryName', 'value': 'DNA-binding_RecF'}]}, {'database': 'InterPro', 'id': 'IPR018078', 'properties': [{'key': 'EntryName', 'value': 'DNA-binding_RecF_CS'}]}, {'database': 'InterPro', 'id': 'IPR027417', 'properties': [{'key': 'EntryName', 'value': 'P-loop_NTPase'}]}, {'database': 'InterPro', 'id': 'IPR003395', 'properties': [{'key': 'EntryName', 'value': 'RecF/RecN/SMC_N'}]}, {'database': 'InterPro', 'id': 'IPR042174', 'properties': [{'key': 'EntryName', 'value': 'RecF_2'}]}, {'database': 'PANTHER', 'id': 'PTHR32182', 'properties': [{'key': 'EntryName', 'value': 'DNA REPLICATION AND REPAIR PROTEIN RECF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR32182:SF0', 'properties': [{'key': 'EntryName', 'value': 'DNA REPLICATION AND REPAIR PROTEIN RECF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02463', 'properties': [{'key': 'EntryName', 'value': 'SMC_N'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF52540', 'properties': [{'key': 'EntryName', 'value': 'P-loop containing nucleoside triphosphate hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00611', 'properties': [{'key': 'EntryName', 'value': 'recf'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS00617', 'properties': [{'key': 'EntryName', 'value': 'RECF_1'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS00618', 'properties': [{'key': 'EntryName', 'value': 'RECF_2'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MHIRSLELRDYRSWPELKVDLEPGITVFIGRNGFGKTNIVEAIGYLAHLSSHRVSSDAPLVRAHAENARVSAVAVNQGRELAAHLLIKPHAANQASLNRTKVRTPRELLGVVKTVLFAPEDLALVKGEPAERRRYLDDIIATRQPRMAGVKADYDKVLKQRNALLKTATIALRRGYGTEEGAAALSTLDTWDGQLARLGAEVMAARFALLNELGPKIYEAYTTIAPESRPAAVNYKTTIDQGLSQFSEFDAGIIEATLLTELAAKRQREIERGSSLVGPHRDDVDLMLGDQPAKGFASHGETWSFALSLRIAEFNLLKSDGTDPILILDDVFSELDAGRREKLVGIAQEVEQVLITAAVHDDLPENLKKVLTAQHTVTVQDTGTGRISLLDVQP', 'length': 394, 'molWeight': 43185, 'crc64': '9601D278C43151E3', 'md5': 'F30DD71E780304A2548587CC174DCEAA'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBCELLULAR LOCATION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Binding site': 1}, 'uniParcId': 'UPI0000039FB2'}}}, {'from': 'CAF18570.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NUD4', 'uniProtkbId': 'Y005_CORGL', 'entryAudit': {'firstPublicDate': '2003-04-04', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 87, 'sequenceVersion': 1}, 'annotationScore': 1.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'value': 'UPF0232 protein Cgl0005/cg0006'}}}, 'genes': [{'orderedLocusNames': [{'value': 'Cgl0005'}, {'value': 'cg0006'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Belongs to the UPF0232 family'}], 'commentType': 'SIMILARITY'}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Erroneous initiation', 'sequence': 'CAF18570.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 178, 'modifier': 'EXACT'}}, 'description': 'UPF0232 protein Cgl0005/cg0006', 'featureId': 'PRO_0000211360'}, {'type': 'Region', 'location': {'start': {'value': 16, 'modifier': 'EXACT'}, 'end': {'value': 55, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Compositional bias', 'location': {'start': {'value': 37, 'modifier': 'EXACT'}, 'end': {'value': 55, 'modifier': 'EXACT'}}, 'description': 'Basic and acidic residues', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}], 'keywords': [{'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97398.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18570.1'}, {'key': 'Status', 'value': 'ALT_INIT'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599256.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003855338.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NUD4', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0006', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97398', 'properties': [{'key': 'ProteinId', 'value': 'BAB97398'}, {'key': 'GeneId', 'value': 'BAB97398'}]}, {'database': 'GeneID', 'id': '69620396', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgb:cg0006', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0005', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.4', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG5512', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_087206_1_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'RWPQIVG', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '5516926at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'HAMAP', 'id': 'MF_00630', 'properties': [{'key': 'EntryName', 'value': 'UPF0232'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR007922', 'properties': [{'key': 'EntryName', 'value': 'DUF721/UPF0232'}]}, {'database': 'InterPro', 'id': 'IPR023007', 'properties': [{'key': 'EntryName', 'value': 'UPF0232_actinobac'}]}, {'database': 'PANTHER', 'id': 'PTHR36456', 'properties': [{'key': 'EntryName', 'value': 'UPF0232 PROTEIN SCO3875'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR36456:SF1', 'properties': [{'key': 'EntryName', 'value': 'UPF0232 PROTEIN SCO3875'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF05258', 'properties': [{'key': 'EntryName', 'value': 'DciA'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MTDPIEQAFERIRAEAMRRNGSVPDLNKNDAFRRPPAPKGGVEKRKKGRASGLDGRQKRYVRGAESLGSVLNKEIQRRGWGKDIAGGWVTSNWEELVGAKIAQHTRVEMIKDKKLFITCDSTAWATNLRMMQRQILQVIAEKVGPNIITELRIFGPQAPSWRKGPLHVKGRGPRDTYG', 'length': 178, 'molWeight': 20058, 'crc64': '55D00847F6EBE238', 'md5': 'ED225FF503F0D187DF2D93B82314F967'}, 'extraAttributes': {'countByCommentType': {'SIMILARITY': 1, 'SEQUENCE CAUTION': 1}, 'countByFeatureType': {'Chain': 1, 'Region': 1, 'Compositional bias': 1}, 'uniParcId': 'UPI0000039FB3'}}}, {'from': 'CAF18580.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q6M8W7', 'secondaryAccessions': ['Q8NUC4'], 'uniProtkbId': 'Y015_CORGL', 'entryAudit': {'firstPublicDate': '2004-08-16', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 91, 'sequenceVersion': 1}, 'annotationScore': 1.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'value': 'UPF0324 membrane protein Cgl0015/cg0018'}}}, 'genes': [{'orderedLocusNames': [{'value': 'Cgl0015'}, {'value': 'cg0018'}]}], 'comments': [{'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Cell membrane', 'id': 'SL-0039'}, 'topology': {'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Multi-pass membrane protein', 'id': 'SL-9909'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Belongs to the UPF0324 family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 345, 'modifier': 'EXACT'}}, 'description': 'UPF0324 membrane protein Cgl0015/cg0018', 'featureId': 'PRO_0000157409'}, {'type': 'Transmembrane', 'location': {'start': {'value': 15, 'modifier': 'EXACT'}, 'end': {'value': 37, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 44, 'modifier': 'EXACT'}, 'end': {'value': 66, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 81, 'modifier': 'EXACT'}, 'end': {'value': 103, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 105, 'modifier': 'EXACT'}, 'end': {'value': 124, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 134, 'modifier': 'EXACT'}, 'end': {'value': 156, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 163, 'modifier': 'EXACT'}, 'end': {'value': 185, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 205, 'modifier': 'EXACT'}, 'end': {'value': 227, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 261, 'modifier': 'EXACT'}, 'end': {'value': 280, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 285, 'modifier': 'EXACT'}, 'end': {'value': 307, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 320, 'modifier': 'EXACT'}, 'end': {'value': 342, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}], 'keywords': [{'id': 'KW-1003', 'category': 'Cellular component', 'name': 'Cell membrane'}, {'id': 'KW-0472', 'category': 'Cellular component', 'name': 'Membrane'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0812', 'category': 'Domain', 'name': 'Transmembrane'}, {'id': 'KW-1133', 'category': 'Domain', 'name': 'Transmembrane helix'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97408.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18580.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599266.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011265442.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q6M8W7', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0018', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97408', 'properties': [{'key': 'ProteinId', 'value': 'BAB97408'}, {'key': 'GeneId', 'value': 'BAB97408'}]}, {'database': 'KEGG', 'id': 'cgb:cg0018', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0015', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.14', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG2855', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_033541_1_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'FGLWCAI', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9766798at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005886', 'properties': [{'key': 'GoTerm', 'value': 'C:plasma membrane'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'InterPro', 'id': 'IPR018383', 'properties': [{'key': 'EntryName', 'value': 'UPF0324_pro'}]}, {'database': 'PANTHER', 'id': 'PTHR30106', 'properties': [{'key': 'EntryName', 'value': 'INNER MEMBRANE PROTEIN YEIH-RELATED'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR30106:SF2', 'properties': [{'key': 'EntryName', 'value': 'UPF0324 INNER MEMBRANE PROTEIN YEIH'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF03601', 'properties': [{'key': 'EntryName', 'value': 'Cons_hypoth698'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSTNLLESTPPFTQLRTGVLQKYTPGLLLCSIAVLIAMIVNHFFSGVSPLIVAIILGIILTNLIQLPASTSPGITLASKKLLRLGIVFLGLQLVFSDILSLGFPMLAVIVCIVAGGIFGTILMGHLLRMKPTQVLLIACGFSICGAAAVAGVEGVTDSEEEEVVTAVALVVIFGTLMIPFIPFATKVLGLSPEIGGMWAGGSIHEIAQVVAAGGVIGGGALGVAVVVKLARVLLLAPIAAILSFRQRRQGYTSPDGKRPPVVPLFILGFLAMVVLRSTVALPDEVIAAGGFLQTALLSAAMFGLGCGVKIQNLIHVGVKPFILAFGSTTLVTSIALAGTLLTHLG', 'length': 345, 'molWeight': 35577, 'crc64': '43BFC76E094A52B9', 'md5': '2580BC0C1CCCB9C0EAF5B28ED4E2E5A7'}, 'extraAttributes': {'countByCommentType': {'SUBCELLULAR LOCATION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Transmembrane': 10}, 'uniParcId': 'UPI0000039FBE'}}}, {'from': 'CAF18608.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NU99', 'uniProtkbId': 'CRGA_CORGL', 'entryAudit': {'firstPublicDate': '2003-04-04', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 100, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}], 'value': 'Cell division protein CrgA'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}], 'value': 'crgA'}, 'orderedLocusNames': [{'value': 'Cgl0040'}, {'value': 'cg0055'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}], 'value': 'Involved in cell division'}], 'commentType': 'FUNCTION'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}], 'value': 'Cell membrane', 'id': 'SL-0039'}, 'topology': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}], 'value': 'Multi-pass membrane protein', 'id': 'SL-9909'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}], 'value': 'Belongs to the CrgA family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 90, 'modifier': 'EXACT'}}, 'description': 'Cell division protein CrgA', 'featureId': 'PRO_0000216812'}, {'type': 'Transmembrane', 'location': {'start': {'value': 38, 'modifier': 'EXACT'}, 'end': {'value': 58, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 67, 'modifier': 'EXACT'}, 'end': {'value': 87, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00631'}]}, {'type': 'Region', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 25, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Compositional bias', 'location': {'start': {'value': 9, 'modifier': 'EXACT'}, 'end': {'value': 25, 'modifier': 'EXACT'}}, 'description': 'Polar residues', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}], 'keywords': [{'id': 'KW-0131', 'category': 'Biological process', 'name': 'Cell cycle'}, {'id': 'KW-0132', 'category': 'Biological process', 'name': 'Cell division'}, {'id': 'KW-1003', 'category': 'Cellular component', 'name': 'Cell membrane'}, {'id': 'KW-0472', 'category': 'Cellular component', 'name': 'Membrane'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0812', 'category': 'Domain', 'name': 'Transmembrane'}, {'id': 'KW-1133', 'category': 'Domain', 'name': 'Transmembrane helix'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97433.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18608.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599292.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003861074.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NU99', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NU99', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0055', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97433', 'properties': [{'key': 'ProteinId', 'value': 'BAB97433'}, {'key': 'GeneId', 'value': 'BAB97433'}]}, {'database': 'KEGG', 'id': 'cgb:cg0055', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0040', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.40', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'ENOG5032ZHR', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_149126_2_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'WNIVVGF', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '5189646at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005886', 'properties': [{'key': 'GoTerm', 'value': 'C:plasma membrane'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0007049', 'properties': [{'key': 'GoTerm', 'value': 'P:cell cycle'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0051301', 'properties': [{'key': 'GoTerm', 'value': 'P:cell division'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'HAMAP', 'id': 'MF_00631', 'properties': [{'key': 'EntryName', 'value': 'CrgA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR009619', 'properties': [{'key': 'EntryName', 'value': 'CrgA'}]}, {'database': 'Pfam', 'id': 'PF06781', 'properties': [{'key': 'EntryName', 'value': 'CrgA'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MPKARVTKNETAPVSSNPSANRTPVKINSAGTPMWYKVIMFAFMIVGLAWLIINYLVGPQIPFMADLGAWNYGIGFGLMIIGLLMTMGWR', 'length': 90, 'molWeight': 9950, 'crc64': 'F9E9E657A4089E5D', 'md5': '15758A81F8279B420CCDA088458273D4'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBCELLULAR LOCATION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Transmembrane': 2, 'Region': 1, 'Compositional bias': 1}, 'uniParcId': 'UPI0000039FD5'}}}, {'from': 'CAF18609.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NU98', 'uniProtkbId': 'PKN1_CORGL', 'entryAudit': {'firstPublicDate': '2003-04-04', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 126, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'value': 'Probable serine/threonine-protein kinase PknB'}, 'ecNumbers': [{'value': '2.7.11.1'}]}}, 'genes': [{'geneName': {'value': 'pknB'}, 'orderedLocusNames': [{'value': 'Cgl0041'}, {'value': 'cg0057'}]}], 'comments': [{'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'ATP + L-seryl-[protein] = ADP + H(+) + O-phospho-L-seryl-[protein]', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:17989'}, {'database': 'Rhea', 'id': 'RHEA-COMP:9863'}, {'database': 'Rhea', 'id': 'RHEA-COMP:11604'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:29999'}, {'database': 'ChEBI', 'id': 'CHEBI:30616'}, {'database': 'ChEBI', 'id': 'CHEBI:83421'}, {'database': 'ChEBI', 'id': 'CHEBI:456216'}], 'ecNumber': '2.7.11.1'}}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'ATP + L-threonyl-[protein] = ADP + H(+) + O-phospho-L-threonyl-[protein]', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:46608'}, {'database': 'Rhea', 'id': 'RHEA-COMP:11060'}, {'database': 'Rhea', 'id': 'RHEA-COMP:11605'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:30013'}, {'database': 'ChEBI', 'id': 'CHEBI:30616'}, {'database': 'ChEBI', 'id': 'CHEBI:61977'}, {'database': 'ChEBI', 'id': 'CHEBI:456216'}], 'ecNumber': '2.7.11.1'}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}], 'value': 'Belongs to the protein kinase superfamily. Ser/Thr protein kinase family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 646, 'modifier': 'EXACT'}}, 'description': 'Probable serine/threonine-protein kinase PknB', 'featureId': 'PRO_0000171197'}, {'type': 'Domain', 'location': {'start': {'value': 9, 'modifier': 'EXACT'}, 'end': {'value': 278, 'modifier': 'EXACT'}}, 'description': 'Protein kinase', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}]}, {'type': 'Domain', 'location': {'start': {'value': 365, 'modifier': 'EXACT'}, 'end': {'value': 431, 'modifier': 'EXACT'}}, 'description': 'PASTA 1', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00528'}]}, {'type': 'Domain', 'location': {'start': {'value': 432, 'modifier': 'EXACT'}, 'end': {'value': 500, 'modifier': 'EXACT'}}, 'description': 'PASTA 2', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00528'}]}, {'type': 'Domain', 'location': {'start': {'value': 501, 'modifier': 'EXACT'}, 'end': {'value': 565, 'modifier': 'EXACT'}}, 'description': 'PASTA 3', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00528'}]}, {'type': 'Domain', 'location': {'start': {'value': 570, 'modifier': 'EXACT'}, 'end': {'value': 638, 'modifier': 'EXACT'}}, 'description': 'PASTA 4', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00528'}]}, {'type': 'Region', 'location': {'start': {'value': 283, 'modifier': 'EXACT'}, 'end': {'value': 333, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Compositional bias', 'location': {'start': {'value': 287, 'modifier': 'EXACT'}, 'end': {'value': 302, 'modifier': 'EXACT'}}, 'description': 'Basic and acidic residues', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Compositional bias', 'location': {'start': {'value': 303, 'modifier': 'EXACT'}, 'end': {'value': 333, 'modifier': 'EXACT'}}, 'description': 'Polar residues', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Active site', 'location': {'start': {'value': 136, 'modifier': 'EXACT'}, 'end': {'value': 136, 'modifier': 'EXACT'}}, 'description': 'Proton acceptor', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}, {'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU10027'}]}, {'type': 'Binding site', 'location': {'start': {'value': 15, 'modifier': 'EXACT'}, 'end': {'value': 23, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}, {'type': 'Binding site', 'location': {'start': {'value': 38, 'modifier': 'EXACT'}, 'end': {'value': 38, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}], 'keywords': [{'id': 'KW-0067', 'category': 'Ligand', 'name': 'ATP-binding'}, {'id': 'KW-0418', 'category': 'Molecular function', 'name': 'Kinase'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0677', 'category': 'Domain', 'name': 'Repeat'}, {'id': 'KW-0723', 'category': 'Molecular function', 'name': 'Serine/threonine-protein kinase'}, {'id': 'KW-0808', 'category': 'Molecular function', 'name': 'Transferase'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97434.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18609.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599293.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013340.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NU98', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NU98', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'DIP', 'id': 'DIP-48321N', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'IntAct', 'id': 'Q8NU98', 'properties': [{'key': 'Interactions', 'value': '1'}]}, {'database': 'STRING', 'id': '196627.cg0057', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97434', 'properties': [{'key': 'ProteinId', 'value': 'BAB97434'}, {'key': 'GeneId', 'value': 'BAB97434'}]}, {'database': 'KEGG', 'id': 'cgb:cg0057', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0041', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.42', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0515', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'eggNOG', 'id': 'COG2815', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_000288_135_2_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'DPDYRYQ', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9762169at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'BRENDA', 'id': '2.7.11.1', 'properties': [{'key': 'OrganismId', 'value': '960'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005524', 'properties': [{'key': 'GoTerm', 'value': 'F:ATP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0106310', 'properties': [{'key': 'GoTerm', 'value': 'F:protein serine kinase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:RHEA'}]}, {'database': 'GO', 'id': 'GO:0004674', 'properties': [{'key': 'GoTerm', 'value': 'F:protein serine/threonine kinase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0006468', 'properties': [{'key': 'GoTerm', 'value': 'P:protein phosphorylation'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'CDD', 'id': 'cd06577', 'properties': [{'key': 'EntryName', 'value': 'PASTA_pknB'}, {'key': 'MatchStatus', 'value': '4'}]}, {'database': 'CDD', 'id': 'cd14014', 'properties': [{'key': 'EntryName', 'value': 'STKc_PknB_like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.30.10.20', 'properties': [{'key': 'EntryName', 'value': '-'}, {'key': 'MatchStatus', 'value': '4'}]}, {'database': 'Gene3D', 'id': '1.10.510.10', 'properties': [{'key': 'EntryName', 'value': 'Transferase(Phosphotransferase) domain 1'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR011009', 'properties': [{'key': 'EntryName', 'value': 'Kinase-like_dom_sf'}]}, {'database': 'InterPro', 'id': 'IPR005543', 'properties': [{'key': 'EntryName', 'value': 'PASTA_dom'}]}, {'database': 'InterPro', 'id': 'IPR000719', 'properties': [{'key': 'EntryName', 'value': 'Prot_kinase_dom'}]}, {'database': 'InterPro', 'id': 'IPR017441', 'properties': [{'key': 'EntryName', 'value': 'Protein_kinase_ATP_BS'}]}, {'database': 'InterPro', 'id': 'IPR008271', 'properties': [{'key': 'EntryName', 'value': 'Ser/Thr_kinase_AS'}]}, {'database': 'PANTHER', 'id': 'PTHR43289', 'properties': [{'key': 'EntryName', 'value': 'MITOGEN-ACTIVATED PROTEIN KINASE KINASE KINASE 20-RELATED'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR43289:SF32', 'properties': [{'key': 'EntryName', 'value': 'SERINE/THREONINE-PROTEIN KINASE PKAA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF03793', 'properties': [{'key': 'EntryName', 'value': 'PASTA'}, {'key': 'MatchStatus', 'value': '4'}]}, {'database': 'Pfam', 'id': 'PF00069', 'properties': [{'key': 'EntryName', 'value': 'Pkinase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SMART', 'id': 'SM00740', 'properties': [{'key': 'EntryName', 'value': 'PASTA'}, {'key': 'MatchStatus', 'value': '3'}]}, {'database': 'SMART', 'id': 'SM00220', 'properties': [{'key': 'EntryName', 'value': 'S_TKc'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF56112', 'properties': [{'key': 'EntryName', 'value': 'Protein kinase-like (PK-like)'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS51178', 'properties': [{'key': 'EntryName', 'value': 'PASTA'}, {'key': 'MatchStatus', 'value': '3'}]}, {'database': 'PROSITE', 'id': 'PS00107', 'properties': [{'key': 'EntryName', 'value': 'PROTEIN_KINASE_ATP'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS50011', 'properties': [{'key': 'EntryName', 'value': 'PROTEIN_KINASE_DOM'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS00108', 'properties': [{'key': 'EntryName', 'value': 'PROTEIN_KINASE_ST'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MTFVIADRYELDAVIGSGGMSEVFAATDTLIGREVAVKMLRIDLAKDPNFRERFRREAQNSGRLSHSSIVAVFDTGEVDKDGTSVPYIVMERVQGRNLREVVTEDGVFTPVEAANILIPVCEALQASHDAGIIHRDVKPANIMITNTGGVKVMDFGIARAVNDSTSAMTQTSAVIGTAQYLSPEQARGKPADARSDIYATGCVMYELVTGKPPFEGESPFAVAYQHVQEDPTPPSDFIADLTPTSAVNVDAVVLTAMAKHPADRYQTASEMAADLGRLSRNAVSHAARAHVETEETPEEPETRFSTRTSTQVAPAAGVAAASTGSGSSSRKRGSRGLTALAIVLSLGVVGVAGAFTYDYFANSSSTATSAIPNVEGLPQQEALTELQAAGFVVNIVEEASADVAEGLVIRANPSVGSEIRQGATVTITVSTGREMINIPDVSGMTLEDAARALEDVGLILNQNVREETSDDVESGLVIDQNPEAGQEVVVGSSVSLTMSSGTESIRVPNLTGMNWSQAEQNLISMGFNPTASYLDSSEPEGEVLSVSSQGTELPKGSSITVEVSNGMLIQAPDLARMSTEQAISALRAAGWTAPDQSLIVGDPIHTAALVDQNKIGFQSPTPATLFRKDAQVQVRLFEFDLAALVQ', 'length': 646, 'molWeight': 68237, 'crc64': '0F3B4980ECF73BFB', 'md5': '8FADA7478B3F1E4B66BDB93593BBF38B'}, 'extraAttributes': {'countByCommentType': {'CATALYTIC ACTIVITY': 2, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Domain': 5, 'Region': 1, 'Compositional bias': 2, 'Active site': 1, 'Binding site': 2}, 'uniParcId': 'UPI0000039FD6'}}}, {'from': 'CAF18610.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NU97', 'secondaryAccessions': ['Q8RQP0'], 'uniProtkbId': 'PKN2_CORGL', 'entryAudit': {'firstPublicDate': '2003-04-04', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 125, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'value': 'Serine/threonine-protein kinases PknA'}, 'ecNumbers': [{'value': '2.7.11.1'}]}}, 'genes': [{'geneName': {'value': 'pknA'}, 'synonyms': [{'value': 'drp72'}], 'orderedLocusNames': [{'value': 'Cgl0042'}, {'value': 'cg0059'}]}], 'comments': [{'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'ATP + L-seryl-[protein] = ADP + H(+) + O-phospho-L-seryl-[protein]', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:17989'}, {'database': 'Rhea', 'id': 'RHEA-COMP:9863'}, {'database': 'Rhea', 'id': 'RHEA-COMP:11604'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:29999'}, {'database': 'ChEBI', 'id': 'CHEBI:30616'}, {'database': 'ChEBI', 'id': 'CHEBI:83421'}, {'database': 'ChEBI', 'id': 'CHEBI:456216'}], 'ecNumber': '2.7.11.1'}}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'ATP + L-threonyl-[protein] = ADP + H(+) + O-phospho-L-threonyl-[protein]', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:46608'}, {'database': 'Rhea', 'id': 'RHEA-COMP:11060'}, {'database': 'Rhea', 'id': 'RHEA-COMP:11605'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:30013'}, {'database': 'ChEBI', 'id': 'CHEBI:30616'}, {'database': 'ChEBI', 'id': 'CHEBI:61977'}, {'database': 'ChEBI', 'id': 'CHEBI:456216'}], 'ecNumber': '2.7.11.1'}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}], 'value': 'Belongs to the protein kinase superfamily. Ser/Thr protein kinase family'}], 'commentType': 'SIMILARITY'}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Frameshift', 'sequence': 'BAB88662.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 469, 'modifier': 'EXACT'}}, 'description': 'Serine/threonine-protein kinases PknA', 'featureId': 'PRO_0000171199'}, {'type': 'Domain', 'location': {'start': {'value': 20, 'modifier': 'EXACT'}, 'end': {'value': 281, 'modifier': 'EXACT'}}, 'description': 'Protein kinase', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}]}, {'type': 'Region', 'location': {'start': {'value': 286, 'modifier': 'EXACT'}, 'end': {'value': 307, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Region', 'location': {'start': {'value': 359, 'modifier': 'EXACT'}, 'end': {'value': 454, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Compositional bias', 'location': {'start': {'value': 359, 'modifier': 'EXACT'}, 'end': {'value': 447, 'modifier': 'EXACT'}}, 'description': 'Polar residues', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Active site', 'location': {'start': {'value': 148, 'modifier': 'EXACT'}, 'end': {'value': 148, 'modifier': 'EXACT'}}, 'description': 'Proton acceptor', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}, {'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU10027'}]}, {'type': 'Binding site', 'location': {'start': {'value': 26, 'modifier': 'EXACT'}, 'end': {'value': 34, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}, {'type': 'Binding site', 'location': {'start': {'value': 49, 'modifier': 'EXACT'}, 'end': {'value': 49, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU00159'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 119, 'modifier': 'EXACT'}, 'end': {'value': 119, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAB88662', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'R', 'alternativeSequences': ['S']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 143, 'modifier': 'EXACT'}, 'end': {'value': 143, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAB88662', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'D', 'alternativeSequences': ['N']}}], 'keywords': [{'id': 'KW-0067', 'category': 'Ligand', 'name': 'ATP-binding'}, {'id': 'KW-0418', 'category': 'Molecular function', 'name': 'Kinase'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0723', 'category': 'Molecular function', 'name': 'Serine/threonine-protein kinase'}, {'id': 'KW-0808', 'category': 'Molecular function', 'name': 'Transferase'}], 'references': [{'citation': {'id': 'CI-E9KPG9AE813B9', 'citationType': 'submission', 'authors': ['Hirano S.', 'Kimura E.', 'Kawahara Y.', 'Sugimoto S.'], 'title': 'drp72 of Corynebacterium glutamicum.', 'publicationDate': 'APR-2002', 'submissionDatabase': 'EMBL/GenBank/DDBJ databases'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]']}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB083046', 'properties': [{'key': 'ProteinId', 'value': 'BAB88662.1'}, {'key': 'Status', 'value': 'ALT_FRAME'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97435.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18610.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599294.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013341.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NU97', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NU97', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'DIP', 'id': 'DIP-48320N', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'IntAct', 'id': 'Q8NU97', 'properties': [{'key': 'Interactions', 'value': '1'}]}, {'database': 'STRING', 'id': '196627.cg0059', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97435', 'properties': [{'key': 'ProteinId', 'value': 'BAB97435'}, {'key': 'GeneId', 'value': 'BAB97435'}]}, {'database': 'KEGG', 'id': 'cgb:cg0059', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0042', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.43', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0515', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_000288_63_44_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'YLCEHSV', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9762169at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'BRENDA', 'id': '2.7.11.1', 'properties': [{'key': 'OrganismId', 'value': '960'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005524', 'properties': [{'key': 'GoTerm', 'value': 'F:ATP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0106310', 'properties': [{'key': 'GoTerm', 'value': 'F:protein serine kinase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:RHEA'}]}, {'database': 'GO', 'id': 'GO:0004674', 'properties': [{'key': 'GoTerm', 'value': 'F:protein serine/threonine kinase activity'}, {'key': 'GoEvidenceType', 'value': 'IMP:CACAO'}], 'evidences': [{'evidenceCode': 'ECO:0000315', 'source': 'PubMed', 'id': '18974047'}]}, {'database': 'GO', 'id': 'GO:0016772', 'properties': [{'key': 'GoTerm', 'value': 'F:transferase activity, transferring phosphorus-containing groups'}, {'key': 'GoEvidenceType', 'value': 'IDA:CACAO'}], 'evidences': [{'evidenceCode': 'ECO:0000314', 'source': 'PubMed', 'id': '18442973'}]}, {'database': 'GO', 'id': 'GO:0006468', 'properties': [{'key': 'GoTerm', 'value': 'P:protein phosphorylation'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'CDD', 'id': 'cd14014', 'properties': [{'key': 'EntryName', 'value': 'STKc_PknB_like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '1.10.510.10', 'properties': [{'key': 'EntryName', 'value': 'Transferase(Phosphotransferase) domain 1'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR011009', 'properties': [{'key': 'EntryName', 'value': 'Kinase-like_dom_sf'}]}, {'database': 'InterPro', 'id': 'IPR000719', 'properties': [{'key': 'EntryName', 'value': 'Prot_kinase_dom'}]}, {'database': 'InterPro', 'id': 'IPR017441', 'properties': [{'key': 'EntryName', 'value': 'Protein_kinase_ATP_BS'}]}, {'database': 'InterPro', 'id': 'IPR008271', 'properties': [{'key': 'EntryName', 'value': 'Ser/Thr_kinase_AS'}]}, {'database': 'PANTHER', 'id': 'PTHR43289', 'properties': [{'key': 'EntryName', 'value': 'MITOGEN-ACTIVATED PROTEIN KINASE KINASE KINASE 20-RELATED'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR43289:SF32', 'properties': [{'key': 'EntryName', 'value': 'SERINE/THREONINE-PROTEIN KINASE PKAA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00069', 'properties': [{'key': 'EntryName', 'value': 'Pkinase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SMART', 'id': 'SM00220', 'properties': [{'key': 'EntryName', 'value': 'S_TKc'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF56112', 'properties': [{'key': 'EntryName', 'value': 'Protein kinase-like (PK-like)'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS00107', 'properties': [{'key': 'EntryName', 'value': 'PROTEIN_KINASE_ATP'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS50011', 'properties': [{'key': 'EntryName', 'value': 'PROTEIN_KINASE_DOM'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS00108', 'properties': [{'key': 'EntryName', 'value': 'PROTEIN_KINASE_ST'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSQEDITGKDRLQELIGADYRLQWIIGHGGMSTVWLADDVVNDREVAIKVLRPEFSDNQEFLNRFRNEAQAAENIDSEHVVATYDYREVPDPAGHTFCFIVMEFVRGESLADLLEREGRLPEDLALDVMEQAAHGLSVIHRMDMVHRDIKPGNMLITANGIVKITDFGIAKAAAAVPLTRTGMVVGTAQYVSPEQAQGKEVTAASDIYSLGVVGYEMMAGRRPFTGDSSVSVAIAHINQAPPQMPTSISAQTRELIGIALRKDPGRRFPDGNEMALAVSAVRLGKRPPQPRTSAMMAQAEAPSPSESTAMLGRVARPATITQEAAPKRGSGIGIGLFIAALLAVIIGAVIYAGTTGILFNDTPEETTTPETITETYTPTVEETTSQWVPPTPPTRSTFTEPETTSHRPTTSEESTSEEPTTEAPTSSRTVPQIPTSTPRTSASVPVETNAPADDLIDAVNGLLDVGGAQ', 'length': 469, 'molWeight': 50375, 'crc64': '53C10AD97BB95595', 'md5': '225DF4FA1C54A00458C7AB8A5869C85C'}, 'extraAttributes': {'countByCommentType': {'CATALYTIC ACTIVITY': 2, 'SIMILARITY': 1, 'SEQUENCE CAUTION': 1}, 'countByFeatureType': {'Chain': 1, 'Domain': 1, 'Region': 2, 'Compositional bias': 1, 'Active site': 1, 'Binding site': 2, 'Sequence conflict': 2}, 'uniParcId': 'UPI0000039FD7'}}}, {'from': 'CAF18640.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'P46396', 'secondaryAccessions': ['P94636'], 'uniProtkbId': 'BIOB_CORGL', 'entryAudit': {'firstPublicDate': '1995-11-01', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-07-11', 'entryVersion': 151, 'sequenceVersion': 3}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Biotin synthase'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': '2.8.1.6'}]}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'bioB'}, 'orderedLocusNames': [{'value': 'Cgl0072'}, {'value': 'cg0095'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Catalyzes the conversion of dethiobiotin (DTB) to biotin by the insertion of a sulfur atom into dethiobiotin via a radical-based mechanism'}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': \"(4R,5S)-dethiobiotin + [sulfur carrier]-SH + 2 reduced [2Fe-2S]-[ferredoxin] + 2 S-adenosyl-L-methionine = 2 5'-deoxyadenosine + [sulfur carrier]-H + biotin + 2 L-methionine + 2 oxidized [2Fe-2S]-[ferredoxin]\", 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:22060'}, {'database': 'Rhea', 'id': 'RHEA-COMP:10000'}, {'database': 'Rhea', 'id': 'RHEA-COMP:10001'}, {'database': 'Rhea', 'id': 'RHEA-COMP:14737'}, {'database': 'Rhea', 'id': 'RHEA-COMP:14739'}, {'database': 'ChEBI', 'id': 'CHEBI:17319'}, {'database': 'ChEBI', 'id': 'CHEBI:29917'}, {'database': 'ChEBI', 'id': 'CHEBI:33737'}, {'database': 'ChEBI', 'id': 'CHEBI:33738'}, {'database': 'ChEBI', 'id': 'CHEBI:57586'}, {'database': 'ChEBI', 'id': 'CHEBI:57844'}, {'database': 'ChEBI', 'id': 'CHEBI:59789'}, {'database': 'ChEBI', 'id': 'CHEBI:64428'}, {'database': 'ChEBI', 'id': 'CHEBI:149473'}], 'ecNumber': '2.8.1.6', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}]}}, {'commentType': 'COFACTOR', 'cofactors': [{'name': '[4Fe-4S] cluster', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:49883'}}], 'note': {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Binds 1 [4Fe-4S] cluster. The cluster is coordinated with 3 cysteines and an exchangeable S-adenosyl-L-methionine.'}]}}, {'commentType': 'COFACTOR', 'cofactors': [{'name': '[2Fe-2S] cluster', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:190135'}}], 'note': {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Binds 1 [2Fe-2S] cluster. The cluster is coordinated with 3 cysteines and 1 arginine.'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Cofactor biosynthesis; biotin biosynthesis; biotin from 7,8-diaminononanoate: step 2/2'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Homodimer'}], 'commentType': 'SUBUNIT'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'value': 'Belongs to the radical SAM superfamily. Biotin synthase family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 334, 'modifier': 'EXACT'}}, 'description': 'Biotin synthase', 'featureId': 'PRO_0000185551'}, {'type': 'Domain', 'location': {'start': {'value': 55, 'modifier': 'EXACT'}, 'end': {'value': 280, 'modifier': 'EXACT'}}, 'description': 'Radical SAM core', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'PROSITE-ProRule', 'id': 'PRU01266'}]}, {'type': 'Binding site', 'location': {'start': {'value': 70, 'modifier': 'EXACT'}, 'end': {'value': 70, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49883'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'ligand': {'name': '[4Fe-4S] cluster', 'id': 'ChEBI:CHEBI:49883', 'note': '4Fe-4S-S-AdoMet'}}, {'type': 'Binding site', 'location': {'start': {'value': 74, 'modifier': 'EXACT'}, 'end': {'value': 74, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49883'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'ligand': {'name': '[4Fe-4S] cluster', 'id': 'ChEBI:CHEBI:49883', 'note': '4Fe-4S-S-AdoMet'}}, {'type': 'Binding site', 'location': {'start': {'value': 77, 'modifier': 'EXACT'}, 'end': {'value': 77, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49883'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'ligand': {'name': '[4Fe-4S] cluster', 'id': 'ChEBI:CHEBI:49883', 'note': '4Fe-4S-S-AdoMet'}}, {'type': 'Binding site', 'location': {'start': {'value': 113, 'modifier': 'EXACT'}, 'end': {'value': 113, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:190135'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'ligand': {'name': '[2Fe-2S] cluster', 'id': 'ChEBI:CHEBI:190135'}}, {'type': 'Binding site', 'location': {'start': {'value': 205, 'modifier': 'EXACT'}, 'end': {'value': 205, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:190135'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'ligand': {'name': '[2Fe-2S] cluster', 'id': 'ChEBI:CHEBI:190135'}}, {'type': 'Binding site', 'location': {'start': {'value': 275, 'modifier': 'EXACT'}, 'end': {'value': 275, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:190135'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01694'}], 'ligand': {'name': '[2Fe-2S] cluster', 'id': 'ChEBI:CHEBI:190135'}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 5, 'modifier': 'EXACT'}, 'end': {'value': 5, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1 and 2', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'G', 'alternativeSequences': ['A']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 88, 'modifier': 'EXACT'}, 'end': {'value': 88, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA03169', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'R', 'alternativeSequences': ['A']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 113, 'modifier': 'EXACT'}, 'end': {'value': 114, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA03169', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'CI', 'alternativeSequences': ['DF']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 232, 'modifier': 'EXACT'}, 'end': {'value': 232, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1 and 2', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'H', 'alternativeSequences': ['D']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 260, 'modifier': 'EXACT'}, 'end': {'value': 260, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA03169', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'R', 'alternativeSequences': ['A']}}], 'keywords': [{'id': 'KW-0001', 'category': 'Ligand', 'name': '2Fe-2S'}, {'id': 'KW-0004', 'category': 'Ligand', 'name': '4Fe-4S'}, {'id': 'KW-0093', 'category': 'Biological process', 'name': 'Biotin biosynthesis'}, {'id': 'KW-0408', 'category': 'Ligand', 'name': 'Iron'}, {'id': 'KW-0411', 'category': 'Ligand', 'name': 'Iron-sulfur'}, {'id': 'KW-0479', 'category': 'Ligand', 'name': 'Metal-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0949', 'category': 'Ligand', 'name': 'S-adenosyl-L-methionine'}, {'id': 'KW-0808', 'category': 'Molecular function', 'name': 'Transferase'}], 'references': [{'citation': {'id': '8173080', 'citationType': 'journal article', 'authors': ['Hatakeyama K.', 'Kohama K.', 'Vertes A.A.', 'Kobayashi M.', 'Kurusu Y.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '8173080'}, {'database': 'DOI', 'id': '10.3109/10425179309020147'}], 'title': 'Analysis of the biotin biosynthesis pathway in coryneform bacteria: cloning and sequencing of the bioB gene from Brevibacterium flavum.', 'publicationDate': '1993', 'journal': 'DNA Seq.', 'firstPage': '87', 'lastPage': '93', 'volume': '4'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]'], 'referenceComments': [{'value': 'MJ233', 'type': 'STRAIN'}]}, {'citation': {'id': '8917070', 'citationType': 'journal article', 'authors': ['Serebriiskii I.G.', 'Vassin V.M.', 'Tsygankov Y.D.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '8917070'}, {'database': 'DOI', 'id': '10.1016/0378-1119(96)00114-x'}], 'title': 'Two new members of the bio B superfamily: cloning, sequencing and expression of bio B genes of Methylobacillus flagellatum and Corynebacterium glutamicum.', 'publicationDate': '1996', 'journal': 'Gene', 'firstPage': '15', 'lastPage': '22', 'volume': '175'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'D14084', 'properties': [{'key': 'ProteinId', 'value': 'BAA03169.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'U31281', 'properties': [{'key': 'ProteinId', 'value': 'AAC44580.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97465.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18640.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'PIR', 'id': 'I40338', 'properties': [{'key': 'EntryName', 'value': 'I40338'}]}, {'database': 'PIR', 'id': 'JC5084', 'properties': [{'key': 'EntryName', 'value': 'JC5084'}]}, {'database': 'RefSeq', 'id': 'NP_599324.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003861135.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'P46396', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'P46396', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0095', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97465', 'properties': [{'key': 'ProteinId', 'value': 'BAB97465'}, {'key': 'GeneId', 'value': 'BAB97465'}]}, {'database': 'KEGG', 'id': 'cgb:cg0095', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0072', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.73', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0502', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_033172_2_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'AMGVHRY', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9786826at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'BRENDA', 'id': '2.8.1.6', 'properties': [{'key': 'OrganismId', 'value': '960'}]}, {'database': 'UniPathway', 'id': 'UPA00078', 'properties': [{'key': 'RectionId', 'value': 'UER00162'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0051537', 'properties': [{'key': 'GoTerm', 'value': 'F:2 iron, 2 sulfur cluster binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0051539', 'properties': [{'key': 'GoTerm', 'value': 'F:4 iron, 4 sulfur cluster binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0004076', 'properties': [{'key': 'GoTerm', 'value': 'F:biotin synthase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0005506', 'properties': [{'key': 'GoTerm', 'value': 'F:iron ion binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0009102', 'properties': [{'key': 'GoTerm', 'value': 'P:biotin biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd01335', 'properties': [{'key': 'EntryName', 'value': 'Radical_SAM'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.20.20.70', 'properties': [{'key': 'EntryName', 'value': 'Aldolase class I'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01694', 'properties': [{'key': 'EntryName', 'value': 'BioB'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR013785', 'properties': [{'key': 'EntryName', 'value': 'Aldolase_TIM'}]}, {'database': 'InterPro', 'id': 'IPR010722', 'properties': [{'key': 'EntryName', 'value': 'BATS_dom'}]}, {'database': 'InterPro', 'id': 'IPR002684', 'properties': [{'key': 'EntryName', 'value': 'Biotin_synth/BioAB'}]}, {'database': 'InterPro', 'id': 'IPR024177', 'properties': [{'key': 'EntryName', 'value': 'Biotin_synthase'}]}, {'database': 'InterPro', 'id': 'IPR006638', 'properties': [{'key': 'EntryName', 'value': 'Elp3/MiaA/NifB-like_rSAM'}]}, {'database': 'InterPro', 'id': 'IPR007197', 'properties': [{'key': 'EntryName', 'value': 'rSAM'}]}, {'database': 'PANTHER', 'id': 'PTHR22976', 'properties': [{'key': 'EntryName', 'value': 'BIOTIN SYNTHASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR22976:SF2', 'properties': [{'key': 'EntryName', 'value': 'BIOTIN SYNTHASE, MITOCHONDRIAL'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF06968', 'properties': [{'key': 'EntryName', 'value': 'BATS'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF04055', 'properties': [{'key': 'EntryName', 'value': 'Radical_SAM'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF001619', 'properties': [{'key': 'EntryName', 'value': 'Biotin_synth'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SFLD', 'id': 'SFLDG01082', 'properties': [{'key': 'EntryName', 'value': 'B12-binding_domain_containing'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SFLD', 'id': 'SFLDG01060', 'properties': [{'key': 'EntryName', 'value': 'BATS_domain_containing'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SFLD', 'id': 'SFLDG01278', 'properties': [{'key': 'EntryName', 'value': 'biotin_synthase_like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SMART', 'id': 'SM00876', 'properties': [{'key': 'EntryName', 'value': 'BATS'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SMART', 'id': 'SM00729', 'properties': [{'key': 'EntryName', 'value': 'Elp3'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF102114', 'properties': [{'key': 'EntryName', 'value': 'Radical SAM enzymes'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00433', 'properties': [{'key': 'EntryName', 'value': 'bioB'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS51918', 'properties': [{'key': 'EntryName', 'value': 'RADICAL_SAM'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MTIPGTILDTARTQVLEQGIGLNQQQLMEVLTLPEEQIPDLMELAHQVRLKWCGEEIEVEGIISLKTGGCPEDCHFCSQSGLFESPVRSVWLDIPNLVEAAKQTAKTGATEFCIVAAVKGPDERLMTQLEEAVLAIHSEVEIEVAASIGTLNKEQVDRLAAAGVHRYNHNLETARSYFPEVVTTHTWEERRETLRLVAEAGMEVCSGGILGMGETLEQRAEFAVQLAELDPHEVPMNFLDPRPGTPFADRELMDSRDALRSIGAFRLAMPHTMLRFAGGRELTLGDKGSEQALLGGINAMIVGNYLTTLGRPMEDDLDMMDRLQLPIKVLNKVI', 'length': 334, 'molWeight': 36856, 'crc64': 'E0EF4E812C6BF096', 'md5': 'B44B1045760B912A10C6D8B898336DB0'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'COFACTOR': 2, 'PATHWAY': 1, 'SUBUNIT': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Domain': 1, 'Binding site': 6, 'Sequence conflict': 5}, 'uniParcId': 'UPI0000036476'}}}, {'from': 'CAF18652.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q9RHM6', 'uniProtkbId': 'URE3_CORGL', 'entryAudit': {'firstPublicDate': '2003-11-28', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2000-05-01', 'entryVersion': 123, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'Urease subunit gamma'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': '3.5.1.5'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'Urea amidohydrolase subunit gamma'}}]}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'ureA'}, 'orderedLocusNames': [{'value': 'Cgl0084'}, {'value': 'cg0113'}]}], 'comments': [{'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': '2 H(+) + H2O + urea = CO2 + 2 NH4(+)', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:20557'}, {'database': 'ChEBI', 'id': 'CHEBI:15377'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:16199'}, {'database': 'ChEBI', 'id': 'CHEBI:16526'}, {'database': 'ChEBI', 'id': 'CHEBI:28938'}], 'ecNumber': '3.5.1.5', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'Nitrogen metabolism; urea degradation; CO(2) and NH(3) from urea (urease route): step 1/1'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'Heterotrimer of UreA (gamma), UreB (beta) and UreC (alpha) subunits. Three heterotrimers associate to form the active enzyme'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}, {'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '15516578'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00739'}], 'value': 'Belongs to the urease gamma subunit family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 100, 'modifier': 'EXACT'}}, 'description': 'Urease subunit gamma', 'featureId': 'PRO_0000098010'}], 'keywords': [{'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0378', 'category': 'Molecular function', 'name': 'Hydrolase'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'OPERON STRUCTURE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '15516578', 'citationType': 'journal article', 'authors': ['Beckers G.', 'Bendt A.K.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '15516578'}, {'database': 'DOI', 'id': '10.1128/jb.186.22.7645-7652.2004'}], 'title': 'Molecular identification of the urea uptake system and transcriptional analysis of urea transporter- and urease-encoding genes in Corynebacterium glutamicum.', 'publicationDate': '2004', 'journal': 'J. Bacteriol.', 'firstPage': '7645', 'lastPage': '7652', 'volume': '186'}, 'referencePositions': ['INDUCTION', 'IDENTIFICATION BY MASS SPECTROMETRY']}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81935.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88552.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97477.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18652.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599336.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003857708.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q9RHM6', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q9RHM6', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0113', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97477', 'properties': [{'key': 'ProteinId', 'value': 'BAB97477'}, {'key': 'GeneId', 'value': 'BAB97477'}]}, {'database': 'GeneID', 'id': '69620493', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgb:cg0113', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0084', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.85', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0831', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_145825_1_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'MQLTPHE', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9797217at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'UniPathway', 'id': 'UPA00258', 'properties': [{'key': 'RectionId', 'value': 'UER00370'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0016151', 'properties': [{'key': 'GoTerm', 'value': 'F:nickel cation binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0009039', 'properties': [{'key': 'GoTerm', 'value': 'F:urease activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0043419', 'properties': [{'key': 'GoTerm', 'value': 'P:urea catabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniPathway'}]}, {'database': 'CDD', 'id': 'cd00390', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.30.280.10', 'properties': [{'key': 'EntryName', 'value': 'Urease, gamma-like subunit'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00739', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR012010', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma'}]}, {'database': 'InterPro', 'id': 'IPR002026', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma/gamma-beta_su'}]}, {'database': 'InterPro', 'id': 'IPR036463', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma_sf'}]}, {'database': 'PANTHER', 'id': 'PTHR33569', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR33569:SF1', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00547', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF001223', 'properties': [{'key': 'EntryName', 'value': 'Urease_gamma'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF54111', 'properties': [{'key': 'EntryName', 'value': 'Urease, gamma-subunit'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00193', 'properties': [{'key': 'EntryName', 'value': 'urease_gam'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MHITPREQEKLMIVVAADLARRRKDRGLKLNHPEAVALITYELIEGARDGRTVADLMSWGSTILTRDDVLEGIPEMIPDIQVEATFDDGTKLVTVHNPIR', 'length': 100, 'molWeight': 11245, 'crc64': 'A48F4DC0EABD9567', 'md5': '27326002340D50917DE1CE9819AFFCE0'}, 'extraAttributes': {'countByCommentType': {'CATALYTIC ACTIVITY': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1}, 'uniParcId': 'UPI000003A004'}}}, {'from': 'CAF18653.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q79VJ4', 'secondaryAccessions': ['Q9L421', 'Q9RHM5'], 'uniProtkbId': 'URE2_CORGL', 'entryAudit': {'firstPublicDate': '2006-05-02', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 110, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'Urease subunit beta'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': '3.5.1.5'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'Urea amidohydrolase subunit beta'}}]}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'ureB'}, 'orderedLocusNames': [{'value': 'Cgl0085'}, {'value': 'cg0114'}]}], 'comments': [{'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': '2 H(+) + H2O + urea = CO2 + 2 NH4(+)', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:20557'}, {'database': 'ChEBI', 'id': 'CHEBI:15377'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:16199'}, {'database': 'ChEBI', 'id': 'CHEBI:16526'}, {'database': 'ChEBI', 'id': 'CHEBI:28938'}], 'ecNumber': '3.5.1.5', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'Nitrogen metabolism; urea degradation; CO(2) and NH(3) from urea (urease route): step 1/1'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'Heterotrimer of UreA (gamma), UreB (beta) and UreC (alpha) subunits. Three heterotrimers associate to form the active enzyme'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}, {'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '15516578'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01954'}], 'value': 'Belongs to the urease beta subunit family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 162, 'modifier': 'EXACT'}}, 'description': 'Urease subunit beta', 'featureId': 'PRO_0000234246'}, {'type': 'Region', 'location': {'start': {'value': 116, 'modifier': 'EXACT'}, 'end': {'value': 162, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Compositional bias', 'location': {'start': {'value': 133, 'modifier': 'EXACT'}, 'end': {'value': 147, 'modifier': 'EXACT'}}, 'description': 'Basic and acidic residues', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Sequence conflict', 'location': {'start': {'value': 103, 'modifier': 'EXACT'}, 'end': {'value': 105, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88553', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'DLK', 'alternativeSequences': ['TQ']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 110, 'modifier': 'EXACT'}, 'end': {'value': 131, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88553', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'EGREDGWRRSSAAGDAPQELPQ', 'alternativeSequences': ['RTEDDGVVLSCCDVHKIAT']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 138, 'modifier': 'EXACT'}, 'end': {'value': 138, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88553', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'G', 'alternativeSequences': ['A']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 147, 'modifier': 'EXACT'}, 'end': {'value': 148, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88553', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'VD', 'alternativeSequences': ['AN']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 157, 'modifier': 'EXACT'}, 'end': {'value': 160, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88553', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {}}], 'keywords': [{'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0378', 'category': 'Molecular function', 'name': 'Hydrolase'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'OPERON STRUCTURE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '15516578', 'citationType': 'journal article', 'authors': ['Beckers G.', 'Bendt A.K.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '15516578'}, {'database': 'DOI', 'id': '10.1128/jb.186.22.7645-7652.2004'}], 'title': 'Molecular identification of the urea uptake system and transcriptional analysis of urea transporter- and urease-encoding genes in Corynebacterium glutamicum.', 'publicationDate': '2004', 'journal': 'J. Bacteriol.', 'firstPage': '7645', 'lastPage': '7652', 'volume': '186'}, 'referencePositions': ['INDUCTION', 'IDENTIFICATION BY MASS SPECTROMETRY']}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88553.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81936.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97478.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18653.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599337.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013377.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q79VJ4', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q79VJ4', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0114', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97478', 'properties': [{'key': 'ProteinId', 'value': 'BAB97478'}, {'key': 'GeneId', 'value': 'BAB97478'}]}, {'database': 'KEGG', 'id': 'cgb:cg0114', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0085', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.86', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0832', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_129707_1_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'FYEVNDA', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9797217at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'UniPathway', 'id': 'UPA00258', 'properties': [{'key': 'RectionId', 'value': 'UER00370'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0035550', 'properties': [{'key': 'GoTerm', 'value': 'C:urease complex'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0009039', 'properties': [{'key': 'GoTerm', 'value': 'F:urease activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0043419', 'properties': [{'key': 'GoTerm', 'value': 'P:urea catabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniPathway'}]}, {'database': 'CDD', 'id': 'cd00407', 'properties': [{'key': 'EntryName', 'value': 'Urease_beta'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '2.10.150.10', 'properties': [{'key': 'EntryName', 'value': 'Urease, beta subunit'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01954', 'properties': [{'key': 'EntryName', 'value': 'Urease_beta'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR002019', 'properties': [{'key': 'EntryName', 'value': 'Urease_beta-like'}]}, {'database': 'InterPro', 'id': 'IPR036461', 'properties': [{'key': 'EntryName', 'value': 'Urease_betasu_sf'}]}, {'database': 'PANTHER', 'id': 'PTHR33569', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR33569:SF1', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00699', 'properties': [{'key': 'EntryName', 'value': 'Urease_beta'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF51278', 'properties': [{'key': 'EntryName', 'value': 'Urease, beta-subunit'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00192', 'properties': [{'key': 'EntryName', 'value': 'urease_beta'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MIPGEYILSSESLTGNVGREAKTIEIINTGDRPVQIGSHFHFAEVNPSISFDRSEGYGFRLDIPSGTAVRLEPGDARTVNLVAIGGDRIVAGFRDLVDGPLEDLKVNVWEGREDGWRRSSAAGDAPQELPQVEAAERGRKLDDATDVDTNVGTEEGFEEGRN', 'length': 162, 'molWeight': 17568, 'crc64': 'FAB0C50E993F8C25', 'md5': '890C4B75E13A026AC7A79D789201C111'}, 'extraAttributes': {'countByCommentType': {'CATALYTIC ACTIVITY': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Region': 1, 'Compositional bias': 1, 'Sequence conflict': 5}, 'uniParcId': 'UPI00000366A2'}}}, {'from': 'CAF18654.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q79VJ3', 'secondaryAccessions': ['Q9L420', 'Q9RHM4'], 'uniProtkbId': 'URE1_CORGL', 'entryAudit': {'firstPublicDate': '2006-05-02', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 121, 'sequenceVersion': 1}, 'annotationScore': 4.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Urease subunit alpha'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': '3.5.1.5'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Urea amidohydrolase subunit alpha'}}]}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'ureC'}, 'orderedLocusNames': [{'value': 'Cgl0086'}, {'value': 'cg0115'}]}], 'comments': [{'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': '2 H(+) + H2O + urea = CO2 + 2 NH4(+)', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:20557'}, {'database': 'ChEBI', 'id': 'CHEBI:15377'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:16199'}, {'database': 'ChEBI', 'id': 'CHEBI:16526'}, {'database': 'ChEBI', 'id': 'CHEBI:28938'}], 'ecNumber': '3.5.1.5', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}]}}, {'commentType': 'COFACTOR', 'cofactors': [{'name': 'Ni cation', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:25516'}}], 'note': {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Binds 2 nickel ions per subunit.'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Nitrogen metabolism; urea degradation; CO(2) and NH(3) from urea (urease route): step 1/1'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Heterotrimer of UreA (gamma), UreB (beta) and UreC (alpha) subunits. Three heterotrimers associate to form the active enzyme'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}, {'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '15516578'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Carboxylation allows a single lysine to coordinate two nickel ions'}], 'commentType': 'PTM'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '10930756'}, {'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}], 'value': 'Cells are urease negative and are no longer able to transport urea'}], 'commentType': 'DISRUPTION PHENOTYPE'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'value': 'Belongs to the metallo-dependent hydrolases superfamily. Urease alpha subunit family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 570, 'modifier': 'EXACT'}}, 'description': 'Urease subunit alpha', 'featureId': 'PRO_0000234152'}, {'type': 'Domain', 'location': {'start': {'value': 132, 'modifier': 'EXACT'}, 'end': {'value': 570, 'modifier': 'EXACT'}}, 'description': 'Urease', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}]}, {'type': 'Active site', 'location': {'start': {'value': 323, 'modifier': 'EXACT'}, 'end': {'value': 323, 'modifier': 'EXACT'}}, 'description': 'Proton donor', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}]}, {'type': 'Binding site', 'location': {'start': {'value': 137, 'modifier': 'EXACT'}, 'end': {'value': 137, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '1'}}, {'type': 'Binding site', 'location': {'start': {'value': 139, 'modifier': 'EXACT'}, 'end': {'value': 139, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '1'}}, {'type': 'Binding site', 'location': {'start': {'value': 220, 'modifier': 'EXACT'}, 'end': {'value': 220, 'modifier': 'EXACT'}}, 'description': 'via carbamate group', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '1'}}, {'type': 'Binding site', 'location': {'start': {'value': 220, 'modifier': 'EXACT'}, 'end': {'value': 220, 'modifier': 'EXACT'}}, 'description': 'via carbamate group', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '2'}}, {'type': 'Binding site', 'location': {'start': {'value': 222, 'modifier': 'EXACT'}, 'end': {'value': 222, 'modifier': 'EXACT'}}, 'description': '', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'substrate'}}, {'type': 'Binding site', 'location': {'start': {'value': 249, 'modifier': 'EXACT'}, 'end': {'value': 249, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '2'}}, {'type': 'Binding site', 'location': {'start': {'value': 275, 'modifier': 'EXACT'}, 'end': {'value': 275, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '2'}}, {'type': 'Binding site', 'location': {'start': {'value': 363, 'modifier': 'EXACT'}, 'end': {'value': 363, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:49786'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}], 'ligand': {'name': 'Ni(2+)', 'id': 'ChEBI:CHEBI:49786', 'label': '1'}}, {'type': 'Modified residue', 'location': {'start': {'value': 220, 'modifier': 'EXACT'}, 'end': {'value': 220, 'modifier': 'EXACT'}}, 'description': 'N6-carboxylysine', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01953'}]}, {'type': 'Sequence conflict', 'location': {'start': {'value': 40, 'modifier': 'EXACT'}, 'end': {'value': 40, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'I', 'alternativeSequences': ['V']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 133, 'modifier': 'EXACT'}, 'end': {'value': 134, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'GI', 'alternativeSequences': ['AL']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 206, 'modifier': 'EXACT'}, 'end': {'value': 206, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'S', 'alternativeSequences': ['A']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 259, 'modifier': 'EXACT'}, 'end': {'value': 263, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'VEDTI', 'alternativeSequences': ['CGRH']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 269, 'modifier': 'EXACT'}, 'end': {'value': 269, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'R', 'alternativeSequences': ['G']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 275, 'modifier': 'EXACT'}, 'end': {'value': 275, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'H', 'alternativeSequences': ['N']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 286, 'modifier': 'EXACT'}, 'end': {'value': 286, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'D', 'alternativeSequences': ['E']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 289, 'modifier': 'EXACT'}, 'end': {'value': 289, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'R', 'alternativeSequences': ['G']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 292, 'modifier': 'EXACT'}, 'end': {'value': 292, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'A', 'alternativeSequences': ['G']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 303, 'modifier': 'EXACT'}, 'end': {'value': 307, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'PTLPY', 'alternativeSequences': ['QRFRN']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 436, 'modifier': 'EXACT'}, 'end': {'value': 436, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'V', 'alternativeSequences': ['L']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 516, 'modifier': 'EXACT'}, 'end': {'value': 516, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88554', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'G', 'alternativeSequences': ['D']}}], 'keywords': [{'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0378', 'category': 'Molecular function', 'name': 'Hydrolase'}, {'id': 'KW-0479', 'category': 'Ligand', 'name': 'Metal-binding'}, {'id': 'KW-0533', 'category': 'Ligand', 'name': 'Nickel'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'DISRUPTION PHENOTYPE', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'OPERON STRUCTURE', 'DISRUPTION PHENOTYPE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '15516578', 'citationType': 'journal article', 'authors': ['Beckers G.', 'Bendt A.K.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '15516578'}, {'database': 'DOI', 'id': '10.1128/jb.186.22.7645-7652.2004'}], 'title': 'Molecular identification of the urea uptake system and transcriptional analysis of urea transporter- and urease-encoding genes in Corynebacterium glutamicum.', 'publicationDate': '2004', 'journal': 'J. Bacteriol.', 'firstPage': '7645', 'lastPage': '7652', 'volume': '186'}, 'referencePositions': ['INDUCTION', 'IDENTIFICATION BY MASS SPECTROMETRY']}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88554.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81937.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97479.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18654.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599338.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013378.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q79VJ3', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q79VJ3', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0115', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'MEROPS', 'id': 'M38.982', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97479', 'properties': [{'key': 'ProteinId', 'value': 'BAB97479'}, {'key': 'GeneId', 'value': 'BAB97479'}]}, {'database': 'KEGG', 'id': 'cgb:cg0115', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0086', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.87', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0804', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_000980_0_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'DIVLWPI', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9802793at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'UniPathway', 'id': 'UPA00258', 'properties': [{'key': 'RectionId', 'value': 'UER00370'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0016151', 'properties': [{'key': 'GoTerm', 'value': 'F:nickel cation binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0009039', 'properties': [{'key': 'GoTerm', 'value': 'F:urease activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0043419', 'properties': [{'key': 'GoTerm', 'value': 'P:urea catabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniPathway'}]}, {'database': 'CDD', 'id': 'cd00375', 'properties': [{'key': 'EntryName', 'value': 'Urease_alpha'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.20.20.140', 'properties': [{'key': 'EntryName', 'value': 'Metal-dependent hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '2.30.40.10', 'properties': [{'key': 'EntryName', 'value': 'Urease, subunit C, domain 1'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01953', 'properties': [{'key': 'EntryName', 'value': 'Urease_alpha'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR006680', 'properties': [{'key': 'EntryName', 'value': 'Amidohydro-rel'}]}, {'database': 'InterPro', 'id': 'IPR011059', 'properties': [{'key': 'EntryName', 'value': 'Metal-dep_hydrolase_composite'}]}, {'database': 'InterPro', 'id': 'IPR032466', 'properties': [{'key': 'EntryName', 'value': 'Metal_Hydrolase'}]}, {'database': 'InterPro', 'id': 'IPR011612', 'properties': [{'key': 'EntryName', 'value': 'Urease_alpha_N_dom'}]}, {'database': 'InterPro', 'id': 'IPR005848', 'properties': [{'key': 'EntryName', 'value': 'Urease_asu'}]}, {'database': 'InterPro', 'id': 'IPR017951', 'properties': [{'key': 'EntryName', 'value': 'Urease_asu_c'}]}, {'database': 'PANTHER', 'id': 'PTHR43440', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR43440:SF1', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF01979', 'properties': [{'key': 'EntryName', 'value': 'Amidohydro_1'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00449', 'properties': [{'key': 'EntryName', 'value': 'Urease_alpha'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PRINTS', 'id': 'PR01752', 'properties': [{'key': 'EntryName', 'value': 'UREASE'}]}, {'database': 'SUPFAM', 'id': 'SSF51338', 'properties': [{'key': 'EntryName', 'value': 'Composite domain of metallo-dependent hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF51556', 'properties': [{'key': 'EntryName', 'value': 'Metallo-dependent hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR01792', 'properties': [{'key': 'EntryName', 'value': 'urease_alph'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS51368', 'properties': [{'key': 'EntryName', 'value': 'UREASE_3'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSFEISRKQYTDLYGPTVGDSVRLADTELFLCVEKDYAAIGEEVAFGGGKVIRDGMGQNGTLVRDVDIPDTVITNVIVLDYTGVYKADVALRDGKIFRIGKAGNPNVMENVDIVIGVATDIIAGEGKILTAGGIDTHVHFLGTDQVNTALASGITTMIGGGTGPSQASMATTVTPGQWNTYNMLSAFEGMPMNFGILGKGHGSSKSPLAEQVRAGAIGLKIHEDWGATPSSINTALEVADDMDIQVALHSDTLNEAGFVEDTIEAIAGRVIHTFHTEGAGGGHAPDLIRVAALPNVLPASTNPTLPYTRNTVEEHLDMVMVAHHLNPDIPEDVAFADSRIRAETIAAEDVLHDMGIFSITSSDSQAMGRVGETITRTWQVADHMKRTRGSLTGDAPYNDNNRLRRFIAKYTINPAIAHGVDYVVGSVEEGKFADLVLWDPKFFGVKPDLVIKGGLMVNSLMGDSNGSIPTPQPRTLRNTWGAFGQAVSRSSITFLSQDAIDANVPDLLNLRKQIRGVRGVRNLTKRDMKLNAEMPDIRVDPETYQVFVNGELITSKPAETVPMARRYFLF', 'length': 570, 'molWeight': 61452, 'crc64': '2C5F5D9F7F5FBB2E', 'md5': 'ED394867F8CFDC10C1786570EA281B75'}, 'extraAttributes': {'countByCommentType': {'CATALYTIC ACTIVITY': 1, 'COFACTOR': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'PTM': 1, 'DISRUPTION PHENOTYPE': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Domain': 1, 'Active site': 1, 'Binding site': 8, 'Modified residue': 1, 'Sequence conflict': 12}, 'uniParcId': 'UPI000003A005'}}}, {'from': 'CAF18655.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q9RHM3', 'secondaryAccessions': ['Q9L419'], 'uniProtkbId': 'UREE_CORGL', 'entryAudit': {'firstPublicDate': '2000-12-01', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-08-02', 'entryVersion': 120, 'sequenceVersion': 2}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '2: Evidence at transcript level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00822'}], 'value': 'Urease accessory protein UreE'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00822'}], 'value': 'ureE'}, 'orderedLocusNames': [{'value': 'Cgl0087'}, {'value': 'cg0116'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00822'}], 'value': 'Involved in urease metallocenter assembly. Binds nickel. Probably functions as a nickel donor during metallocenter assembly'}], 'commentType': 'FUNCTION'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00822'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00822'}], 'value': 'Belongs to the UreE family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 157, 'modifier': 'EXACT'}}, 'description': 'Urease accessory protein UreE', 'featureId': 'PRO_0000067631'}, {'type': 'Sequence conflict', 'location': {'start': {'value': 42, 'modifier': 'EXACT'}, 'end': {'value': 42, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88555', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'P', 'alternativeSequences': ['S']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 52, 'modifier': 'EXACT'}, 'end': {'value': 57, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88555', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'NHGHPI', 'alternativeSequences': ['ITGTPS']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 69, 'modifier': 'EXACT'}, 'end': {'value': 69, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88555', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'K', 'alternativeSequences': ['R']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 133, 'modifier': 'EXACT'}, 'end': {'value': 157, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'DEHGIEYQRTELVPPIPFRHSGHTH', 'alternativeSequences': ['G']}}], 'keywords': [{'id': 'KW-0143', 'category': 'Molecular function', 'name': 'Chaperone'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0533', 'category': 'Ligand', 'name': 'Nickel'}, {'id': 'KW-0996', 'category': 'Biological process', 'name': 'Nickel insertion'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'OPERON STRUCTURE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88555.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81938.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97480.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18655.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599339.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013379.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q9RHM3', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q9RHM3', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0116', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97480', 'properties': [{'key': 'ProteinId', 'value': 'BAB97480'}, {'key': 'GeneId', 'value': 'BAB97480'}]}, {'database': 'KEGG', 'id': 'cgb:cg0116', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0087', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.88', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG2371', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_093757_3_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'YVDLEWF', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9810882at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0016151', 'properties': [{'key': 'GoTerm', 'value': 'F:nickel cation binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0051082', 'properties': [{'key': 'GoTerm', 'value': 'F:unfolded protein binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006457', 'properties': [{'key': 'GoTerm', 'value': 'P:protein folding'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0065003', 'properties': [{'key': 'GoTerm', 'value': 'P:protein-containing complex assembly'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0019627', 'properties': [{'key': 'GoTerm', 'value': 'P:urea metabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'CDD', 'id': 'cd00571', 'properties': [{'key': 'EntryName', 'value': 'UreE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '2.60.260.20', 'properties': [{'key': 'EntryName', 'value': 'Urease metallochaperone UreE, N-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.30.70.790', 'properties': [{'key': 'EntryName', 'value': 'UreE, C-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00822', 'properties': [{'key': 'EntryName', 'value': 'UreE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR012406', 'properties': [{'key': 'EntryName', 'value': 'UreE'}]}, {'database': 'InterPro', 'id': 'IPR007864', 'properties': [{'key': 'EntryName', 'value': 'UreE_C_dom'}]}, {'database': 'InterPro', 'id': 'IPR004029', 'properties': [{'key': 'EntryName', 'value': 'UreE_N'}]}, {'database': 'InterPro', 'id': 'IPR036118', 'properties': [{'key': 'EntryName', 'value': 'UreE_N_sf'}]}, {'database': 'Pfam', 'id': 'PF05194', 'properties': [{'key': 'EntryName', 'value': 'UreE_C'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02814', 'properties': [{'key': 'EntryName', 'value': 'UreE_N'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF036402', 'properties': [{'key': 'EntryName', 'value': 'Ureas_acces_UreE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SMART', 'id': 'SM00988', 'properties': [{'key': 'EntryName', 'value': 'UreE_N'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF69737', 'properties': [{'key': 'EntryName', 'value': 'Urease metallochaperone UreE, C-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF69287', 'properties': [{'key': 'EntryName', 'value': 'Urease metallochaperone UreE, N-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MIITAIDTNIYDQPEFVEGRDVIGVRFEDLVLDKRIQRVALPGGEELGLRLNHGHPILREGDVLKADDKTVFVVEIIPTDVLVITPSDIHQMGFVAHSLGNRHLPAQFSKPGELTEKAAMIVQYDHTVVSFLDEHGIEYQRTELVPPIPFRHSGHTH', 'length': 157, 'molWeight': 17644, 'crc64': '4BBAA3381FCD5C21', 'md5': 'AF98D331066343E33EA9DA00EFEBD268'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Sequence conflict': 4}, 'uniParcId': 'UPI000003A006'}}}, {'from': 'CAF18656.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q79VJ2', 'secondaryAccessions': ['Q9L418', 'Q9RHM2'], 'uniProtkbId': 'UREF_CORGL', 'entryAudit': {'firstPublicDate': '2008-07-22', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 99, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '2: Evidence at transcript level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01385'}], 'value': 'Urease accessory protein UreF'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01385'}], 'value': 'ureF'}, 'orderedLocusNames': [{'value': 'Cgl0088'}, {'value': 'cg0117'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01385'}], 'value': 'Required for maturation of urease via the functional incorporation of the urease nickel metallocenter'}], 'commentType': 'FUNCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01385'}], 'value': 'UreD, UreF and UreG form a complex that acts as a GTP-hydrolysis-dependent molecular chaperone, activating the urease apoprotein by helping to assemble the nickel containing metallocenter of UreC. The UreE protein probably delivers the nickel'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01385'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01385'}], 'value': 'Belongs to the UreF family'}], 'commentType': 'SIMILARITY'}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Erroneous initiation', 'sequence': 'BAA88556.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 226, 'modifier': 'EXACT'}}, 'description': 'Urease accessory protein UreF', 'featureId': 'PRO_0000344116'}, {'type': 'Sequence conflict', 'location': {'start': {'value': 2, 'modifier': 'EXACT'}, 'end': {'value': 22, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88556', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'DLDADFLLLHLSDSALPTGAF', 'alternativeSequences': ['MISTLISAVALSGSATSTGRV']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 26, 'modifier': 'EXACT'}, 'end': {'value': 30, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88556', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'FGFET', 'alternativeSequences': ['LGLKN']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 39, 'modifier': 'EXACT'}, 'end': {'value': 46, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88556', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'NAEEFQDW', 'alternativeSequences': ['MQRSSKTG']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 179, 'modifier': 'EXACT'}, 'end': {'value': 179, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88556', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'V', 'alternativeSequences': ['M']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 193, 'modifier': 'EXACT'}, 'end': {'value': 193, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88556', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'R', 'alternativeSequences': ['L']}}], 'keywords': [{'id': 'KW-0143', 'category': 'Molecular function', 'name': 'Chaperone'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0996', 'category': 'Biological process', 'name': 'Nickel insertion'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'PROBABLE OPERON STRUCTURE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88556.1'}, {'key': 'Status', 'value': 'ALT_INIT'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81939.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97481.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18656.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599340.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013380.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q79VJ2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q79VJ2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0117', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'DNASU', 'id': '3342891', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97481', 'properties': [{'key': 'ProteinId', 'value': 'BAB97481'}, {'key': 'GeneId', 'value': 'BAB97481'}]}, {'database': 'KEGG', 'id': 'cgb:cg0117', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0088', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.89', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0830', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_049215_4_2_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'TQYTRLF', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9798772at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0016151', 'properties': [{'key': 'GoTerm', 'value': 'F:nickel cation binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006807', 'properties': [{'key': 'GoTerm', 'value': 'P:nitrogen compound metabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'Gene3D', 'id': '1.10.4190.10', 'properties': [{'key': 'EntryName', 'value': 'Urease accessory protein UreF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01385', 'properties': [{'key': 'EntryName', 'value': 'UreF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR002639', 'properties': [{'key': 'EntryName', 'value': 'UreF'}]}, {'database': 'InterPro', 'id': 'IPR038277', 'properties': [{'key': 'EntryName', 'value': 'UreF_sf'}]}, {'database': 'PANTHER', 'id': 'PTHR33620', 'properties': [{'key': 'EntryName', 'value': 'UREASE ACCESSORY PROTEIN F'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR33620:SF1', 'properties': [{'key': 'EntryName', 'value': 'UREASE ACCESSORY PROTEIN F'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF01730', 'properties': [{'key': 'EntryName', 'value': 'UreF'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF009467', 'properties': [{'key': 'EntryName', 'value': 'Ureas_acces_UreF'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MDLDADFLLLHLSDSALPTGAFAHSFGFETYMDAERITNAEEFQDWLKVLLKVQLTSSDALAMRMFYATPTVSELKRLDERLFAGTPAREIREANARMGTRMAEIVAETYSVPLIVEYLELIQHRELSGHPALALALATHSAGIDVDRAIHAHLTATVSSLIQNAVRGIPLGQMAGQRVMFAMREHIGAAVKRSANLDEIDFCSGDPGLDISQMVHETQRARLFMS', 'length': 226, 'molWeight': 25000, 'crc64': '66DFBFB9B075F1AC', 'md5': 'CB0DDFBE18F8593EE754F5AFD63E1804'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'SIMILARITY': 1, 'SEQUENCE CAUTION': 1}, 'countByFeatureType': {'Chain': 1, 'Sequence conflict': 5}, 'uniParcId': 'UPI00000366A9'}}}, {'from': 'CAF18657.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q79VJ1', 'secondaryAccessions': ['Q9L417', 'Q9RHM1'], 'uniProtkbId': 'UREG_CORGL', 'entryAudit': {'firstPublicDate': '2008-09-02', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 107, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '2: Evidence at transcript level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'value': 'Urease accessory protein UreG'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'value': 'ureG'}, 'orderedLocusNames': [{'value': 'Cgl0089'}, {'value': 'cg0118'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'value': 'Facilitates the functional incorporation of the urease nickel metallocenter. This process requires GTP hydrolysis, probably effectuated by UreG'}], 'commentType': 'FUNCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'value': 'Homodimer. UreD, UreF and UreG form a complex that acts as a GTP-hydrolysis-dependent molecular chaperone, activating the urease apoprotein by helping to assemble the nickel containing metallocenter of UreC. The UreE protein probably delivers the nickel'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'value': 'Belongs to the SIMIBI class G3E GTPase family. UreG subfamily'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 205, 'modifier': 'EXACT'}}, 'description': 'Urease accessory protein UreG', 'featureId': 'PRO_0000347384'}, {'type': 'Binding site', 'location': {'start': {'value': 10, 'modifier': 'EXACT'}, 'end': {'value': 17, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:37565'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01389'}], 'ligand': {'name': 'GTP', 'id': 'ChEBI:CHEBI:37565'}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 198, 'modifier': 'EXACT'}, 'end': {'value': 198, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88557', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'M', 'alternativeSequences': ['I']}}], 'keywords': [{'id': 'KW-0143', 'category': 'Molecular function', 'name': 'Chaperone'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0342', 'category': 'Ligand', 'name': 'GTP-binding'}, {'id': 'KW-0996', 'category': 'Biological process', 'name': 'Nickel insertion'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'PROBABLE OPERON STRUCTURE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88557.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81940.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97482.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18657.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599341.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003857703.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q79VJ1', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q79VJ1', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0118', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97482', 'properties': [{'key': 'ProteinId', 'value': 'BAB97482'}, {'key': 'GeneId', 'value': 'BAB97482'}]}, {'database': 'GeneID', 'id': '69620498', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgb:cg0118', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0089', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.90', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0378', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_072144_1_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'TQEDAQH', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9802035at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0005525', 'properties': [{'key': 'GoTerm', 'value': 'F:GTP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0003924', 'properties': [{'key': 'GoTerm', 'value': 'F:GTPase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0016151', 'properties': [{'key': 'GoTerm', 'value': 'F:nickel cation binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006807', 'properties': [{'key': 'GoTerm', 'value': 'P:nitrogen compound metabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'CDD', 'id': 'cd05540', 'properties': [{'key': 'EntryName', 'value': 'UreG'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.40.50.300', 'properties': [{'key': 'EntryName', 'value': 'P-loop containing nucleotide triphosphate hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01389', 'properties': [{'key': 'EntryName', 'value': 'UreG'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR003495', 'properties': [{'key': 'EntryName', 'value': 'CobW/HypB/UreG_dom'}]}, {'database': 'InterPro', 'id': 'IPR027417', 'properties': [{'key': 'EntryName', 'value': 'P-loop_NTPase'}]}, {'database': 'InterPro', 'id': 'IPR004400', 'properties': [{'key': 'EntryName', 'value': 'UreG'}]}, {'database': 'PANTHER', 'id': 'PTHR31715', 'properties': [{'key': 'EntryName', 'value': 'UREASE ACCESSORY PROTEIN G'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR31715:SF0', 'properties': [{'key': 'EntryName', 'value': 'UREASE ACCESSORY PROTEIN G'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02492', 'properties': [{'key': 'EntryName', 'value': 'cobW'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF005624', 'properties': [{'key': 'EntryName', 'value': 'Ni-bind_GTPase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF52540', 'properties': [{'key': 'EntryName', 'value': 'P-loop containing nucleoside triphosphate hydrolases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00101', 'properties': [{'key': 'EntryName', 'value': 'ureG'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MGPIRIGVGGPVGAGKTQLVERITRALIDEVSMAAITNDIYTIEDAKILAANGVLPEERIVGIETGGCPHTAIREDTSMNDAAIKDLVERFPDLELIFVESGGDNLSATFSPELVDFSIYIIDVAQGEKIPRKAGQGMIKSDLFIINKTDLAPYVGANLDVMVEDAKAFRKNKPFCLTNLRTDDGLDKVLEWIRHEVMMQDLQEA', 'length': 205, 'molWeight': 22418, 'crc64': '202DA9A718F03D86', 'md5': 'D321236940B59A682A6DE3485C6AAAC3'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Binding site': 1, 'Sequence conflict': 1}, 'uniParcId': 'UPI00000366AA'}}}, {'from': 'CAF18658.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q79VJ0', 'secondaryAccessions': ['Q9L416', 'Q9RHM0'], 'uniProtkbId': 'URED_CORGL', 'entryAudit': {'firstPublicDate': '2008-06-10', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-07-05', 'entryVersion': 102, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '2: Evidence at transcript level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01384'}], 'value': 'Urease accessory protein UreD'}}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01384'}], 'value': 'ureD1'}, 'orderedLocusNames': [{'value': 'Cgl0090'}, {'value': 'cg0119'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250'}], 'value': 'Probably acts in the maturation of urease via the functional incorporation of the urease nickel metallocenter (By similarity). Required for urease expression'}], 'commentType': 'FUNCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01384'}], 'value': 'UreD, UreF and UreG form a complex that acts as a GTP-hydrolysis-dependent molecular chaperone, activating the urease apoprotein by helping to assemble the nickel containing metallocenter of UreC. The UreE protein probably delivers the nickel'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01384'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '11328647'}], 'value': 'By urea and nitrogen starvation'}], 'commentType': 'INDUCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '10930756'}], 'value': 'Cells are urease negative and are no longer able to transport urea'}], 'commentType': 'DISRUPTION PHENOTYPE'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01384'}], 'value': 'Belongs to the UreD family'}], 'commentType': 'SIMILARITY'}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Frameshift', 'sequence': 'BAA88558.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 283, 'modifier': 'EXACT'}}, 'description': 'Urease accessory protein UreD', 'featureId': 'PRO_0000340448'}, {'type': 'Region', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 21, 'modifier': 'EXACT'}}, 'description': 'Disordered', 'evidences': [{'evidenceCode': 'ECO:0000256', 'source': 'SAM', 'id': 'MobiDB-lite'}]}, {'type': 'Sequence conflict', 'location': {'start': {'value': 222, 'modifier': 'EXACT'}, 'end': {'value': 222, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88558', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'L', 'alternativeSequences': ['I']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 241, 'modifier': 'EXACT'}, 'end': {'value': 241, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88558', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'L', 'alternativeSequences': ['E']}}, {'type': 'Sequence conflict', 'location': {'start': {'value': 258, 'modifier': 'EXACT'}, 'end': {'value': 258, 'modifier': 'EXACT'}}, 'description': 'in Ref. 1; BAA88558', 'evidences': [{'evidenceCode': 'ECO:0000305'}], 'alternativeSequence': {'originalSequence': 'I', 'alternativeSequences': ['V']}}], 'keywords': [{'id': 'KW-0143', 'category': 'Molecular function', 'name': 'Chaperone'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0996', 'category': 'Biological process', 'name': 'Nickel insertion'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '11328647', 'citationType': 'journal article', 'authors': ['Puskas L.G.', 'Inui M.', 'Yukawa H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '11328647'}, {'database': 'DOI', 'id': '10.3109/10425170009033989'}], 'title': 'Structure of the urease operon of Corynebacterium glutamicum.', 'publicationDate': '2000', 'journal': 'DNA Seq.', 'firstPage': '383', 'lastPage': '394', 'volume': '11'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'INDUCTION'], 'referenceComments': [{'value': 'ATCC 13869 / DSMZ 1412 / NCIMB 9567', 'type': 'STRAIN'}]}, {'citation': {'id': '10930756', 'citationType': 'journal article', 'authors': ['Nolden L.', 'Beckers G.', 'Moeckel B.', 'Pfefferle W.', 'Nampoothiri K.M.', 'Kraemer R.', 'Burkovski A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10930756'}, {'database': 'DOI', 'id': '10.1111/j.1574-6968.2000.tb09248.x'}], 'title': 'Urease of Corynebacterium glutamicum: organization of corresponding genes and investigation of activity.', 'publicationDate': '2000', 'journal': 'FEMS Microbiol. Lett.', 'firstPage': '305', 'lastPage': '310', 'volume': '189'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'PROBABLE OPERON STRUCTURE', 'DISRUPTION PHENOTYPE'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AB029154', 'properties': [{'key': 'ProteinId', 'value': 'BAA88558.1'}, {'key': 'Status', 'value': 'ALT_FRAME'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'AJ251883', 'properties': [{'key': 'ProteinId', 'value': 'CAB81941.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97483.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18658.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599342.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013381.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q79VJ0', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q79VJ0', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0119', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97483', 'properties': [{'key': 'ProteinId', 'value': 'BAB97483'}, {'key': 'GeneId', 'value': 'BAB97483'}]}, {'database': 'KEGG', 'id': 'cgb:cg0119', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0090', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.91', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0829', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_056339_5_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'CQHIIVH', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9807968at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0016151', 'properties': [{'key': 'GoTerm', 'value': 'F:nickel cation binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006807', 'properties': [{'key': 'GoTerm', 'value': 'P:nitrogen compound metabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'HAMAP', 'id': 'MF_01384', 'properties': [{'key': 'EntryName', 'value': 'UreD'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR002669', 'properties': [{'key': 'EntryName', 'value': 'UreD'}]}, {'database': 'PANTHER', 'id': 'PTHR33643', 'properties': [{'key': 'EntryName', 'value': 'UREASE ACCESSORY PROTEIN D'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR33643:SF1', 'properties': [{'key': 'EntryName', 'value': 'UREASE ACCESSORY PROTEIN D'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF01774', 'properties': [{'key': 'EntryName', 'value': 'UreD'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MTQTQPVGTLRLTIDDQGPQGQSRAVEQFHQGALRVIRPHYLDDSGQVCYTIIAIGGGYLGGDVYEQQFTIKDNAKALITTQSATKIYRTPQGPATQHTEINVGENAVLEYLADQTIAYREATYHQFTKVALHPSATFVMSEQITPGWHPDGKHFAYDEMRLHTEITDSTTGRLVLLDNLLLRPDSREGSFGWTEQYTHSGQMIVMGEGVDKQLVAELNEQLAAHPDVYGAVNFLSAPGTLLRGFIARTLSNRTEELINLHEHIASLLRGRWRGQEPVNLRKY', 'length': 283, 'molWeight': 31646, 'crc64': 'BE1B05AE876E7417', 'md5': '31DA20C4CB31AA9C19C8145F21E81CC5'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'INDUCTION': 1, 'DISRUPTION PHENOTYPE': 1, 'SIMILARITY': 1, 'SEQUENCE CAUTION': 1}, 'countByFeatureType': {'Chain': 1, 'Region': 1, 'Sequence conflict': 3}, 'uniParcId': 'UPI00000366A7'}}}, {'from': 'CAF18681.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q9X713', 'uniProtkbId': 'PANC_CORGL', 'entryAudit': {'firstPublicDate': '2000-05-30', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '1999-11-01', 'entryVersion': 126, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'value': 'Pantothenate synthetase'}, 'shortNames': [{'value': 'PS'}], 'ecNumbers': [{'value': '6.3.2.1'}]}, 'alternativeNames': [{'fullName': {'value': 'Pantoate--beta-alanine ligase'}}, {'fullName': {'value': 'Pantoate-activating enzyme'}}]}, 'genes': [{'geneName': {'value': 'panC'}, 'orderedLocusNames': [{'value': 'Cgl0113'}, {'value': 'cg0148'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '10223988'}], 'value': 'Catalyzes the condensation of pantoate with beta-alanine in an ATP-dependent reaction via a pantoyl-adenylate intermediate'}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': '(R)-pantoate + ATP + beta-alanine = (R)-pantothenate + AMP + diphosphate + H(+)', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:10912'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:15980'}, {'database': 'ChEBI', 'id': 'CHEBI:29032'}, {'database': 'ChEBI', 'id': 'CHEBI:30616'}, {'database': 'ChEBI', 'id': 'CHEBI:33019'}, {'database': 'ChEBI', 'id': 'CHEBI:57966'}, {'database': 'ChEBI', 'id': 'CHEBI:456215'}], 'ecNumber': '6.3.2.1', 'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '10223988'}]}}, {'texts': [{'value': 'Cofactor biosynthesis; (R)-pantothenate biosynthesis; (R)-pantothenate from (R)-pantoate and beta-alanine: step 1/1'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250'}], 'value': 'Homodimer'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250'}], 'value': 'The reaction proceeds by a bi uni uni bi ping pong mechanism'}], 'commentType': 'MISCELLANEOUS'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Belongs to the pantothenate synthetase family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 279, 'modifier': 'EXACT'}}, 'description': 'Pantothenate synthetase', 'featureId': 'PRO_0000128224'}, {'type': 'Active site', 'location': {'start': {'value': 33, 'modifier': 'EXACT'}, 'end': {'value': 33, 'modifier': 'EXACT'}}, 'description': 'Proton donor', 'evidences': [{'evidenceCode': 'ECO:0000250'}]}, {'type': 'Binding site', 'location': {'start': {'value': 26, 'modifier': 'EXACT'}, 'end': {'value': 33, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}, {'type': 'Binding site', 'location': {'start': {'value': 57, 'modifier': 'EXACT'}, 'end': {'value': 57, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:15980'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': '(R)-pantoate', 'id': 'ChEBI:CHEBI:15980'}}, {'type': 'Binding site', 'location': {'start': {'value': 57, 'modifier': 'EXACT'}, 'end': {'value': 57, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:57966'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'beta-alanine', 'id': 'ChEBI:CHEBI:57966'}}, {'type': 'Binding site', 'location': {'start': {'value': 147, 'modifier': 'EXACT'}, 'end': {'value': 150, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}, {'type': 'Binding site', 'location': {'start': {'value': 153, 'modifier': 'EXACT'}, 'end': {'value': 153, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:15980'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': '(R)-pantoate', 'id': 'ChEBI:CHEBI:15980'}}, {'type': 'Binding site', 'location': {'start': {'value': 176, 'modifier': 'EXACT'}, 'end': {'value': 176, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}, {'type': 'Binding site', 'location': {'start': {'value': 184, 'modifier': 'EXACT'}, 'end': {'value': 187, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}], 'keywords': [{'id': 'KW-0067', 'category': 'Ligand', 'name': 'ATP-binding'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0436', 'category': 'Molecular function', 'name': 'Ligase'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-0566', 'category': 'Biological process', 'name': 'Pantothenate biosynthesis'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '10223988', 'citationType': 'journal article', 'authors': ['Eggeling L.', 'Sahm H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10223988'}, {'database': 'DOI', 'id': '10.1128/aem.65.5.1973-1979.1999'}], 'title': 'D-pantothenate synthesis in Corynebacterium glutamicum and use of panBC and genes encoding L-valine synthesis for D-pantothenate overproduction.', 'publicationDate': '1999', 'journal': 'Appl. Environ. Microbiol.', 'firstPage': '1973', 'lastPage': '1979', 'volume': '65'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'FUNCTION', 'CATALYTIC ACTIVITY'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'X96580', 'properties': [{'key': 'ProteinId', 'value': 'CAA65398.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97506.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18681.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'PIR', 'id': 'T47120', 'properties': [{'key': 'EntryName', 'value': 'T47120'}]}, {'database': 'RefSeq', 'id': 'NP_599366.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013399.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q9X713', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q9X713', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0148', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97506', 'properties': [{'key': 'ProteinId', 'value': 'BAB97506'}, {'key': 'GeneId', 'value': 'BAB97506'}]}, {'database': 'KEGG', 'id': 'cgb:cg0148', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0113', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.116', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0414', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_047148_0_2_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'CNHKLEP', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9773087at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'UniPathway', 'id': 'UPA00028', 'properties': [{'key': 'RectionId', 'value': 'UER00005'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0005524', 'properties': [{'key': 'GoTerm', 'value': 'F:ATP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0004592', 'properties': [{'key': 'GoTerm', 'value': 'F:pantoate-beta-alanine ligase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0015940', 'properties': [{'key': 'GoTerm', 'value': 'P:pantothenate biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd00560', 'properties': [{'key': 'EntryName', 'value': 'PanC'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.40.50.620', 'properties': [{'key': 'EntryName', 'value': 'HUPs'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.30.1300.10', 'properties': [{'key': 'EntryName', 'value': 'Pantoate-beta-alanine ligase, C-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00158', 'properties': [{'key': 'EntryName', 'value': 'PanC'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR003721', 'properties': [{'key': 'EntryName', 'value': 'Pantoate_ligase'}]}, {'database': 'InterPro', 'id': 'IPR042176', 'properties': [{'key': 'EntryName', 'value': 'Pantoate_ligase_C'}]}, {'database': 'InterPro', 'id': 'IPR014729', 'properties': [{'key': 'EntryName', 'value': 'Rossmann-like_a/b/a_fold'}]}, {'database': 'PANTHER', 'id': 'PTHR21299', 'properties': [{'key': 'EntryName', 'value': 'CYTIDYLATE KINASE/PANTOATE-BETA-ALANINE LIGASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR21299:SF1', 'properties': [{'key': 'EntryName', 'value': 'PANTOATE--BETA-ALANINE LIGASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02569', 'properties': [{'key': 'EntryName', 'value': 'Pantoate_ligase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF52374', 'properties': [{'key': 'EntryName', 'value': 'Nucleotidylyl transferase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00018', 'properties': [{'key': 'EntryName', 'value': 'panC'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MQVATTKQALIDALLHHKSVGLVPTMGALHSGHASLVKAARAENDTVVASIFVNPLQFEALGDCDDYRNYPRQLDADLALLEEAGVDIVFAPDVEEMYPGGLPLVWARTGSIGTKLEGASRPGHFDGVATVVAKLFNLVRPDRAYFGQKDAQQVAVIRRLVADLDIPVEIRPVPIIRGADGLAESSRNQRLSADQRAQALVLPQVLSGLQRRKAAGEALDIQGARDTLASADGVRLDHLEIVDPATLEPLEIDGLLTQPALVVGAIFVGPVRLIDNIEL', 'length': 279, 'molWeight': 29888, 'crc64': 'EE4CF5EB4198A768', 'md5': '87D41F217E5385D63665AEBF12E0C34B'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'MISCELLANEOUS': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Active site': 1, 'Binding site': 7}, 'uniParcId': 'UPI00000339AE'}}}, {'from': 'CAF18682.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q9X712', 'uniProtkbId': 'PANB_CORGL', 'entryAudit': {'firstPublicDate': '2000-12-01', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2003-04-23', 'entryVersion': 128, 'sequenceVersion': 2}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'value': '3-methyl-2-oxobutanoate hydroxymethyltransferase'}, 'ecNumbers': [{'value': '2.1.2.11'}]}, 'alternativeNames': [{'fullName': {'value': 'Ketopantoate hydroxymethyltransferase'}, 'shortNames': [{'value': 'KPHMT'}]}]}, 'genes': [{'geneName': {'value': 'panB'}, 'orderedLocusNames': [{'value': 'Cgl0114'}, {'value': 'cg0149'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '10223988'}], 'value': 'Catalyzes the reversible reaction in which hydroxymethyl group from 5,10-methylenetetrahydrofolate is transferred onto alpha-ketoisovalerate to form ketopantoate'}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': '(6R)-5,10-methylene-5,6,7,8-tetrahydrofolate + 3-methyl-2-oxobutanoate + H2O = (6S)-5,6,7,8-tetrahydrofolate + 2-dehydropantoate', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:11824'}, {'database': 'ChEBI', 'id': 'CHEBI:11561'}, {'database': 'ChEBI', 'id': 'CHEBI:11851'}, {'database': 'ChEBI', 'id': 'CHEBI:15377'}, {'database': 'ChEBI', 'id': 'CHEBI:15636'}, {'database': 'ChEBI', 'id': 'CHEBI:57453'}], 'ecNumber': '2.1.2.11', 'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '10223988'}]}}, {'commentType': 'COFACTOR', 'cofactors': [{'name': 'Mg(2+)', 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:18420'}}], 'note': {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250'}], 'value': 'Binds 1 Mg(2+) ion per subunit.'}]}}, {'texts': [{'value': 'Cofactor biosynthesis; (R)-pantothenate biosynthesis; (R)-pantoate from 3-methyl-2-oxobutanoate: step 1/2'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250'}], 'value': 'Homodecamer; pentamer of dimers'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Belongs to the PanB family'}], 'commentType': 'SIMILARITY'}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Erroneous initiation', 'sequence': 'BAB97507.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Erroneous initiation', 'sequence': 'CAA65397.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Erroneous initiation', 'sequence': 'CAF18682.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 269, 'modifier': 'EXACT'}}, 'description': '3-methyl-2-oxobutanoate hydroxymethyltransferase', 'featureId': 'PRO_0000184838'}, {'type': 'Active site', 'location': {'start': {'value': 187, 'modifier': 'EXACT'}, 'end': {'value': 187, 'modifier': 'EXACT'}}, 'description': 'Proton acceptor', 'evidences': [{'evidenceCode': 'ECO:0000250'}]}, {'type': 'Binding site', 'location': {'start': {'value': 50, 'modifier': 'EXACT'}, 'end': {'value': 51, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:11851'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': '3-methyl-2-oxobutanoate', 'id': 'ChEBI:CHEBI:11851'}}, {'type': 'Binding site', 'location': {'start': {'value': 50, 'modifier': 'EXACT'}, 'end': {'value': 50, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:18420'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'Mg(2+)', 'id': 'ChEBI:CHEBI:18420'}}, {'type': 'Binding site', 'location': {'start': {'value': 89, 'modifier': 'EXACT'}, 'end': {'value': 89, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:11851'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': '3-methyl-2-oxobutanoate', 'id': 'ChEBI:CHEBI:11851'}}, {'type': 'Binding site', 'location': {'start': {'value': 89, 'modifier': 'EXACT'}, 'end': {'value': 89, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:18420'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'Mg(2+)', 'id': 'ChEBI:CHEBI:18420'}}, {'type': 'Binding site', 'location': {'start': {'value': 119, 'modifier': 'EXACT'}, 'end': {'value': 119, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:11851'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': '3-methyl-2-oxobutanoate', 'id': 'ChEBI:CHEBI:11851'}}, {'type': 'Binding site', 'location': {'start': {'value': 121, 'modifier': 'EXACT'}, 'end': {'value': 121, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:18420'}], 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'ligand': {'name': 'Mg(2+)', 'id': 'ChEBI:CHEBI:18420'}}], 'keywords': [{'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0460', 'category': 'Ligand', 'name': 'Magnesium'}, {'id': 'KW-0479', 'category': 'Ligand', 'name': 'Metal-binding'}, {'id': 'KW-0566', 'category': 'Biological process', 'name': 'Pantothenate biosynthesis'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0808', 'category': 'Molecular function', 'name': 'Transferase'}], 'references': [{'citation': {'id': '10223988', 'citationType': 'journal article', 'authors': ['Eggeling L.', 'Sahm H.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10223988'}, {'database': 'DOI', 'id': '10.1128/aem.65.5.1973-1979.1999'}], 'title': 'D-pantothenate synthesis in Corynebacterium glutamicum and use of panBC and genes encoding L-valine synthesis for D-pantothenate overproduction.', 'publicationDate': '1999', 'journal': 'Appl. Environ. Microbiol.', 'firstPage': '1973', 'lastPage': '1979', 'volume': '65'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'FUNCTION', 'CATALYTIC ACTIVITY'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'X96580', 'properties': [{'key': 'ProteinId', 'value': 'CAA65397.1'}, {'key': 'Status', 'value': 'ALT_INIT'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97507.1'}, {'key': 'Status', 'value': 'ALT_INIT'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18682.1'}, {'key': 'Status', 'value': 'ALT_INIT'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'PIR', 'id': 'T47119', 'properties': [{'key': 'EntryName', 'value': 'T47119'}]}, {'database': 'RefSeq', 'id': 'NP_599367.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013400.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q9X712', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q9X712', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0149', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97507', 'properties': [{'key': 'ProteinId', 'value': 'BAB97507'}, {'key': 'GeneId', 'value': 'BAB97507'}]}, {'database': 'KEGG', 'id': 'cgb:cg0149', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0114', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.117', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0413', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_036645_1_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'VLVWTDM', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9781789at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'UniPathway', 'id': 'UPA00028', 'properties': [{'key': 'RectionId', 'value': 'UER00003'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0003864', 'properties': [{'key': 'GoTerm', 'value': 'F:3-methyl-2-oxobutanoate hydroxymethyltransferase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0046872', 'properties': [{'key': 'GoTerm', 'value': 'F:metal ion binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0015940', 'properties': [{'key': 'GoTerm', 'value': 'P:pantothenate biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd06557', 'properties': [{'key': 'EntryName', 'value': 'KPHMT-like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.20.20.60', 'properties': [{'key': 'EntryName', 'value': 'Phosphoenolpyruvate-binding domains'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00156', 'properties': [{'key': 'EntryName', 'value': 'PanB'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR003700', 'properties': [{'key': 'EntryName', 'value': 'Pantoate_hydroxy_MeTrfase'}]}, {'database': 'InterPro', 'id': 'IPR015813', 'properties': [{'key': 'EntryName', 'value': 'Pyrv/PenolPyrv_Kinase-like_dom'}]}, {'database': 'InterPro', 'id': 'IPR040442', 'properties': [{'key': 'EntryName', 'value': 'Pyrv_Kinase-like_dom_sf'}]}, {'database': 'PANTHER', 'id': 'PTHR20881', 'properties': [{'key': 'EntryName', 'value': '3-METHYL-2-OXOBUTANOATE HYDROXYMETHYLTRANSFERASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR20881:SF0', 'properties': [{'key': 'EntryName', 'value': '3-METHYL-2-OXOBUTANOATE HYDROXYMETHYLTRANSFERASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02548', 'properties': [{'key': 'EntryName', 'value': 'Pantoate_transf'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF000388', 'properties': [{'key': 'EntryName', 'value': 'Pantoate_hydroxy_MeTrfase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF51621', 'properties': [{'key': 'EntryName', 'value': 'Phosphoenolpyruvate/pyruvate domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00222', 'properties': [{'key': 'EntryName', 'value': 'panB'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSGIDAKKIRTRHFREAKVNGQKVSVLTSYDALSARIFDEAGVDMLLVGDSAANVVLGRDTTLSITLDEMIVLAKAVTIATKRALVVVDLPFGTYEVSPNQAVESAIRVMRETGAAAVKIEGGVEIAQTIRRIVDAGIPVVGHIGYTPQSEHSLGGHVVQGRGASSGKLIADARALEQAGAFAVVLEMVPAEAAREVTEDLSITTIGIGAGNGTDGQVLVWQDAFGLNRGKKPRFVREYATLGDSLHDAAQAYIADIHAGTFPGEAESF', 'length': 269, 'molWeight': 28324, 'crc64': 'C494D4FF3D2A0047', 'md5': 'AA66879465229D6F341FE27E83EDF3D1'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'COFACTOR': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'SIMILARITY': 1, 'SEQUENCE CAUTION': 3}, 'countByFeatureType': {'Chain': 1, 'Active site': 1, 'Binding site': 6}, 'uniParcId': 'UPI00000363EA'}}}, {'from': 'CAF18684.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NU33', 'uniProtkbId': '3MGH_CORGL', 'entryAudit': {'firstPublicDate': '2002-10-10', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2004-04-26', 'entryVersion': 116, 'sequenceVersion': 2}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00527'}], 'value': 'Putative 3-methyladenine DNA glycosylase'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00527'}], 'value': '3.2.2.-'}]}}, 'genes': [{'geneName': {'value': 'mag'}, 'orderedLocusNames': [{'value': 'Cgl0116'}, {'value': 'cg0151'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00527'}], 'value': 'Belongs to the DNA glycosylase MPG family'}], 'commentType': 'SIMILARITY'}, {'commentType': 'SEQUENCE CAUTION', 'sequenceCautionType': 'Erroneous initiation', 'sequence': 'BAB97509.1', 'evidences': [{'evidenceCode': 'ECO:0000305'}]}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 189, 'modifier': 'EXACT'}}, 'description': 'Putative 3-methyladenine DNA glycosylase', 'featureId': 'PRO_0000100083'}], 'keywords': [{'id': 'KW-0227', 'category': 'Biological process', 'name': 'DNA damage'}, {'id': 'KW-0234', 'category': 'Biological process', 'name': 'DNA repair'}, {'id': 'KW-0378', 'category': 'Molecular function', 'name': 'Hydrolase'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97509.1'}, {'key': 'Status', 'value': 'ALT_INIT'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18684.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_694632.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011265466.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NU33', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NU33', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0151', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97509', 'properties': [{'key': 'ProteinId', 'value': 'BAB97509'}, {'key': 'GeneId', 'value': 'BAB97509'}]}, {'database': 'KEGG', 'id': 'cgb:cg0151', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0116', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.119', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG2094', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_060471_3_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'VEAYHHT', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9794313at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0003905', 'properties': [{'key': 'GoTerm', 'value': 'F:alkylbase DNA N-glycosylase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0003677', 'properties': [{'key': 'GoTerm', 'value': 'F:DNA binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0006284', 'properties': [{'key': 'GoTerm', 'value': 'P:base-excision repair'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'CDD', 'id': 'cd00540', 'properties': [{'key': 'EntryName', 'value': 'AAG'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.10.300.10', 'properties': [{'key': 'EntryName', 'value': 'Methylpurine-DNA glycosylase (MPG)'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00527', 'properties': [{'key': 'EntryName', 'value': '3MGH'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR011034', 'properties': [{'key': 'EntryName', 'value': 'Formyl_transferase-like_C_sf'}]}, {'database': 'InterPro', 'id': 'IPR003180', 'properties': [{'key': 'EntryName', 'value': 'MPG'}]}, {'database': 'InterPro', 'id': 'IPR036995', 'properties': [{'key': 'EntryName', 'value': 'MPG_sf'}]}, {'database': 'PANTHER', 'id': 'PTHR10429', 'properties': [{'key': 'EntryName', 'value': 'DNA-3-METHYLADENINE GLYCOSYLASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR10429:SF0', 'properties': [{'key': 'EntryName', 'value': 'DNA-3-METHYLADENINE GLYCOSYLASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02245', 'properties': [{'key': 'EntryName', 'value': 'Pur_DNA_glyco'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF50486', 'properties': [{'key': 'EntryName', 'value': 'FMT C-terminal domain-like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00567', 'properties': [{'key': 'EntryName', 'value': '3mg'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MPIDFLQPADIVAPQLLGCTLTHGGVGIRITEVEAYLDSTDEAAHTYRGKTPRNAAMFGPGGHMYVYISYGIHRAGNIVCGPEGTGQGVLLRAGEVVSGESIAQSRRGEGIPHARLAQGPGNFGQALGLEISDNHASVFGPSFLISDRVETPEIVRGPRIGISKNTEALLRFWIPNDPTVSGRRGYPKE', 'length': 189, 'molWeight': 20162, 'crc64': '0752BA7C882D746C', 'md5': '3D99EE81082DC71F0256C7D69902C083'}, 'extraAttributes': {'countByCommentType': {'SIMILARITY': 1, 'SEQUENCE CAUTION': 1}, 'countByFeatureType': {'Chain': 1}, 'uniParcId': 'UPI0000236BA5'}}}, {'from': 'CAF18702.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q9X4N0', 'uniProtkbId': 'PAND_CORGL', 'entryAudit': {'firstPublicDate': '2001-09-26', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '1999-11-01', 'entryVersion': 138, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Aspartate 1-decarboxylase'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': '4.1.1.11'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Aspartate alpha-decarboxylase'}}], 'contains': [{'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Aspartate 1-decarboxylase beta chain'}}}, {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Aspartate 1-decarboxylase alpha chain'}}}], 'flag': 'Precursor'}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'panD'}, 'orderedLocusNames': [{'value': 'Cgl0135'}, {'value': 'cg0172'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Catalyzes the pyruvoyl-dependent decarboxylation of aspartate to produce beta-alanine'}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'H(+) + L-aspartate = beta-alanine + CO2', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:19497'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:16526'}, {'database': 'ChEBI', 'id': 'CHEBI:29991'}, {'database': 'ChEBI', 'id': 'CHEBI:57966'}], 'ecNumber': '4.1.1.11', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}]}}, {'commentType': 'COFACTOR', 'cofactors': [{'name': 'pyruvate', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:15361'}}], 'note': {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Binds 1 pyruvoyl group covalently per subunit.'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Cofactor biosynthesis; (R)-pantothenate biosynthesis; beta-alanine from L-aspartate: step 1/1'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Heterooctamer of four alpha and four beta subunits'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Cytoplasm', 'id': 'SL-0086'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Is synthesized initially as an inactive proenzyme, which is activated by self-cleavage at a specific serine bond to produce a beta-subunit with a hydroxyl group at its C-terminus and an alpha-subunit with a pyruvoyl group at its N-terminus'}], 'commentType': 'PTM'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'value': 'Belongs to the PanD family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 24, 'modifier': 'EXACT'}}, 'description': 'Aspartate 1-decarboxylase beta chain', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'featureId': 'PRO_0000023067'}, {'type': 'Chain', 'location': {'start': {'value': 25, 'modifier': 'EXACT'}, 'end': {'value': 136, 'modifier': 'EXACT'}}, 'description': 'Aspartate 1-decarboxylase alpha chain', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'featureId': 'PRO_0000023068'}, {'type': 'Active site', 'location': {'start': {'value': 25, 'modifier': 'EXACT'}, 'end': {'value': 25, 'modifier': 'EXACT'}}, 'description': 'Schiff-base intermediate with substrate; via pyruvic acid', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}]}, {'type': 'Active site', 'location': {'start': {'value': 58, 'modifier': 'EXACT'}, 'end': {'value': 58, 'modifier': 'EXACT'}}, 'description': 'Proton donor', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}]}, {'type': 'Binding site', 'location': {'start': {'value': 57, 'modifier': 'EXACT'}, 'end': {'value': 57, 'modifier': 'EXACT'}}, 'description': '', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'ligand': {'name': 'substrate'}}, {'type': 'Binding site', 'location': {'start': {'value': 73, 'modifier': 'EXACT'}, 'end': {'value': 75, 'modifier': 'EXACT'}}, 'description': '', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}], 'ligand': {'name': 'substrate'}}, {'type': 'Modified residue', 'location': {'start': {'value': 25, 'modifier': 'EXACT'}, 'end': {'value': 25, 'modifier': 'EXACT'}}, 'description': 'Pyruvic acid (Ser)', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00446'}]}], 'keywords': [{'id': 'KW-0068', 'category': 'PTM', 'name': 'Autocatalytic cleavage'}, {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0210', 'category': 'Molecular function', 'name': 'Decarboxylase'}, {'id': 'KW-0456', 'category': 'Molecular function', 'name': 'Lyase'}, {'id': 'KW-0566', 'category': 'Biological process', 'name': 'Pantothenate biosynthesis'}, {'id': 'KW-0670', 'category': 'Ligand', 'name': 'Pyruvate'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0704', 'category': 'Ligand', 'name': 'Schiff base'}, {'id': 'KW-0865', 'category': 'PTM', 'name': 'Zymogen'}], 'references': [{'citation': {'id': '10103247', 'citationType': 'journal article', 'authors': ['Dusch N.', 'Puehler A.', 'Kalinowski J.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '10103247'}, {'database': 'DOI', 'id': '10.1128/aem.65.4.1530-1539.1999'}], 'title': 'Expression of the Corynebacterium glutamicum panD gene encoding L-aspartate-alpha-decarboxylase leads to pantothenate overproduction in Escherichia coli.', 'publicationDate': '1999', 'journal': 'Appl. Environ. Microbiol.', 'firstPage': '1530', 'lastPage': '1539', 'volume': '65'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'CHARACTERIZATION'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AF116184', 'properties': [{'key': 'ProteinId', 'value': 'AAD28430.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97528.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18702.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599388.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003857183.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q9X4N0', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q9X4N0', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0172', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97528', 'properties': [{'key': 'ProteinId', 'value': 'BAB97528'}, {'key': 'GeneId', 'value': 'BAB97528'}]}, {'database': 'GeneID', 'id': '69620590', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgb:cg0172', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0135', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.139', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0853', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_115305_2_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'HEPKLVF', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9803983at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'BRENDA', 'id': '4.1.1.11', 'properties': [{'key': 'OrganismId', 'value': '960'}]}, {'database': 'UniPathway', 'id': 'UPA00028', 'properties': [{'key': 'RectionId', 'value': 'UER00002'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0004068', 'properties': [{'key': 'GoTerm', 'value': 'F:aspartate 1-decarboxylase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006523', 'properties': [{'key': 'GoTerm', 'value': 'P:alanine biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0015940', 'properties': [{'key': 'GoTerm', 'value': 'P:pantothenate biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd06919', 'properties': [{'key': 'EntryName', 'value': 'Asp_decarbox'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '2.40.40.20', 'properties': [{'key': 'EntryName', 'value': '-'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00446', 'properties': [{'key': 'EntryName', 'value': 'PanD'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR009010', 'properties': [{'key': 'EntryName', 'value': 'Asp_de-COase-like_dom_sf'}]}, {'database': 'InterPro', 'id': 'IPR003190', 'properties': [{'key': 'EntryName', 'value': 'Asp_decarbox'}]}, {'database': 'PANTHER', 'id': 'PTHR21012', 'properties': [{'key': 'EntryName', 'value': 'ASPARTATE 1-DECARBOXYLASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR21012:SF0', 'properties': [{'key': 'EntryName', 'value': 'ASPARTATE 1-DECARBOXYLASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02261', 'properties': [{'key': 'EntryName', 'value': 'Asp_decarbox'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF006246', 'properties': [{'key': 'EntryName', 'value': 'Asp_decarbox'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF50692', 'properties': [{'key': 'EntryName', 'value': 'ADC-like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00223', 'properties': [{'key': 'EntryName', 'value': 'panD'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MLRTILGSKIHRATVTQADLDYVGSVTIDADLVHAAGLIEGEKVAIVDITNGARLETYVIVGDAGTGNICINGAAAHLINPGDLVIIMSYLQATDAEAKAYEPKIVHVDADNRIVALGNDLAEALPGSGLLTSRSI', 'length': 136, 'molWeight': 14147, 'crc64': 'A9747386D58A5E2D', 'md5': 'CEF6265E39A7BD82ABD7AFAB8B84ADE9'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'COFACTOR': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'PTM': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 2, 'Active site': 2, 'Binding site': 2, 'Modified residue': 1}, 'uniParcId': 'UPI00000339AC'}}}, {'from': 'CAF18731.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NTY7', 'secondaryAccessions': ['Q6M8J3'], 'uniProtkbId': 'IOLG_CORGL', 'entryAudit': {'firstPublicDate': '2008-10-14', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 115, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'Inositol 2-dehydrogenase'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': '1.1.1.18'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'Myo-inositol 2-dehydrogenase'}, 'shortNames': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'MI 2-dehydrogenase'}]}]}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'iolG'}, 'orderedLocusNames': [{'value': 'Cgl0164'}, {'value': 'cg0204'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'Involved in the oxidation of myo-inositol (MI) to 2-keto-myo-inositol (2KMI or 2-inosose)'}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'myo-inositol + NAD(+) = H(+) + NADH + scyllo-inosose', 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:16949'}, {'database': 'ChEBI', 'id': 'CHEBI:15378'}, {'database': 'ChEBI', 'id': 'CHEBI:17268'}, {'database': 'ChEBI', 'id': 'CHEBI:17811'}, {'database': 'ChEBI', 'id': 'CHEBI:57540'}, {'database': 'ChEBI', 'id': 'CHEBI:57945'}], 'ecNumber': '1.1.1.18', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'Homotetramer'}], 'commentType': 'SUBUNIT'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01671'}], 'value': 'Belongs to the Gfo/Idh/MocA family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 337, 'modifier': 'EXACT'}}, 'description': 'Inositol 2-dehydrogenase', 'featureId': 'PRO_0000352566'}], 'keywords': [{'id': 'KW-0520', 'category': 'Ligand', 'name': 'NAD'}, {'id': 'KW-0560', 'category': 'Molecular function', 'name': 'Oxidoreductase'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97557.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18731.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599416.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013436.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NTY7', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NTY7', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0204', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97557', 'properties': [{'key': 'ProteinId', 'value': 'BAB97557'}, {'key': 'GeneId', 'value': 'BAB97557'}]}, {'database': 'KEGG', 'id': 'cgb:cg0204', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0164', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.168', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0673', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_023194_0_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'RKPVMCE', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '256869at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'BRENDA', 'id': '1.1.1.18', 'properties': [{'key': 'OrganismId', 'value': '960'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0050112', 'properties': [{'key': 'GoTerm', 'value': 'F:inositol 2-dehydrogenase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0000166', 'properties': [{'key': 'GoTerm', 'value': 'F:nucleotide binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0019310', 'properties': [{'key': 'GoTerm', 'value': 'P:inositol catabolic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'Gene3D', 'id': '3.40.50.720', 'properties': [{'key': 'EntryName', 'value': 'NAD(P)-binding Rossmann-like Domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01671', 'properties': [{'key': 'EntryName', 'value': 'IolG'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR004104', 'properties': [{'key': 'EntryName', 'value': 'Gfo/Idh/MocA-like_OxRdtase_C'}]}, {'database': 'InterPro', 'id': 'IPR000683', 'properties': [{'key': 'EntryName', 'value': 'Gfo/Idh/MocA-like_OxRdtase_N'}]}, {'database': 'InterPro', 'id': 'IPR023794', 'properties': [{'key': 'EntryName', 'value': 'MI/DCI_dehydrogenase'}]}, {'database': 'InterPro', 'id': 'IPR036291', 'properties': [{'key': 'EntryName', 'value': 'NAD(P)-bd_dom_sf'}]}, {'database': 'PANTHER', 'id': 'PTHR43593', 'properties': [{'key': 'EntryName', 'value': '-'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR43593:SF1', 'properties': [{'key': 'EntryName', 'value': 'INOSITOL 2-DEHYDROGENASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF01408', 'properties': [{'key': 'EntryName', 'value': 'GFO_IDH_MocA'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02894', 'properties': [{'key': 'EntryName', 'value': 'GFO_IDH_MocA_C'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF55347', 'properties': [{'key': 'EntryName', 'value': 'Glyceraldehyde-3-phosphate dehydrogenase-like, C-terminal domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF51735', 'properties': [{'key': 'EntryName', 'value': 'NAD(P)-binding Rossmann-fold domains'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSKSLRVGVVGAGAMGADHIDRINNRTSGAHISAIIEPDAARAAAAAEDAPGAQAFTRIEDAIAADAVDAVLIAVPGQFHEPVLVPALEAGLPILCEKPLTPDSESSLRIVELEQKLDKPHIQVGFMRRFDPEYNNLRKLVESGEAGELLMLRGLHRNPSVGESYTQSMLITDSVVHEFDVIPWLAGSRVVSVEVKYPKTSSLAHSGLKEPILVIMELENGVLVDVEMNVNIQFGYQVATEAVFEKGLARIGQPSGMQRWRDGEFLINEHTDFTTRFATAYDRQIQSWVDAVHEGTLVAGPNAWDGYLVALSCEAGVKALDGGVIPVDAAPRPDFYA', 'length': 337, 'molWeight': 36431, 'crc64': 'F5EF6EAFA65924A3', 'md5': 'DD3435C932FABEF59E4E22DEF1FDCDC7'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'SUBUNIT': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1}, 'uniParcId': 'UPI000003A04D'}}}, {'from': 'CAF18758.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NTW4', 'secondaryAccessions': ['Q6M8G6'], 'uniProtkbId': 'AFTA_CORGL', 'entryAudit': {'firstPublicDate': '2013-01-09', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 84, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000250', 'source': 'UniProtKB', 'id': 'P9WN03'}], 'value': 'Galactan 5-O-arabinofuranosyltransferase'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000250', 'source': 'UniProtKB', 'id': 'P9WN03'}], 'value': '2.4.2.46'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000305', 'source': 'PubMed', 'id': '16595677'}], 'value': 'Arabinofuranosyltransferase AftA'}}]}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000303', 'source': 'PubMed', 'id': '16595677'}], 'value': 'aftA'}, 'orderedLocusNames': [{'value': 'Cgl0188'}, {'value': 'cg0236'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250', 'source': 'UniProtKB', 'id': 'P9WN03'}, {'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '16595677'}], 'value': \"Involved in the biosynthesis of the arabinogalactan (AG) region of the mycolylarabinogalactan-peptidoglycan (mAGP) complex, an essential component of the cell wall (PubMed:16595677). Catalyzes the addition of the first key arabinofuranosyl (Araf) residue from the sugar donor decaprenyl-phospho-arabinose (DPA) on the C-5 of a 6-linked galactofuranosyl (Galf) of the galactan domain, thus 'priming' the galactan for further elaboration by other arabinofuranosyltransferases (By similarity)\"}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': 'Adds an alpha-D-arabinofuranosyl group from trans,octacis-decaprenylphospho-beta-D-arabinofuranose at the 5-O-position of the eighth, tenth and twelfth galactofuranose unit of the galactofuranan chain of [beta-D-galactofuranosyl-(1->5)-beta-D-galactofuranosyl-(1->6)]14-beta-D-galactofuranosyl-(1->5)-beta-D-galactofuranosyl-(1->4)-alpha-L-rhamnopyranosyl-(1->3)-N-acetyl-alpha-D-glucosaminyl-diphospho-trans,octacis-decaprenol.', 'ecNumber': '2.4.2.46', 'evidences': [{'evidenceCode': 'ECO:0000250', 'source': 'UniProtKB', 'id': 'P9WN03'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '16595677'}], 'value': 'Cell wall biogenesis; cell wall polysaccharide biosynthesis'}], 'commentType': 'PATHWAY'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255'}], 'value': 'Cell membrane', 'id': 'SL-0039'}, 'topology': {'evidences': [{'evidenceCode': 'ECO:0000255'}], 'value': 'Multi-pass membrane protein', 'id': 'SL-9909'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '16595677'}], 'value': 'Cells lacking this gene have a cell wall that shows a complete absence of arabinose resulting in a truncated cell wall structure possessing only a galactan core with a concomitant loss of cell wall-bound corynomycolic acids'}], 'commentType': 'DISRUPTION PHENOTYPE'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Belongs to the glycosyltransferase 85 family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 661, 'modifier': 'EXACT'}}, 'description': 'Galactan 5-O-arabinofuranosyltransferase', 'featureId': 'PRO_0000420576'}, {'type': 'Transmembrane', 'location': {'start': {'value': 26, 'modifier': 'EXACT'}, 'end': {'value': 46, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 64, 'modifier': 'EXACT'}, 'end': {'value': 84, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 108, 'modifier': 'EXACT'}, 'end': {'value': 128, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 194, 'modifier': 'EXACT'}, 'end': {'value': 214, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 217, 'modifier': 'EXACT'}, 'end': {'value': 237, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 243, 'modifier': 'EXACT'}, 'end': {'value': 263, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 265, 'modifier': 'EXACT'}, 'end': {'value': 285, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 286, 'modifier': 'EXACT'}, 'end': {'value': 306, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 312, 'modifier': 'EXACT'}, 'end': {'value': 332, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 362, 'modifier': 'EXACT'}, 'end': {'value': 382, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 393, 'modifier': 'EXACT'}, 'end': {'value': 413, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 418, 'modifier': 'EXACT'}, 'end': {'value': 438, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}, {'type': 'Transmembrane', 'location': {'start': {'value': 458, 'modifier': 'EXACT'}, 'end': {'value': 478, 'modifier': 'EXACT'}}, 'description': 'Helical', 'evidences': [{'evidenceCode': 'ECO:0000255'}]}], 'keywords': [{'id': 'KW-1003', 'category': 'Cellular component', 'name': 'Cell membrane'}, {'id': 'KW-0961', 'category': 'Biological process', 'name': 'Cell wall biogenesis/degradation'}, {'id': 'KW-0328', 'category': 'Molecular function', 'name': 'Glycosyltransferase'}, {'id': 'KW-0472', 'category': 'Cellular component', 'name': 'Membrane'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0808', 'category': 'Molecular function', 'name': 'Transferase'}, {'id': 'KW-0812', 'category': 'Domain', 'name': 'Transmembrane'}, {'id': 'KW-1133', 'category': 'Domain', 'name': 'Transmembrane helix'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '16595677', 'citationType': 'journal article', 'authors': ['Alderwick L.J.', 'Seidel M.', 'Sahm H.', 'Besra G.S.', 'Eggeling L.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '16595677'}, {'database': 'DOI', 'id': '10.1074/jbc.m600045200'}], 'title': 'Identification of a novel arabinofuranosyltransferase (AftA) involved in cell wall arabinan biosynthesis in Mycobacterium tuberculosis.', 'publicationDate': '2006', 'journal': 'J. Biol. Chem.', 'firstPage': '15653', 'lastPage': '15661', 'volume': '281'}, 'referencePositions': ['FUNCTION IN ARABINAN BIOSYNTHESIS', 'DISRUPTION PHENOTYPE', 'PATHWAY'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97581.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18758.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599441.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NTW4', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0236', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'CAZy', 'id': 'GT85', 'properties': [{'key': 'FamilyName', 'value': 'Glycosyltransferase Family 85'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97581', 'properties': [{'key': 'ProteinId', 'value': 'BAB97581'}, {'key': 'GeneId', 'value': 'BAB97581'}]}, {'database': 'KEGG', 'id': 'cgb:cg0236', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0188', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.192', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'ENOG502ZB59', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_021304_0_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'YPAGWFW', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '4775300at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'BRENDA', 'id': '2.4.2.46', 'properties': [{'key': 'OrganismId', 'value': '960'}]}, {'database': 'UniPathway', 'id': 'UPA00963', 'properties': [{'key': 'RectionId', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0005886', 'properties': [{'key': 'GoTerm', 'value': 'C:plasma membrane'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-SubCell'}]}, {'database': 'GO', 'id': 'GO:0016757', 'properties': [{'key': 'GoTerm', 'value': 'F:glycosyltransferase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0045227', 'properties': [{'key': 'GoTerm', 'value': 'P:capsule polysaccharide biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniPathway'}]}, {'database': 'GO', 'id': 'GO:0044038', 'properties': [{'key': 'GoTerm', 'value': 'P:cell wall macromolecule biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0071555', 'properties': [{'key': 'GoTerm', 'value': 'P:cell wall organization'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'InterPro', 'id': 'IPR020959', 'properties': [{'key': 'EntryName', 'value': 'ArabinofuranosylTrfase_AftA_C'}]}, {'database': 'InterPro', 'id': 'IPR020963', 'properties': [{'key': 'EntryName', 'value': 'ArabinofuranosylTrfase_AftA_N'}]}, {'database': 'Pfam', 'id': 'PF12249', 'properties': [{'key': 'EntryName', 'value': 'AftA_C'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF12250', 'properties': [{'key': 'EntryName', 'value': 'AftA_N'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MNPYGPSRVPEGEVYRPDRLNRKATLVAIVGAAILAFAFALVLWMGLKQTNLPAFGPSNVTRAVASATIAAVLIVTGFLTWLWLRDEHQSNPRWELEDVKPRPKWRTALTYLASYLSPAALVVAVLAIPLSATRLYLDGISVDQGFRTQFLTRMADDIGLSDMNYIDMPTFYPAGWFWLGGRLANLLGLPGWEAFQPWAIVSMAVAASVLVPVWQRITGSLPVATGIALVTTCIILAMNSEEPYAAIVAMGIPAMLVLASRIAKGDKFALAGGIIYLGVSATFYTLFTGAIALSAVAVCIVVAAIVQRSIKPLLWLAVLGGGSIVIALISWGPYLLASINGAERSGDSATHYLPLEGTQFPVPFLASSVVGLLCLVGLIYLVVRFHNNEVRAMWVGIAVFYAWMGMSMAITLLGNTLLGFRLDTVLVLIFATAGVLGIADFRLASVYQLYPTQITERTATHLTNLIVVLVLLGGLYYAQDLPQKNARAIDLAYTDTDGYGERADLYPAGAARYYKDINDHLLDQGFEPSETVVLTDELDFMSYYPYRGYQAFTSHYANPLGEFGNRNAFIEDLAIRSWDELADPQQFSDALNTSPWTIPEVFIFRGSIDDPDAGWKYDVAEDLYPNNPNVRFRGVYFNPESFDQMWQTKQVGPFVVVTHNE', 'length': 661, 'molWeight': 72570, 'crc64': 'B8F00CF82DE4EF65', 'md5': 'C9B39010315593B94B9A44F96F4905B8'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'PATHWAY': 1, 'SUBCELLULAR LOCATION': 1, 'DISRUPTION PHENOTYPE': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Transmembrane': 13}, 'uniParcId': 'UPI000003A067'}}}, {'from': 'CAF18782.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NTU1', 'uniProtkbId': 'MOAC_CORGL', 'entryAudit': {'firstPublicDate': '2002-09-19', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-09-19', 'entryVersion': 108, 'sequenceVersion': 1}, 'annotationScore': 3.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': 'Cyclic pyranopterin monophosphate synthase'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': '4.6.1.17'}]}, 'alternativeNames': [{'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': 'Molybdenum cofactor biosynthesis protein C'}}]}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': 'moaC'}, 'orderedLocusNames': [{'value': 'Cgl0211'}, {'value': 'cg0260'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': \"Catalyzes the conversion of (8S)-3',8-cyclo-7,8-dihydroguanosine 5'-triphosphate to cyclic pyranopterin monophosphate (cPMP)\"}], 'commentType': 'FUNCTION'}, {'commentType': 'CATALYTIC ACTIVITY', 'reaction': {'name': \"(8S)-3',8-cyclo-7,8-dihydroguanosine 5'-triphosphate = cyclic pyranopterin phosphate + diphosphate\", 'reactionCrossReferences': [{'database': 'Rhea', 'id': 'RHEA:49580'}, {'database': 'ChEBI', 'id': 'CHEBI:33019'}, {'database': 'ChEBI', 'id': 'CHEBI:59648'}, {'database': 'ChEBI', 'id': 'CHEBI:131766'}], 'ecNumber': '4.6.1.17', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': 'Cofactor biosynthesis; molybdopterin biosynthesis'}], 'commentType': 'PATHWAY'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': 'Homohexamer; trimer of dimers'}], 'commentType': 'SUBUNIT'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'value': 'Belongs to the MoaC family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 156, 'modifier': 'EXACT'}}, 'description': 'Cyclic pyranopterin monophosphate synthase', 'featureId': 'PRO_0000097796'}, {'type': 'Active site', 'location': {'start': {'value': 126, 'modifier': 'EXACT'}, 'end': {'value': 126, 'modifier': 'EXACT'}}, 'description': '', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}]}, {'type': 'Binding site', 'location': {'start': {'value': 75, 'modifier': 'EXACT'}, 'end': {'value': 77, 'modifier': 'EXACT'}}, 'description': '', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'ligand': {'name': 'substrate'}}, {'type': 'Binding site', 'location': {'start': {'value': 111, 'modifier': 'EXACT'}, 'end': {'value': 112, 'modifier': 'EXACT'}}, 'description': '', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01224'}], 'ligand': {'name': 'substrate'}}], 'keywords': [{'id': 'KW-0456', 'category': 'Molecular function', 'name': 'Lyase'}, {'id': 'KW-0501', 'category': 'Biological process', 'name': 'Molybdenum cofactor biosynthesis'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97604.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18782.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599464.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013475.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NTU1', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NTU1', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0260', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97604', 'properties': [{'key': 'ProteinId', 'value': 'BAB97604'}, {'key': 'GeneId', 'value': 'BAB97604'}]}, {'database': 'KEGG', 'id': 'cgb:cg0260', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0211', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.215', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0315', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_074693_1_1_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'EASCTGK', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9794429at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'UniPathway', 'id': 'UPA00344', 'properties': [{'key': 'RectionId', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0061799', 'properties': [{'key': 'GoTerm', 'value': 'F:cyclic pyranopterin monophosphate synthase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006777', 'properties': [{'key': 'GoTerm', 'value': 'P:Mo-molybdopterin cofactor biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'CDD', 'id': 'cd01420', 'properties': [{'key': 'EntryName', 'value': 'MoaC_PE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.30.70.640', 'properties': [{'key': 'EntryName', 'value': 'Molybdopterin cofactor biosynthesis C (MoaC) domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01224_B', 'properties': [{'key': 'EntryName', 'value': 'MoaC_B'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR023045', 'properties': [{'key': 'EntryName', 'value': 'Mo_CF_biosynth-C'}]}, {'database': 'InterPro', 'id': 'IPR036522', 'properties': [{'key': 'EntryName', 'value': 'MoaC_sf'}]}, {'database': 'InterPro', 'id': 'IPR002820', 'properties': [{'key': 'EntryName', 'value': 'Mopterin_CF_biosynth-C_dom'}]}, {'database': 'Pfam', 'id': 'PF01967', 'properties': [{'key': 'EntryName', 'value': 'MoaC'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF55040', 'properties': [{'key': 'EntryName', 'value': 'Molybdenum cofactor biosynthesis protein C, MoaC'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00581', 'properties': [{'key': 'EntryName', 'value': 'moaC'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MSELTHVRADGSAHMVDVTGKNETSRTAVAEGFVKMRGDVVKQLFSAGLPKGDALPVARIAGIMGAKKTPDIIPLCHPLPLGKITVDFFELTDGVRIEASVKTRGVTGVEMEALTAVSTAALTVYDMIKAVDKMAVIDGIRVLSKTGGKSGDWSVQ', 'length': 156, 'molWeight': 16438, 'crc64': '36BA4D6B8326DE64', 'md5': '80FDE8AB0AAE89C0FA8B422B6EB5F695'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'CATALYTIC ACTIVITY': 1, 'PATHWAY': 1, 'SUBUNIT': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Active site': 1, 'Binding site': 2}, 'uniParcId': 'UPI0000036497'}}}, {'from': 'CAF18789.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NTT4', 'uniProtkbId': 'PATR_CORGL', 'entryAudit': {'firstPublicDate': '2004-04-26', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 108, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'value': 'Probable phenylalanine aminotransferase'}, 'ecNumbers': [{'value': '2.6.1.-'}]}}, 'genes': [{'geneName': {'value': 'pat'}, 'orderedLocusNames': [{'value': 'Cgl0218'}, {'value': 'cg0267'}]}], 'comments': [{'texts': [{'value': 'May catalyze the transamination reaction in phenylalanine biosynthesis'}], 'commentType': 'FUNCTION'}, {'commentType': 'COFACTOR', 'cofactors': [{'name': \"pyridoxal 5'-phosphate\", 'evidences': [{'evidenceCode': 'ECO:0000250'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:597326'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000305'}], 'value': 'Belongs to the class-II pyridoxal-phosphate-dependent aminotransferase family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 341, 'modifier': 'EXACT'}}, 'description': 'Probable phenylalanine aminotransferase', 'featureId': 'PRO_0000153512'}, {'type': 'Modified residue', 'location': {'start': {'value': 213, 'modifier': 'EXACT'}, 'end': {'value': 213, 'modifier': 'EXACT'}}, 'description': 'N6-(pyridoxal phosphate)lysine', 'evidences': [{'evidenceCode': 'ECO:0000250'}]}], 'keywords': [{'id': 'KW-0032', 'category': 'Molecular function', 'name': 'Aminotransferase'}, {'id': 'KW-0663', 'category': 'Ligand', 'name': 'Pyridoxal phosphate'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0808', 'category': 'Molecular function', 'name': 'Transferase'}], 'references': [{'citation': {'id': '12948641', 'citationType': 'journal article', 'authors': ['McHardy A.C.', 'Tauch A.', 'Rueckert C.', 'Puehler A.', 'Kalinowski J.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948641'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00161-5'}], 'title': 'Genome-based analysis of biosynthetic aminotransferase genes of Corynebacterium glutamicum.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '229', 'lastPage': '240', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [GENOMIC DNA]', 'PROBABLE FUNCTION'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'AY238319', 'properties': [{'key': 'ProteinId', 'value': 'AAO92310.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97611.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18789.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599471.2', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013481.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NTT4', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NTT4', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0267', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'DNASU', 'id': '1021282', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97611', 'properties': [{'key': 'ProteinId', 'value': 'BAB97611'}, {'key': 'GeneId', 'value': 'BAB97611'}]}, {'database': 'KEGG', 'id': 'cgb:cg0267', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0218', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.222', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0079', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_017584_3_3_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'YPDMACT', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9809616at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0004400', 'properties': [{'key': 'GoTerm', 'value': 'F:histidinol-phosphate transaminase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0030170', 'properties': [{'key': 'GoTerm', 'value': 'F:pyridoxal phosphate binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0000105', 'properties': [{'key': 'GoTerm', 'value': 'P:histidine biosynthetic process'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'CDD', 'id': 'cd00609', 'properties': [{'key': 'EntryName', 'value': 'AAT_like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.90.1150.10', 'properties': [{'key': 'EntryName', 'value': 'Aspartate Aminotransferase, domain 1'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Gene3D', 'id': '3.40.640.10', 'properties': [{'key': 'EntryName', 'value': 'Type I PLP-dependent aspartate aminotransferase-like (Major domain)'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01023', 'properties': [{'key': 'EntryName', 'value': 'HisC_aminotrans_2'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01513', 'properties': [{'key': 'EntryName', 'value': 'Phe_aminotrans_2'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR001917', 'properties': [{'key': 'EntryName', 'value': 'Aminotrans_II_pyridoxalP_BS'}]}, {'database': 'InterPro', 'id': 'IPR004839', 'properties': [{'key': 'EntryName', 'value': 'Aminotransferase_I/II'}]}, {'database': 'InterPro', 'id': 'IPR005861', 'properties': [{'key': 'EntryName', 'value': 'HisP_aminotrans'}]}, {'database': 'InterPro', 'id': 'IPR024892', 'properties': [{'key': 'EntryName', 'value': 'Pat'}]}, {'database': 'InterPro', 'id': 'IPR015424', 'properties': [{'key': 'EntryName', 'value': 'PyrdxlP-dep_Trfase'}]}, {'database': 'InterPro', 'id': 'IPR015421', 'properties': [{'key': 'EntryName', 'value': 'PyrdxlP-dep_Trfase_major'}]}, {'database': 'InterPro', 'id': 'IPR015422', 'properties': [{'key': 'EntryName', 'value': 'PyrdxlP-dep_Trfase_small'}]}, {'database': 'PANTHER', 'id': 'PTHR43643:SF3', 'properties': [{'key': 'EntryName', 'value': 'HISTIDINOL-PHOSPHATE AMINOTRANSFERASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR43643', 'properties': [{'key': 'EntryName', 'value': 'HISTIDINOL-PHOSPHATE AMINOTRANSFERASE 2'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00155', 'properties': [{'key': 'EntryName', 'value': 'Aminotran_1_2'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF53383', 'properties': [{'key': 'EntryName', 'value': 'PLP-dependent transferases'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR01141', 'properties': [{'key': 'EntryName', 'value': 'hisC'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PROSITE', 'id': 'PS00599', 'properties': [{'key': 'EntryName', 'value': 'AA_TRANSFER_CLASS_2'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MIRADLATIPTYVPGRRLVDATKLSSNEVSFSPLPAAVDAVTEATWGANRYPDMGAVELREALAEHLEVEFDQVTVGCGSSALCQQLVQATCAQGDEVIFPWRSFEAYPIFAQVAGATPVAIPLTADQNHDLDAMAAAITDKTRLIFICNPNNPSGTTITQAQFDNFMEKVPNDVVVGLDEAYFEFNRADDTPVATEEIHRHDNVIGLRTFSKAYGLAGLRVGYAFGNAEIIAAMNKVAIPFAVNSAAQAAALASLNSADELMERVEETVEKRDAVVSALGAAPTQANFVWLPGEGAAELAAKLAEHGIVIRAFPEGARISVTNAEETDKLLRAWEAINAG', 'length': 341, 'molWeight': 36450, 'crc64': '5DC10FBEFFC1C67B', 'md5': 'FAD03231053FE8D7CA71FD46F7037435'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'COFACTOR': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Modified residue': 1}, 'uniParcId': 'UPI000003A082'}}}, {'from': 'CAF18807.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NTR6', 'secondaryAccessions': ['Q6M8B9'], 'uniProtkbId': 'GLUQ_CORGL', 'entryAudit': {'firstPublicDate': '2005-10-11', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 114, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': 'Glutamyl-Q tRNA(Asp) synthetase'}, 'shortNames': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': 'Glu-Q-RSs'}], 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': '6.1.1.-'}]}}, 'genes': [{'geneName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': 'gluQ'}, 'orderedLocusNames': [{'value': 'Cgl0236'}, {'value': 'cg0289'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': 'Catalyzes the tRNA-independent activation of glutamate in presence of ATP and the subsequent transfer of glutamate onto a tRNA(Asp). Glutamate is transferred on the 2-amino-5-(4,5-dihydroxy-2-cyclopenten-1-yl) moiety of the queuosine in the wobble position of the QUC anticodon'}], 'commentType': 'FUNCTION'}, {'commentType': 'COFACTOR', 'cofactors': [{'name': 'Zn(2+)', 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'cofactorCrossReference': {'database': 'ChEBI', 'id': 'CHEBI:29105'}}], 'note': {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': 'Binds 1 zinc ion per subunit.'}]}}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'value': 'Belongs to the class-I aminoacyl-tRNA synthetase family. GluQ subfamily'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 293, 'modifier': 'EXACT'}}, 'description': 'Glutamyl-Q tRNA(Asp) synthetase', 'featureId': 'PRO_0000208296'}, {'type': 'Motif', 'location': {'start': {'value': 7, 'modifier': 'EXACT'}, 'end': {'value': 17, 'modifier': 'EXACT'}}, 'description': \"'HIGH' region\"}, {'type': 'Motif', 'location': {'start': {'value': 236, 'modifier': 'EXACT'}, 'end': {'value': 240, 'modifier': 'EXACT'}}, 'description': \"'KMSKS' region\"}, {'type': 'Binding site', 'location': {'start': {'value': 4, 'modifier': 'EXACT'}, 'end': {'value': 8, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29985'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'L-glutamate', 'id': 'ChEBI:CHEBI:29985'}}, {'type': 'Binding site', 'location': {'start': {'value': 40, 'modifier': 'EXACT'}, 'end': {'value': 40, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29985'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'L-glutamate', 'id': 'ChEBI:CHEBI:29985'}}, {'type': 'Binding site', 'location': {'start': {'value': 92, 'modifier': 'EXACT'}, 'end': {'value': 92, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29105'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'Zn(2+)', 'id': 'ChEBI:CHEBI:29105'}}, {'type': 'Binding site', 'location': {'start': {'value': 94, 'modifier': 'EXACT'}, 'end': {'value': 94, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29105'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'Zn(2+)', 'id': 'ChEBI:CHEBI:29105'}}, {'type': 'Binding site', 'location': {'start': {'value': 113, 'modifier': 'EXACT'}, 'end': {'value': 113, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29105'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'Zn(2+)', 'id': 'ChEBI:CHEBI:29105'}}, {'type': 'Binding site', 'location': {'start': {'value': 117, 'modifier': 'EXACT'}, 'end': {'value': 117, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29105'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'Zn(2+)', 'id': 'ChEBI:CHEBI:29105'}}, {'type': 'Binding site', 'location': {'start': {'value': 180, 'modifier': 'EXACT'}, 'end': {'value': 180, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29985'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'L-glutamate', 'id': 'ChEBI:CHEBI:29985'}}, {'type': 'Binding site', 'location': {'start': {'value': 198, 'modifier': 'EXACT'}, 'end': {'value': 198, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:29985'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'L-glutamate', 'id': 'ChEBI:CHEBI:29985'}}, {'type': 'Binding site', 'location': {'start': {'value': 239, 'modifier': 'EXACT'}, 'end': {'value': 239, 'modifier': 'EXACT'}}, 'description': '', 'featureCrossReferences': [{'database': 'ChEBI', 'id': 'CHEBI:30616'}], 'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_01428'}], 'ligand': {'name': 'ATP', 'id': 'ChEBI:CHEBI:30616'}}], 'keywords': [{'id': 'KW-0030', 'category': 'Molecular function', 'name': 'Aminoacyl-tRNA synthetase'}, {'id': 'KW-0067', 'category': 'Ligand', 'name': 'ATP-binding'}, {'id': 'KW-0436', 'category': 'Molecular function', 'name': 'Ligase'}, {'id': 'KW-0479', 'category': 'Ligand', 'name': 'Metal-binding'}, {'id': 'KW-0547', 'category': 'Ligand', 'name': 'Nucleotide-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}, {'id': 'KW-0862', 'category': 'Ligand', 'name': 'Zinc'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97629.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18807.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599489.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_011013493.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NTR6', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NTR6', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0289', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97629', 'properties': [{'key': 'ProteinId', 'value': 'BAB97629'}, {'key': 'GeneId', 'value': 'BAB97629'}]}, {'database': 'KEGG', 'id': 'cgb:cg0289', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0236', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.241', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0008', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_015768_0_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'WLLRMED', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9807503at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0004812', 'properties': [{'key': 'GoTerm', 'value': 'F:aminoacyl-tRNA ligase activity'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0005524', 'properties': [{'key': 'GoTerm', 'value': 'F:ATP binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]}, {'database': 'GO', 'id': 'GO:0008270', 'properties': [{'key': 'GoTerm', 'value': 'F:zinc ion binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0006424', 'properties': [{'key': 'GoTerm', 'value': 'P:glutamyl-tRNA aminoacylation'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'GO', 'id': 'GO:0006400', 'properties': [{'key': 'GoTerm', 'value': 'P:tRNA modification'}, {'key': 'GoEvidenceType', 'value': 'IEA:InterPro'}]}, {'database': 'Gene3D', 'id': '3.40.50.620', 'properties': [{'key': 'EntryName', 'value': 'HUPs'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_01428', 'properties': [{'key': 'EntryName', 'value': 'Glu_Q_tRNA_synth'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR022380', 'properties': [{'key': 'EntryName', 'value': 'Glu-Q_tRNA(Asp)_Synthase'}]}, {'database': 'InterPro', 'id': 'IPR000924', 'properties': [{'key': 'EntryName', 'value': 'Glu/Gln-tRNA-synth'}]}, {'database': 'InterPro', 'id': 'IPR020058', 'properties': [{'key': 'EntryName', 'value': 'Glu/Gln-tRNA-synth_Ib_cat-dom'}]}, {'database': 'InterPro', 'id': 'IPR014729', 'properties': [{'key': 'EntryName', 'value': 'Rossmann-like_a/b/a_fold'}]}, {'database': 'PANTHER', 'id': 'PTHR43311', 'properties': [{'key': 'EntryName', 'value': 'GLUTAMATE--TRNA LIGASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR43311:SF1', 'properties': [{'key': 'EntryName', 'value': 'GLUTAMYL-Q TRNA(ASP) SYNTHETASE'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF00749', 'properties': [{'key': 'EntryName', 'value': 'tRNA-synt_1c'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PRINTS', 'id': 'PR00987', 'properties': [{'key': 'EntryName', 'value': 'TRNASYNTHGLU'}]}, {'database': 'SUPFAM', 'id': 'SSF52374', 'properties': [{'key': 'EntryName', 'value': 'Nucleotidylyl transferase'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR03838', 'properties': [{'key': 'EntryName', 'value': 'queuosine_YadB'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MAGRYAPSPSGDLHFGNLRTALLAWLFARSEGKKFLMRVEDIDEQRSSKESAESQLADLSALGLDWDGDVLYQSTRYDAYRAALEKLDTYECYCSRRDIQEASRAPHVAPGVYPGTCRGLKEEERVEKRATLAAQNRHPAIRLRAQVTSFDFHDRLRGPQTGPVDDFILLRGGQEPGWAYNLAVVVDDAYQGVDQVVRGDDLLDSAARQAYLGSLLGTPAPEYIHVPLVLNAHGQRLAKRDGAVTLKEMLIDAPLHTIFSRLASSLGYEGVNSAPQLLEIFDPTTLSREPFIY', 'length': 293, 'molWeight': 32552, 'crc64': 'AF73DCC9E1BD8F09', 'md5': '8F5CA0F186B18F9FDBEEEDA933303394'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'COFACTOR': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1, 'Motif': 2, 'Binding site': 9}, 'uniParcId': 'UPI000003A096'}}}, {'from': 'CAF18814.1', 'to': {'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'Q8NTQ9', 'uniProtkbId': 'Y243_CORGL', 'entryAudit': {'firstPublicDate': '2002-11-08', 'lastAnnotationUpdateDate': '2023-06-28', 'lastSequenceUpdateDate': '2002-10-01', 'entryVersion': 104, 'sequenceVersion': 1}, 'annotationScore': 2.0, 'organism': {'scientificName': 'Corynebacterium glutamicum (strain ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025)', 'taxonId': 196627, 'lineage': ['Bacteria', 'Actinomycetota', 'Actinomycetes', 'Mycobacteriales', 'Corynebacteriaceae', 'Corynebacterium']}, 'proteinExistence': '3: Inferred from homology', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00274'}], 'value': 'Nucleoid-associated protein Cgl0243/cg0297'}}}, 'genes': [{'orderedLocusNames': [{'value': 'Cgl0243'}, {'value': 'cg0297'}]}], 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00274'}], 'value': 'Binds to DNA and alters its conformation. May be involved in regulation of gene expression, nucleoid organization and DNA protection'}], 'commentType': 'FUNCTION'}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00274'}], 'value': 'Homodimer'}], 'commentType': 'SUBUNIT'}, {'commentType': 'SUBCELLULAR LOCATION', 'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00274'}], 'value': 'Cytoplasm, nucleoid', 'id': 'SL-0187'}}]}, {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000255', 'source': 'HAMAP-Rule', 'id': 'MF_00274'}], 'value': 'Belongs to the YbaB/EbfC family'}], 'commentType': 'SIMILARITY'}], 'features': [{'type': 'Chain', 'location': {'start': {'value': 1, 'modifier': 'EXACT'}, 'end': {'value': 103, 'modifier': 'EXACT'}}, 'description': 'Nucleoid-associated protein Cgl0243/cg0297', 'featureId': 'PRO_0000170388'}], 'keywords': [{'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'}, {'id': 'KW-0238', 'category': 'Molecular function', 'name': 'DNA-binding'}, {'id': 'KW-1185', 'category': 'Technical term', 'name': 'Reference proteome'}], 'references': [{'citation': {'id': '12743753', 'citationType': 'journal article', 'authors': ['Ikeda M.', 'Nakagawa S.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12743753'}, {'database': 'DOI', 'id': '10.1007/s00253-003-1328-1'}], 'title': 'The Corynebacterium glutamicum genome: features and impacts on biotechnological processes.', 'publicationDate': '2003', 'journal': 'Appl. Microbiol. Biotechnol.', 'firstPage': '99', 'lastPage': '109', 'volume': '62'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}, {'citation': {'id': '12948626', 'citationType': 'journal article', 'authors': ['Kalinowski J.', 'Bathe B.', 'Bartels D.', 'Bischoff N.', 'Bott M.', 'Burkovski A.', 'Dusch N.', 'Eggeling L.', 'Eikmanns B.J.', 'Gaigalat L.', 'Goesmann A.', 'Hartmann M.', 'Huthmacher K.', 'Kraemer R.', 'Linke B.', 'McHardy A.C.', 'Meyer F.', 'Moeckel B.', 'Pfefferle W.', 'Puehler A.', 'Rey D.A.', 'Rueckert C.', 'Rupp O.', 'Sahm H.', 'Wendisch V.F.', 'Wiegraebe I.', 'Tauch A.'], 'citationCrossReferences': [{'database': 'PubMed', 'id': '12948626'}, {'database': 'DOI', 'id': '10.1016/s0168-1656(03)00154-8'}], 'title': 'The complete Corynebacterium glutamicum ATCC 13032 genome sequence and its impact on the production of L-aspartate-derived amino acids and vitamins.', 'publicationDate': '2003', 'journal': 'J. Biotechnol.', 'firstPage': '5', 'lastPage': '25', 'volume': '104'}, 'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]'], 'referenceComments': [{'value': 'ATCC 13032 / DSM 20300 / BCRC 11384 / JCM 1318 / LMG 3730 / NCIMB 10025', 'type': 'STRAIN'}]}], 'uniProtKBCrossReferences': [{'database': 'EMBL', 'id': 'BA000036', 'properties': [{'key': 'ProteinId', 'value': 'BAB97636.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'EMBL', 'id': 'BX927148', 'properties': [{'key': 'ProteinId', 'value': 'CAF18814.1'}, {'key': 'Status', 'value': '-'}, {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]}, {'database': 'RefSeq', 'id': 'NP_599496.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_003450.3'}]}, {'database': 'RefSeq', 'id': 'WP_003855676.1', 'properties': [{'key': 'NucleotideSequenceId', 'value': 'NC_006958.1'}]}, {'database': 'AlphaFoldDB', 'id': 'Q8NTQ9', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'SMR', 'id': 'Q8NTQ9', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'STRING', 'id': '196627.cg0297', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'EnsemblBacteria', 'id': 'BAB97636', 'properties': [{'key': 'ProteinId', 'value': 'BAB97636'}, {'key': 'GeneId', 'value': 'BAB97636'}]}, {'database': 'GeneID', 'id': '69620702', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgb:cg0297', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'KEGG', 'id': 'cgl:Cgl0243', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'PATRIC', 'id': 'fig|196627.13.peg.248', 'properties': [{'key': 'GeneDesignation', 'value': '-'}]}, {'database': 'eggNOG', 'id': 'COG0718', 'properties': [{'key': 'ToxonomicScope', 'value': 'Bacteria'}]}, {'database': 'HOGENOM', 'id': 'CLU_140930_4_0_11', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'OMA', 'id': 'MGNMMKQ', 'properties': [{'key': 'Fingerprint', 'value': '-'}]}, {'database': 'OrthoDB', 'id': '9809370at2', 'properties': [{'key': 'Description', 'value': '-'}]}, {'database': 'Proteomes', 'id': 'UP000000582', 'properties': [{'key': 'Component', 'value': 'Chromosome'}]}, {'database': 'GO', 'id': 'GO:0043590', 'properties': [{'key': 'GoTerm', 'value': 'C:bacterial nucleoid'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0005737', 'properties': [{'key': 'GoTerm', 'value': 'C:cytoplasm'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'GO', 'id': 'GO:0003677', 'properties': [{'key': 'GoTerm', 'value': 'F:DNA binding'}, {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-UniRule'}]}, {'database': 'Gene3D', 'id': '3.30.1310.10', 'properties': [{'key': 'EntryName', 'value': 'Nucleoid-associated protein YbaB-like domain'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'HAMAP', 'id': 'MF_00274', 'properties': [{'key': 'EntryName', 'value': 'DNA_YbaB_EbfC'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'InterPro', 'id': 'IPR036894', 'properties': [{'key': 'EntryName', 'value': 'YbaB-like_sf'}]}, {'database': 'InterPro', 'id': 'IPR004401', 'properties': [{'key': 'EntryName', 'value': 'YbaB/EbfC'}]}, {'database': 'PANTHER', 'id': 'PTHR33449', 'properties': [{'key': 'EntryName', 'value': 'NUCLEOID-ASSOCIATED PROTEIN YBAB'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PANTHER', 'id': 'PTHR33449:SF1', 'properties': [{'key': 'EntryName', 'value': 'NUCLEOID-ASSOCIATED PROTEIN YBAB'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'Pfam', 'id': 'PF02575', 'properties': [{'key': 'EntryName', 'value': 'YbaB_DNA_bd'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'PIRSF', 'id': 'PIRSF004555', 'properties': [{'key': 'EntryName', 'value': 'UCP004555'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'SUPFAM', 'id': 'SSF82607', 'properties': [{'key': 'EntryName', 'value': 'YbaB-like'}, {'key': 'MatchStatus', 'value': '1'}]}, {'database': 'TIGRFAMs', 'id': 'TIGR00103', 'properties': [{'key': 'EntryName', 'value': 'DNA_YbaB_EbfC'}, {'key': 'MatchStatus', 'value': '1'}]}], 'sequence': {'value': 'MTQPDMSQILAQAQQMQAQLQAAQQEILATTVVGNAGNGLVTVTMAGNGEVSAVTVDPKVVDPEDVETLQDLLLGAFKDAHNKVANVAEEKMGPLSQGMGGLF', 'length': 103, 'molWeight': 10722, 'crc64': '6A774D6328CA10E9', 'md5': '96EC189A4193962DD7500C6378D4F75A'}, 'extraAttributes': {'countByCommentType': {'FUNCTION': 1, 'SUBUNIT': 1, 'SUBCELLULAR LOCATION': 1, 'SIMILARITY': 1}, 'countByFeatureType': {'Chain': 1}, 'uniParcId': 'UPI000003A09F'}}}]}\n"
     ]
    }
   ],
   "source": [
    "#mapping_uniprot = uniprot_id_mapping(DF_annot.ncbi_protein.fillna(''),input_id='EMBL',output_id='ACC',\n",
    "#                             input_name='ncbi_protein',output_name='uniprot')\n",
    "#mapping_uniprot.head()\n",
    "\n",
    "\n",
    "#mapping_uniprot = uniprot_id_mapping(DF_annot.ncbi_protein.fillna(''),input_id='P_REFSEQ_AC',output_id='ACC',\n",
    "#                                     input_name='ncbi_protein',output_name='uniprot')\n",
    "#mapping_uniprot.head()\n",
    "\n",
    "import json\n",
    "import requests\n",
    "import time\n",
    "\n",
    "URL = 'https://rest.uniprot.org/idmapping'\n",
    "#IDS = DF_annot.index.fillna('').values.tolist()\n",
    "IDS = DF_annot.ncbi_protein.fillna('').values.tolist()\n",
    "\n",
    "\n",
    "params = {\n",
    "   'from': 'EMBL-GenBank-DDBJ_CDS',\n",
    "   'to': 'UniProtKB',\n",
    "   'ids': ' '.join(IDS)\n",
    "}\n",
    "\n",
    "response = requests.post(f'{URL}/run', params)\n",
    "\n",
    "job_id = response.json()['jobId']\n",
    "job_status = requests.get(f'{URL}/status/{job_id}')\n",
    "d = job_status.json()\n",
    "\n",
    "# Make three attemps to get the results\n",
    "for i in range(3):\n",
    "    if d.get(\"job_status\") == 'FINISHED' or d.get('results'):\n",
    "        job_results = requests.get(f'{URL}/results/{job_id}')\n",
    "        results = job_results.json()\n",
    "        for obj in results['results']:\n",
    "            print(f'{obj[\"from\"]}\\t{obj[\"to\"]}')\n",
    "        break\n",
    "    time.sleep(1)\n",
    "print(IDS)\n",
    "print(response.json())\n",
    "print(job_status.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'mapping_uniprot' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/qf/5c8btmkn3pz76_zg3zmb_11h0000gn/T/ipykernel_84634/3674106361.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Merge with current annotation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mDF_annot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mDF_annot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmapping_uniprot\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'ncbi_protein'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mDF_annot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'locus_tag'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minplace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mDF_annot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'mapping_uniprot' is not defined"
     ]
    }
   ],
   "source": [
    "# Merge with current annotation\n",
    "DF_annot = pd.merge(DF_annot.reset_index(),mapping_uniprot,how='left',on='ncbi_protein')\n",
    "DF_annot.set_index('locus_tag',inplace=True)\n",
    "DF_annot.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add Biocyc Operon information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To obtain operon information from Biocyc, follow the steps below\n",
    "\n",
    "1. Go to [Biocyc.org](https://biocyc.org/) (you may need to create an account and/or login)\n",
    "2. Change the organism database to your organism/strain\n",
    "3. Select **SmartTables** -> **Special SmartTables**\n",
    "4. Select **\"All genes of \\<organism\\>\"**\n",
    "5. Select the **\"Gene Name\"** column\n",
    "6. Under **\"ADD TRANSFORM COLUMN\"** select **\"Genes in same transcription unit\"**\n",
    "7. Select the **\"Genes in same transcription unit\"** column\n",
    "8. Under **\"ADD PROPERTY COLUMN\"** select **\"Accession-1\"**\n",
    "9. Under **OPERATIONS**, select **\"Export\"** -> **\"to Spreadsheet File...\"**\n",
    "10. Select **\"common names\"** and click **\"Export smarttable\"**\n",
    "11. Add file location below and run the code cell\n",
    "\n",
    "Summary of Corynebacterium glutamicum DSM 20300 = ATCC 13032, version 27.0\n",
    "Tier 3 Uncurated Database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "biocyc_file = os.path.join('..','data','external','biocyc_annotations.txt')\n",
    "\n",
    "DF_biocyc = pd.read_csv(biocyc_file,sep='\\t')\n",
    "\n",
    "# Remove genes with no accession\n",
    "DF_biocyc = DF_biocyc[DF_biocyc['Accession-1'].notnull()]\n",
    "\n",
    "# Set the accession (i.e. locus tag) as index\n",
    "DF_biocyc = DF_biocyc.set_index('Accession-1').sort_values('Left-End-Position')\n",
    "\n",
    "# Specific for B. subtilis: Fix locus tags\n",
    "DF_biocyc.index = DF_biocyc.index.str.replace('BSU','BSU_')\n",
    "\n",
    "# Only keep genes in the final annotation file\n",
    "DF_biocyc = DF_biocyc.reindex(DF_annot.index)\n",
    "\n",
    "# Reformat transcription units\n",
    "DF_biocyc['operon_list'] = DF_biocyc['Accession-1.1'].apply(reformat_biocyc_tu)\n",
    "\n",
    "# Fill None with locus tags\n",
    "DF_biocyc['operon_list'].fillna(DF_biocyc.index.to_series(), inplace=True)\n",
    "\n",
    "DF_biocyc.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Assign unique IDs to operons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following code assigns unique names to each operon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all operons\n",
    "operons = DF_biocyc['operon_list'].unique()\n",
    "\n",
    "# Map each operon to a unique string\n",
    "operon_dict = {operon: \"Op\"+str(i) for i, operon in enumerate(operons)}\n",
    "\n",
    "# Add names to dataframe\n",
    "DF_biocyc['operon'] = [operon_dict[op] for op in DF_biocyc[\"operon_list\"]]\n",
    "\n",
    "DF_biocyc.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, merge the Biocyc information with the main annotation DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_annot['operon'] = DF_biocyc['operon']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and save annotation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, we will re-order the annotation columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if 'old_locus_tag' in DF_annot.columns:\n",
    "    order = ['gene_name','accession','old_locus_tag','start','end','strand','gene_product','COG','uniprot','operon']\n",
    "else:\n",
    "    order = ['gene_name','accession','start','end','strand','gene_product','COG','uniprot','operon']\n",
    "    \n",
    "DF_annot = DF_annot[order]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_annot.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final statistics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following graphs show how much information is available for the organism."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "sns.set_style('ticks')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig,ax = plt.subplots()\n",
    "DF_annot.count().plot(kind='bar',ax=ax)\n",
    "ax.set_ylabel('# of Values',fontsize=18)\n",
    "ax.tick_params(labelsize=16)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fill missing values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some organisms are missing gene names, so these will be filled with locus tag gene names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Fill in missing gene names with locus tag names\n",
    "DF_annot['tmp_name'] = DF_annot.copy().index.tolist()\n",
    "DF_annot.gene_name.fillna(DF_annot.tmp_name,inplace=True)\n",
    "DF_annot.drop('tmp_name',axis=1,inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " COG letters will also be converted to the full name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fill missing COGs with X\n",
    "DF_annot['COG'].fillna('X',inplace=True)\n",
    "\n",
    "# Change single letter COG annotation to full description\n",
    "DF_annot['COG'] = DF_annot.COG.apply(cog2str)\n",
    "\n",
    "counts = DF_annot.COG.value_counts()\n",
    "plt.pie(counts.values,labels=counts.index);"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Uncomment the following line to save the gene annotation dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_annot.to_csv(os.path.join('..','data','processed_data','gene_info.csv'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GO Annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To start, download the GO Annotations for your organism from AmiGO 2\n",
    "\n",
    "1. Go to [AmiGO 2](http://amigo.geneontology.org/amigo/search/annotation)\n",
    "1. Filter for your organism\n",
    "1. Click ``CustomDL``\n",
    "1. Drag ``GO class (direct)`` to the end of your Selected Fields\n",
    "1. Enter the location of your GO annotation file below and run the following code block"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "go_file = os.path.join('..','data','external','GO_annotations.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_GO = pd.read_csv(go_file,sep='\\t',header=None,usecols=[2,17])\n",
    "DF_GO.columns = ['gene_name','gene_ontology']\n",
    "DF_GO.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Convert the gene names to gene locus tags, and drop gene names that cannot be converted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "name2num = {v:k for k,v in DF_annot.gene_name.to_dict().items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_GO['gene_id'] = [name2num[x] if x in name2num.keys() else None for x in DF_GO.gene_name]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_GO.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we remove null entries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_GO = DF_GO[DF_GO.gene_id.notnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_GO.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Uncomment the line below to save the annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DF_GO[['gene_id','gene_name','gene_ontology']].to_csv(os.path.join('..','data','external','GO_annotations_curated.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
